// maz-ui/components/MazAnimatedCounter.mjs
import { defineComponent as v, useCssVars as _, useSlots as h, computed as l, ref as b, watch as C, openBlock as p, createElementBlock as m, normalizeClass as $, createElementVNode as A, renderSlot as f, createTextVNode as o, toDisplayString as a, createCommentVNode as d } from "vue";
import './assets/MazAnimatedCounter.css';const S = { class: "maz-sr-only" }, g = {
  key: 0,
  class: "m-animated-counter__fix"
}, k = /* @__PURE__ */ v({
  __name: "MazAnimatedCounter",
  props: {
    count: {},
    duration: { default: 1e3 },
    prefix: { default: void 0 },
    suffix: { default: void 0 }
  },
  setup(n) {
    _((e) => ({
      "7cf84c3b": x.value,
      a2b7049a: e.count
    }));
    const t = n, s = h(), r = l(() => !!t.prefix || !!s.prefix), u = l(() => !!t.suffix || !!s.suffix), i = b(!0);
    C(
      () => t.count,
      (e, c) => {
        e !== c && (i.value = !1, setTimeout(() => {
          i.value = !0;
        }, 100));
      }
    );
    const x = l(() => `${t.duration}ms`);
    return (e, c) => (p(), m(
      "span",
      {
        class: $(["m-animated-counter", {
          "--animated": i.value,
          "--prefixed": r.value,
          "--suffixed": u.value
        }])
      },
      [
        A("span", S, [
          f(e.$slots, "prefix", {}, () => [
            o(
              a(e.prefix),
              1
              /* TEXT */
            )
          ], !0),
          o(
            " " + a(e.count) + " ",
            1
            /* TEXT */
          ),
          f(e.$slots, "suffix", {}, () => [
            o(
              a(e.suffix),
              1
              /* TEXT */
            )
          ], !0)
        ]),
        r.value || u.value ? (p(), m("span", g, [
          d(' @slot Prefix slot - Add a prefix next to the number (e.g: "$") '),
          f(e.$slots, "prefix", {}, () => [
            o(
              a(e.prefix),
              1
              /* TEXT */
            )
          ], !0),
          d(' @slot Suffix slot - Add a suffix next to the number (e.g: "%") '),
          f(e.$slots, "suffix", {}, () => [
            o(
              a(e.suffix),
              1
              /* TEXT */
            )
          ], !0)
        ])) : d("v-if", !0)
      ],
      2
      /* CLASS */
    ));
  }
}), y = (n, t) => {
  const s = n.__vccOpts || n;
  for (const [r, u] of t)
    s[r] = u;
  return s;
}, V = /* @__PURE__ */ y(k, [["__scopeId", "data-v-95ad1bb8"]]);
export {
  V as default
};
