// maz-ui/components/MazAccordion.mjs
import { defineComponent as C, mergeModels as A, useCssVars as V, useModel as x, ref as p, onMounted as O, onUnmounted as B, openBlock as f, createElementBlock as b, normalizeClass as g, createElementVNode as u, mergeProps as I, createCommentVNode as L, renderSlot as h, computed as $, normalizeStyle as w, withDirectives as E, vShow as z, getCurrentInstance as N, useSlots as j, Fragment as D, renderList as F, createBlock as R, withCtx as M, unref as y, createVNode as S } from "vue";
import './assets/MazAccordion.css';const P = ["aria-hidden"], T = /* @__PURE__ */ C({
  inheritAttrs: !1,
  __name: "MazExpandAnimation",
  props: /* @__PURE__ */ A({
    duration: { default: "300ms" },
    timingFunction: { default: "ease-in-out" }
  }, {
    modelValue: { type: Boolean },
    modelModifiers: {}
  }),
  emits: ["update:modelValue"],
  setup(r) {
    V((n) => ({
      "3628b947": n.duration,
      "239277b5": n.timingFunction
    }));
    const a = x(r, "modelValue"), t = p(!a.value), l = p();
    function i() {
      a.value || (t.value = !0);
    }
    function m() {
      a.value && (t.value = !1);
    }
    return O(() => {
      var n, s;
      (n = l.value) == null || n.addEventListener("transitionstart", i, !1), (s = l.value) == null || s.addEventListener("transitionend", m, !1);
    }), B(() => {
      var n, s;
      (n = l.value) == null || n.removeEventListener("transitionstart", i, !1), (s = l.value) == null || s.removeEventListener("transitionend", m, !1);
    }), (n, s) => (f(), b("div", {
      ref_key: "expandAnimationRef",
      ref: l,
      class: g(["m-expand-animation", { "m-expand-animation--expanded": a.value }]),
      "aria-hidden": !a.value,
      role: "region"
    }, [
      u(
        "div",
        I({ class: "m-expand-animation__inner" }, n.$attrs, {
          class: {
            "--overflow-hidden": t.value
          }
        }),
        [
          L(" @slot Default Slot - Display content "),
          h(n.$slots, "default", {}, void 0, !0)
        ],
        16
        /* FULL_PROPS */
      )
    ], 10, P));
  }
}), k = (r, a) => {
  const t = r.__vccOpts || r;
  for (const [l, i] of a)
    t[l] = i;
  return t;
}, U = /* @__PURE__ */ k(T, [["__scopeId", "data-v-b7ca6b97"]]), q = { class: "inner" }, H = /* @__PURE__ */ C({
  __name: "MazCardSpotlight",
  props: {
    color: { default: "primary" },
    noElevation: { type: Boolean, default: !1 },
    padding: { type: Boolean, default: !0 },
    contentClass: { default: void 0 },
    contentStyle: { type: [Boolean, null, String, Object, Array], default: void 0 },
    innerOpacity: { default: 0.95 }
  },
  setup(r) {
    V((d) => ({
      "4157dd1a": d.innerOpacity,
      "22a94e2a": n.value
    }));
    const a = r, t = p(), l = p(), i = p(!1);
    function m({ clientX: d, clientY: c }) {
      var e, _, o;
      i.value = !0;
      const v = (e = l.value) == null ? void 0 : e.getBoundingClientRect();
      v && ((o = (_ = t.value) == null ? void 0 : _.animate) == null || o.call(
        _,
        [
          {
            transform: `translate(${d - v.left - v.width / 2}px,${c - v.top - v.height / 2}px)`
          }
        ],
        {
          duration: 300,
          fill: "forwards"
        }
      ));
    }
    const n = $(() => `var(--maz-color-${a.color}-alpha)`), s = $(() => `var(--maz-color-${a.color}-alpha-20)`);
    return O(() => {
      window.addEventListener("mousemove", m);
    }), B(() => {
      window.removeEventListener("mousemove", m);
    }), (d, c) => (f(), b(
      "div",
      {
        class: g(["m-card-spotlight", { "maz-elevation": !d.noElevation }]),
        style: w({ backgroundColor: s.value })
      },
      [
        u("div", q, [
          u(
            "div",
            {
              class: g(["content", [{ "maz-p-4": d.padding }, d.contentClass]])
            },
            [
              h(d.$slots, "default", {}, void 0, !0)
            ],
            2
            /* CLASS */
          )
        ]),
        E(u(
          "div",
          {
            ref_key: "blobElement",
            ref: t,
            class: "blob",
            style: w({ backgroundColor: n.value })
          },
          null,
          4
          /* STYLE */
        ), [
          [z, i.value]
        ]),
        E(u(
          "div",
          {
            ref_key: "fakeblobElement",
            ref: l,
            class: "fakeblob"
          },
          null,
          512
          /* NEED_PATCH */
        ), [
          [z, i.value]
        ])
      ],
      6
      /* CLASS, STYLE */
    ));
  }
}), W = /* @__PURE__ */ k(H, [["__scopeId", "data-v-5d8ac6ae"]]), G = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "1em",
  height: "1em",
  fill: "none",
  viewBox: "0 0 24 24"
}, J = /* @__PURE__ */ u(
  "path",
  {
    stroke: "currentColor",
    "stroke-linecap": "round",
    "stroke-linejoin": "round",
    "stroke-width": "1.5",
    d: "M12 4.5v15m7.5-7.5h-15"
  },
  null,
  -1
  /* HOISTED */
), K = [
  J
];
function Q(r, a) {
  return f(), b("svg", G, [...K]);
}
const X = { render: Q };
function Y({
  componentName: r,
  providedId: a
}) {
  const t = N();
  return $(() => a ?? `${r}-${t == null ? void 0 : t.uid}`);
}
const Z = {
  class: "m-accordion",
  role: "presentation"
}, ee = ["id", "aria-controls", "aria-expanded", "onClick"], te = /* @__PURE__ */ C({
  __name: "MazAccordion",
  props: {
    id: { default: void 0 },
    modelValue: { default: 0 },
    contentClass: { default: void 0 }
  },
  emits: ["update:model-value"],
  setup(r, { emit: a }) {
    const t = r, l = a, i = Y({
      componentName: "MazAccordion",
      providedId: t.id
    }), m = j(), n = $(
      () => Object.keys(m).filter((e) => e.startsWith("title-")).length
    ), s = p(t.modelValue), d = $({
      get: () => t.modelValue || s.value,
      set: (e) => {
        s.value = e, l("update:model-value", e);
      }
    });
    function c(e) {
      return d.value === e;
    }
    function v(e) {
      d.value === e ? (s.value = 0, l("update:model-value", 0)) : (s.value = e, l("update:model-value", e));
    }
    return (e, _) => (f(), b("div", Z, [
      (f(!0), b(
        D,
        null,
        F(n.value, (o) => (f(), R(
          W,
          {
            key: o,
            class: "m-accordion__spotlight",
            padding: !1
          },
          {
            default: M(() => [
              u("button", {
                id: `step-${o}-${y(i)}`,
                class: "m-accordion__header",
                "aria-controls": `step-${o}-${y(i)}`,
                "aria-expanded": c(o),
                onClick: (ne) => v(o)
              }, [
                h(e.$slots, `title-${o}`, {
                  isOpen: c(o)
                }, void 0, !0),
                S(y(X), {
                  class: g(["header-icon", { "--rotate": c(o) }])
                }, null, 8, ["class"])
              ], 8, ee),
              S(U, {
                "animation-duration": "300ms",
                "model-value": c(o),
                "aria-labelledby": `step-${o}-${y(i)}`
              }, {
                default: M(() => [
                  u(
                    "div",
                    {
                      class: g(["m-accordion__content", e.contentClass])
                    },
                    [
                      h(e.$slots, "content", {
                        isOpen: c(o)
                      }, void 0, !0),
                      h(e.$slots, `content-${o}`, {
                        isOpen: c(o)
                      }, void 0, !0)
                    ],
                    2
                    /* CLASS */
                  )
                ]),
                _: 2
                /* DYNAMIC */
              }, 1032, ["model-value", "aria-labelledby"])
            ]),
            _: 2
            /* DYNAMIC */
          },
          1024
          /* DYNAMIC_SLOTS */
        ))),
        128
        /* KEYED_FRAGMENT */
      ))
    ]));
  }
}), ae = /* @__PURE__ */ k(te, [["__scopeId", "data-v-3413d328"]]);
export {
  ae as default
};
