// maz-ui/components/MazFullscreenLoader.mjs
import { defineComponent as s, defineAsyncComponent as l, onMounted as m, onUnmounted as c, openBlock as o, createElementBlock as n, createVNode as d, unref as u, renderSlot as i, createCommentVNode as f } from "vue";
import './assets/MazFullscreenLoader.css';const p = { class: "maz-fixed maz-inset-0 maz-z-default-backdrop maz-flex maz-flex-col maz-gap-2 maz-bg-overlay maz-text-center maz-backdrop-blur maz-flex-center" }, z = { key: 0 }, y = /* @__PURE__ */ s({
  __name: "MazFullscreenLoader",
  props: {
    color: { default: "primary" },
    size: { default: "3em" }
  },
  setup(_) {
    const t = l(() => import("./chunks/MazSpinner-nH44Htog.mjs"));
    function a() {
      document.documentElement.classList.add("--maz-fullscreen-loader-present");
    }
    async function r() {
      document.documentElement.classList.remove("--maz-fullscreen-loader-present");
    }
    return m(() => {
      a();
    }), c(() => {
      r();
    }), (e, k) => (o(), n("div", p, [
      d(u(t), {
        color: e.color,
        size: e.size
      }, null, 8, ["color", "size"]),
      e.$slots.default ? (o(), n("span", z, [
        i(e.$slots, "default")
      ])) : f("v-if", !0)
    ]));
  }
});
export {
  y as default
};
