// maz-ui/components/MazInputCode.mjs
import { defineComponent as N, useCssVars as S, ref as V, watch as M, computed as m, openBlock as v, createElementBlock as g, normalizeClass as w, normalizeStyle as P, Fragment as $, renderList as q, createCommentVNode as x, createElementVNode as D, mergeProps as E } from "vue";
import './assets/MazInputCode.css';const F = ["disabled"], K = ["inputmode", "pattern", "required", "value", "onInput", "onKeydown", "onClick"], T = /* @__PURE__ */ N({
  inheritAttrs: !1,
  __name: "MazInputCode",
  props: {
    style: { default: void 0 },
    class: { default: void 0 },
    modelValue: { default: void 0 },
    codeLength: { default: 4 },
    type: { default: "text" },
    acceptAlpha: { type: Boolean, default: !1 },
    required: { type: Boolean, default: !1 },
    disabled: { type: Boolean, default: !1 },
    error: { type: Boolean, default: !1 },
    success: { type: Boolean, default: !1 },
    warning: { type: Boolean, default: !1 },
    size: { default: "md" },
    color: { default: "primary" }
  },
  emits: ["update:model-value", "completed"],
  setup(i, { emit: f }) {
    S((e) => ({
      "126eacc0": C.value
    }));
    const a = i, d = f, u = V([]), l = V(/* @__PURE__ */ new Map());
    M(
      () => a.modelValue,
      (e, t) => {
        e !== t && (l.value = z());
      },
      { immediate: !0 }
    );
    const p = m({
      get: () => l.value,
      set: (e) => {
        const t = L(e);
        d("update:model-value", t), (t == null ? void 0 : t.toString().length) === a.codeLength && d("completed");
      }
    });
    function z(e = a.modelValue) {
      const t = /* @__PURE__ */ new Map();
      for (const o of Array.from({ length: a.codeLength }, (n, r) => r))
        if (e === void 0)
          t.set(o + 1, void 0);
        else {
          const n = [...e.toString()];
          t.set(o + 1, n[o]);
        }
      return t;
    }
    function I(e, t) {
      var n;
      const o = h((n = e.target) == null ? void 0 : n.value);
      p.value = l.value.set(t, o), o && c(t);
    }
    function L(e) {
      const t = [...e.values()].join("");
      return a.type === "text" ? t : t && !Number.isNaN(Number(t)) ? Number(t) : void 0;
    }
    function h(e) {
      var t;
      return (t = a.acceptAlpha ? e.match(/^[\w.]$/) : e.match(/\d+/g)) == null ? void 0 : t.toString();
    }
    function B(e, t) {
      const o = l.value.get(t);
      if (e.key === "ArrowRight" && c(t), e.key === "ArrowLeft" && c(t - 2), e.key === "Backspace" && !o) {
        const n = t - 1 < 0 ? 0 : t - 1;
        p.value = l.value.set(n, void 0), c(n - 1);
      }
    }
    function A(e) {
      var n;
      const t = (n = e.clipboardData) == null ? void 0 : n.getData("text");
      if (!t)
        return;
      const o = Array.from({ length: a.codeLength }, (r, s) => ({
        index: s + 1,
        value: [...t.toString()][s] ?? void 0
      }));
      for (const { index: r, value: s } of o) {
        const b = s ? h(s) : void 0;
        b && l.value.set(r, b);
      }
      setTimeout(() => {
        const r = _(), s = r >= a.codeLength ? a.codeLength - 1 : r;
        c(s, !1);
      }, 0);
    }
    function _() {
      return [...l.value.values()].filter(Boolean).length;
    }
    function c(e, t = !0) {
      setTimeout(() => {
        const o = u.value[e];
        e + 1 > a.codeLength || !o || (o.focus(), t && y(e));
      }, 0);
    }
    function y(e) {
      const t = u.value[e];
      e + 1 > a.codeLength || !t || t.select();
    }
    const k = m(() => a.error ? "!maz-border-danger" : a.success ? "!maz-border-success" : a.warning ? "!maz-border-warning" : ""), C = m(() => `var(--maz-color-${a.color})`);
    return (e, t) => (v(), g("fieldset", {
      class: w(["m-input-code", [e.size ? `--${e.size}` : void 0, a.class]]),
      disabled: e.disabled,
      style: P(e.style)
    }, [
      (v(!0), g(
        $,
        null,
        q(e.codeLength, (o) => (v(), g(
          "div",
          {
            key: o,
            class: w(["input-wrapper", k.value])
          },
          [
            x(" eslint-disable-next-line vuejs-accessibility/form-control-has-label "),
            D("input", E({
              ref_for: !0,
              ref: (n) => {
                u.value[o - 1] = n;
              },
              type: "text",
              minlength: "1",
              maxlength: "1",
              inputmode: e.acceptAlpha ? "text" : "numeric",
              pattern: e.acceptAlpha ? "[a-zA-Z0-9]{1}" : "[0-9]{1}",
              autocomplete: "do-not-autofill",
              required: e.required
            }, e.$attrs, {
              value: p.value.get(o),
              onInput: (n) => I(n, o),
              onKeydown: (n) => B(n, o),
              onClick: (n) => y(o - 1),
              onPaste: A
            }), null, 16, K)
          ],
          2
          /* CLASS */
        ))),
        128
        /* KEYED_FRAGMENT */
      ))
    ], 14, F));
  }
}), j = (i, f) => {
  const a = i.__vccOpts || i;
  for (const [d, u] of f)
    a[d] = u;
  return a;
}, W = /* @__PURE__ */ j(T, [["__scopeId", "data-v-afab8168"]]);
export {
  W as default
};
