// maz-ui/modules/chunks/MazBtn-1g7UvzQ_.mjs
import { defineComponent as F, defineAsyncComponent as h, useAttrs as j, useSlots as E, onBeforeMount as O, computed as l, openBlock as n, createBlock as a, resolveDynamicComponent as r, normalizeClass as s, withCtx as R, createElementBlock as u, createCommentVNode as t, renderSlot as d, unref as c } from "vue";
import { _ as w } from "./index-D7R1dvsf.mjs";
import '../assets/MazBtn.css';const A = /* @__PURE__ */ F({
  __name: "MazBtn",
  props: {
    variant: { default: "button" },
    size: { default: "md" },
    color: { default: "primary" },
    type: { default: "button" },
    rounded: { type: Boolean },
    roundedSize: { default: "lg" },
    noRounded: { type: Boolean },
    outline: { type: Boolean },
    pastel: { type: Boolean },
    block: { type: Boolean },
    noUnderline: { type: Boolean },
    loading: { type: Boolean },
    disabled: { type: Boolean },
    fab: { type: Boolean },
    icon: { type: [String, Function, Object], default: void 0 },
    leftIcon: { type: [String, Function, Object], default: void 0 },
    rightIcon: { type: [String, Function, Object], default: void 0 },
    noPadding: { type: Boolean },
    noElevation: { type: Boolean },
    contentClass: { default: void 0 }
  },
  setup(g) {
    const o = g, z = h(() => import("./MazSpinner--BVvGFIO.mjs")), f = h(() => import("./MazIcon-BdDO3aVl.mjs")), { href: k, to: B } = j(), p = E();
    O(() => {
      o.icon && !o.fab && console.error('[maz-ui](MazBtn) the prop "icon" must be used only with "fab" props');
    });
    const v = l(() => k ? "a" : B ? "router-link" : "button"), I = l(
      () => o.pastel ? `--${o.color}-pastel` : o.outline ? `--${o.color}-outline` : `--${o.color}`
    ), b = l(
      () => (o.loading || o.disabled) && v.value === "button"
    ), C = l(() => b.value ? "--cursor-default" : "--cursor-pointer"), $ = l(() => `--is-${o.variant}`), i = l(() => o.loading && o.variant === "button"), m = l(() => !!p["left-icon"] || o.leftIcon), y = l(() => !!p["right-icon"] || o.rightIcon), S = l(() => m.value || y.value), M = l(() => o.fab && (o.icon || !!p.icon)), T = l(() => v.value === "button" ? o.type : void 0);
    return (e, D) => (n(), a(r(v.value), {
      disabled: b.value || void 0,
      class: s(["m-btn", [
        `--${e.size}`,
        !e.fab && e.roundedSize && `--rounded-${e.roundedSize}`,
        I.value,
        C.value,
        $.value,
        {
          "--block": e.block,
          "--no-underline": e.noUnderline,
          "--fab": e.fab,
          "--loading": e.loading,
          "--disabled": b.value,
          "--icon": S.value,
          "--rounded": e.rounded,
          "--no-rounded": e.noRounded,
          "--no-padding": e.noPadding,
          "--no-elevation": e.noElevation
        }
      ]]),
      type: T.value
    }, {
      default: R(() => [
        m.value ? (n(), u(
          "div",
          {
            key: 0,
            class: s(["m-btn__icon-left maz-flex maz-flex-center", { "maz-invisible": i.value }])
          },
          [
            t(`
        @slot left-icon - The icon to display on the left of the button
      `),
            d(e.$slots, "left-icon", {}, () => [
              typeof e.leftIcon == "string" ? (n(), a(c(f), {
                key: 0,
                name: e.leftIcon
              }, null, 8, ["name"])) : e.leftIcon ? (n(), a(r(e.leftIcon), { key: 1 })) : t("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : t("v-if", !0),
        M.value ? (n(), u(
          "div",
          {
            key: 1,
            class: s(["m-btn__icon", { "maz-invisible": i.value }])
          },
          [
            t(`
        @slot icon - The icon to display on the fab button
      `),
            d(e.$slots, "icon", {}, () => [
              typeof e.icon == "string" ? (n(), a(c(f), {
                key: 0,
                name: e.icon
              }, null, 8, ["name"])) : e.icon ? (n(), a(r(e.icon), { key: 1 })) : t("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : t("v-if", !0),
        e.$slots.default ? (n(), u(
          "span",
          {
            key: 2,
            class: s([{ "maz-invisible": i.value }, e.contentClass])
          },
          [
            t(`
        @slot default - The content of the button
      `),
            d(e.$slots, "default", {}, void 0, !0)
          ],
          2
          /* CLASS */
        )) : t("v-if", !0),
        y.value ? (n(), u(
          "div",
          {
            key: 3,
            class: s(["m-btn__icon-right", { "maz-invisible": i.value }])
          },
          [
            t(`
        @slot right-icon - The icon to display on the right of the button
      `),
            d(e.$slots, "right-icon", {}, () => [
              typeof e.rightIcon == "string" ? (n(), a(c(f), {
                key: 0,
                name: e.rightIcon
              }, null, 8, ["name"])) : e.rightIcon ? (n(), a(r(e.rightIcon), { key: 1 })) : t("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : t("v-if", !0),
        i.value ? (n(), a(c(z), {
          key: 4,
          class: "m-btn-loader",
          size: "2em",
          color: e.color
        }, null, 8, ["color"])) : t("v-if", !0)
      ]),
      _: 3
      /* FORWARDED */
    }, 8, ["disabled", "class", "type"]));
  }
}), U = /* @__PURE__ */ w(A, [["__scopeId", "data-v-48f14785"]]);
export {
  U as default
};
