// maz-ui/components/chunks/MazBottomSheet-DpI8wFgc.mjs
import { defineComponent as S, ref as L, onMounted as V, watch as D, openBlock as c, createBlock as k, Teleport as P, createVNode as M, Transition as $, withCtx as v, createElementBlock as A, normalizeClass as b, createElementVNode as d, withModifiers as N, mergeProps as O, renderSlot as z, createCommentVNode as C, defineAsyncComponent as I, unref as w, pushScopeId as T, popScopeId as x } from "vue";
import '../assets/MazBottomSheet.css';const g = "--backdrop-present", K = /* @__PURE__ */ S({
  inheritAttrs: !1,
  __name: "MazBackdrop",
  props: {
    modelValue: { type: Boolean, default: !1 },
    teleportSelector: { default: "body" },
    beforeClose: { type: Function, default: void 0 },
    persistent: { type: Boolean, default: !1 },
    noCloseOnEscKey: { type: Boolean, default: !1 },
    transitionName: { default: "backdrop-anim" },
    backdropClass: { default: void 0 },
    backdropContentClass: { default: void 0 }
  },
  emits: ["open", "close", "update:model-value", "before-close"],
  setup(t, { expose: l, emit: r }) {
    const o = t, n = r;
    function a() {
      document.documentElement.classList.add(g);
    }
    async function s() {
      document.querySelector(".m-backdrop.--present") || document.documentElement.classList.remove(g);
    }
    const i = L(o.modelValue);
    function m() {
      _(!1);
    }
    async function _(e) {
      var f;
      e || (n("before-close"), await ((f = o.beforeClose) == null ? void 0 : f.call(o))), i.value = e;
    }
    function y() {
      n("open");
    }
    function h() {
      n("update:model-value", !1), n("close"), p();
    }
    function B() {
      o.persistent || m();
    }
    function u(e) {
      !o.noCloseOnEscKey && e.key === "Escape" && !o.persistent && m();
    }
    function E() {
      a(), document.addEventListener("keyup", u, !1);
    }
    function p() {
      document.removeEventListener("keyup", u), s();
    }
    return V(() => {
      o.modelValue ? E() : p();
    }), D(
      () => o.modelValue,
      (e) => {
        i.value = e, e ? E() : p();
      }
    ), l({
      onBackdropAnimationEnter: y,
      onBackdropAnimationLeave: h,
      onBackdropClicked: B,
      close: m,
      present: i,
      toggleModal: _,
      onKeyPress: u
    }), (e, f) => (c(), k(P, { to: e.teleportSelector }, [
      M($, {
        appear: "",
        name: e.transitionName,
        onAfterEnter: y,
        onAfterLeave: h
      }, {
        default: v(() => [
          i.value ? (c(), A(
            "div",
            {
              key: 0,
              class: b(["m-backdrop --present", [e.backdropClass]]),
              tabindex: "-1",
              role: "dialog"
            },
            [
              d(
                "button",
                {
                  class: b(["m-backdrop-overlay", { "--disabled": e.persistent }]),
                  tabindex: "-1",
                  onClick: N(B, ["self"])
                },
                null,
                2
                /* CLASS */
              ),
              d(
                "div",
                O({
                  class: ["m-backdrop-content", e.backdropContentClass]
                }, e.$attrs, {
                  role: "document",
                  tabindex: "0"
                }),
                [
                  z(e.$slots, "default", { close: m })
                ],
                16
                /* FULL_PROPS */
              )
            ],
            2
            /* CLASS */
          )) : C("v-if", !0)
        ]),
        _: 3
        /* FORWARDED */
      }, 8, ["name"])
    ], 8, ["to"]));
  }
}), F = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "1em",
  height: "1em",
  fill: "none",
  viewBox: "0 0 24 24"
}, j = /* @__PURE__ */ d(
  "path",
  {
    stroke: "currentColor",
    "stroke-linecap": "round",
    "stroke-linejoin": "round",
    "stroke-width": "1.5",
    d: "M6 18 18 6M6 6l12 12"
  },
  null,
  -1
  /* HOISTED */
), q = [
  j
];
function U(t, l) {
  return c(), A("svg", F, [...q]);
}
const X = { render: U }, G = (t) => (T("data-v-8350510b"), t = t(), x(), t), H = /* @__PURE__ */ G(() => /* @__PURE__ */ d(
  "div",
  { class: "m-bottom-sheet__content-wrapper" },
  [
    /* @__PURE__ */ d("p", null, "Default content")
  ],
  -1
  /* HOISTED */
)), J = /* @__PURE__ */ S({
  __name: "MazBottomSheet",
  props: {
    noClose: { type: Boolean, default: !1 },
    noPadding: { type: Boolean, default: !1 }
  },
  emits: ["update:model-value", "open", "close"],
  setup(t, { emit: l }) {
    const r = l, o = I(() => import("./MazBtn-D2tIsob2.mjs"));
    return (n, a) => (c(), k(K, {
      "transition-name": "bottom-sheet-anim",
      "backdrop-class": "--bottom-sheet",
      onClose: a[0] || (a[0] = (s) => n.$emit("close", s)),
      onOpen: a[1] || (a[1] = (s) => n.$emit("open", s)),
      "onUpdate:modelValue": a[2] || (a[2] = (s) => r("update:model-value", s))
    }, {
      default: v(({ close: s }) => [
        d(
          "div",
          {
            class: b(["m-bottom-sheet__container", {
              "maz-py-6": !t.noPadding
            }])
          },
          [
            C(" Slot content "),
            z(n.$slots, "default", { close: s }, () => [
              H
            ], !0),
            t.noClose ? C("v-if", !0) : (c(), k(w(o), {
              key: 0,
              size: "sm",
              class: "m-bottom-sheet__close",
              color: "transparent",
              onClick: s
            }, {
              default: v(() => [
                M(w(X), { class: "maz-text-lg" })
              ]),
              _: 2
              /* DYNAMIC */
            }, 1032, ["onClick"]))
          ],
          2
          /* CLASS */
        )
      ]),
      _: 3
      /* FORWARDED */
    }));
  }
}), Q = (t, l) => {
  const r = t.__vccOpts || t;
  for (const [o, n] of l)
    r[o] = n;
  return r;
}, W = /* @__PURE__ */ Q(J, [["__scopeId", "data-v-8350510b"]]);
export {
  W as M,
  Q as _
};
