// maz-ui/components/chunks/MazDialog-DXaeCp-b.mjs
import { defineComponent as T, ref as N, onMounted as S, watch as O, openBlock as d, createBlock as y, Teleport as F, createVNode as L, Transition as P, withCtx as k, createElementBlock as C, normalizeClass as g, createElementVNode as m, withModifiers as H, mergeProps as h, renderSlot as u, createCommentVNode as a, defineAsyncComponent as V, useAttrs as K, computed as w, createTextVNode as W, toDisplayString as I, unref as z } from "vue";
import '../assets/MazDialog.css';const M = "--backdrop-present", q = /* @__PURE__ */ T({
  inheritAttrs: !1,
  __name: "MazBackdrop",
  props: {
    modelValue: { type: Boolean, default: !1 },
    teleportSelector: { default: "body" },
    beforeClose: { type: Function, default: void 0 },
    persistent: { type: Boolean, default: !1 },
    noCloseOnEscKey: { type: Boolean, default: !1 },
    transitionName: { default: "backdrop-anim" },
    backdropClass: { default: void 0 },
    backdropContentClass: { default: void 0 }
  },
  emits: ["open", "close", "update:model-value", "before-close"],
  setup(r, { expose: c, emit: i }) {
    const t = r, l = i;
    function p() {
      document.documentElement.classList.add(M);
    }
    async function e() {
      document.querySelector(".m-backdrop.--present") || document.documentElement.classList.remove(M);
    }
    const s = N(t.modelValue);
    function n() {
      B(!1);
    }
    async function B(o) {
      var b;
      o || (l("before-close"), await ((b = t.beforeClose) == null ? void 0 : b.call(t))), s.value = o;
    }
    function E() {
      l("open");
    }
    function A() {
      l("update:model-value", !1), l("close"), v();
    }
    function $() {
      t.persistent || n();
    }
    function f(o) {
      !t.noCloseOnEscKey && o.key === "Escape" && !t.persistent && n();
    }
    function D() {
      p(), document.addEventListener("keyup", f, !1);
    }
    function v() {
      document.removeEventListener("keyup", f), e();
    }
    return S(() => {
      t.modelValue ? D() : v();
    }), O(
      () => t.modelValue,
      (o) => {
        s.value = o, o ? D() : v();
      }
    ), c({
      onBackdropAnimationEnter: E,
      onBackdropAnimationLeave: A,
      onBackdropClicked: $,
      close: n,
      present: s,
      toggleModal: B,
      onKeyPress: f
    }), (o, b) => (d(), y(F, { to: o.teleportSelector }, [
      L(P, {
        appear: "",
        name: o.transitionName,
        onAfterEnter: E,
        onAfterLeave: A
      }, {
        default: k(() => [
          s.value ? (d(), C(
            "div",
            {
              key: 0,
              class: g(["m-backdrop --present", [o.backdropClass]]),
              tabindex: "-1",
              role: "dialog"
            },
            [
              m(
                "button",
                {
                  class: g(["m-backdrop-overlay", { "--disabled": o.persistent }]),
                  tabindex: "-1",
                  onClick: H($, ["self"])
                },
                null,
                2
                /* CLASS */
              ),
              m(
                "div",
                h({
                  class: ["m-backdrop-content", o.backdropContentClass]
                }, o.$attrs, {
                  role: "document",
                  tabindex: "0"
                }),
                [
                  u(o.$slots, "default", { close: n })
                ],
                16
                /* FULL_PROPS */
              )
            ],
            2
            /* CLASS */
          )) : a("v-if", !0)
        ]),
        _: 3
        /* FORWARDED */
      }, 8, ["name"])
    ], 8, ["to"]));
  }
}), U = {
  key: 0,
  id: "dialogTitle",
  class: "maz-my-0 maz-text-xl maz-font-semibold"
}, X = {
  id: "dialogDesc",
  class: "m-dialog-content"
}, j = {
  key: 0,
  class: "m-dialog-footer"
}, G = /* @__PURE__ */ T({
  __name: "MazDialog",
  props: {
    modelValue: { type: Boolean },
    title: { default: void 0 },
    noClose: { type: Boolean, default: !1 },
    width: { default: "500px" },
    maxWidth: { default: "95vw" },
    maxHeight: { default: "95vh" },
    scrollable: { type: Boolean, default: !1 },
    persistent: { type: Boolean, default: !1 },
    teleportSelector: {},
    beforeClose: {},
    noCloseOnEscKey: { type: Boolean },
    transitionName: {},
    backdropClass: {},
    backdropContentClass: {}
  },
  emits: ["open", "close", "update:model-value"],
  setup(r) {
    const c = V(() => import("./MazBtn-Vas62hFD.mjs")), i = V(() => import("./x-mark-BP8zWe_C.mjs")), t = K(), l = w(() => ({
      ...t,
      class: void 0,
      style: void 0
    })), p = w(() => ({
      class: t.class,
      style: t.style
    }));
    return (e, s) => (d(), y(q, h(l.value, {
      persistent: e.persistent,
      "model-value": e.modelValue,
      "transition-name": "modal-anim",
      "aria-labelledby": "dialogTitle",
      "aria-describedby": "dialogDesc",
      onClose: s[0] || (s[0] = (n) => e.$emit("close", n)),
      onOpen: s[1] || (s[1] = (n) => e.$emit("open", n)),
      "onUpdate:modelValue": s[2] || (s[2] = (n) => e.$emit("update:model-value", n))
    }), {
      default: k(({ close: n }) => [
        m(
          "div",
          h({
            class: ["m-dialog", { "--scrollable": e.scrollable }],
            role: "dialog",
            "aria-modal": "true",
            style: [{ width: e.width, maxWidth: e.maxWidth, maxHeight: e.maxHeight }]
          }, p.value),
          [
            a(`
          @slot Header slot
            @binding {Function} close close function
        `),
            u(e.$slots, "header", { close: n }, () => [
              m(
                "div",
                {
                  class: g(["m-dialog-header", { "--has-title": e.$slots.title || e.title }])
                },
                [
                  e.$slots.title || e.title ? (d(), C("h2", U, [
                    a(`
                @slot Title slot in the header
              `),
                    u(e.$slots, "title", {}, () => [
                      W(
                        I(e.title),
                        1
                        /* TEXT */
                      )
                    ], !0)
                  ])) : a("v-if", !0),
                  !e.noClose && !e.persistent ? (d(), y(z(c), {
                    key: 1,
                    class: "m-dialog-closebtn",
                    color: "transparent",
                    onClick: n
                  }, {
                    default: k(() => [
                      L(z(i), { class: "maz-text-lg" })
                    ]),
                    _: 2
                    /* DYNAMIC */
                  }, 1032, ["onClick"])) : a("v-if", !0)
                ],
                2
                /* CLASS */
              )
            ], !0),
            m("div", X, [
              a(`
            @slot Default content
              @binding {Function} close close function
          `),
              u(e.$slots, "default", { close: n }, void 0, !0)
            ]),
            e.$slots.footer ? (d(), C("div", j, [
              a(`
            @slot Footer slot
              @binding {Function} close close function
          `),
              u(e.$slots, "footer", { close: n }, void 0, !0)
            ])) : a("v-if", !0)
          ],
          16
          /* FULL_PROPS */
        )
      ]),
      _: 3
      /* FORWARDED */
    }, 16, ["persistent", "model-value"]));
  }
}), J = (r, c) => {
  const i = r.__vccOpts || r;
  for (const [t, l] of c)
    i[t] = l;
  return i;
}, R = /* @__PURE__ */ J(G, [["__scopeId", "data-v-20ca3bcd"]]);
export {
  R as M,
  J as _
};
