// maz-ui/components/chunks/MazInput-D177bUMh.mjs
import { getCurrentInstance as K, computed as r, defineComponent as Q, defineAsyncComponent as f, ref as z, onMounted as W, useSlots as X, openBlock as n, createElementBlock as p, normalizeClass as g, normalizeStyle as Y, createElementVNode as k, createCommentVNode as o, renderSlot as I, createBlock as d, unref as s, resolveDynamicComponent as C, withDirectives as Z, mergeProps as x, toHandlers as _, vModelDynamic as ee, createTextVNode as te, toDisplayString as oe, withModifiers as ne, withCtx as V, createVNode as S } from "vue";
import '../assets/MazInput.css';function ae({
  componentName: i,
  providedId: u
}) {
  const t = K();
  return r(() => u ?? `${i}-${t == null ? void 0 : t.uid}`);
}
function le(i, u) {
  let t;
  return function(...l) {
    clearTimeout(t), t = setTimeout(() => {
      i.apply(this, l);
    }, u);
  };
}
const re = {
  key: 0,
  class: "m-input-wrapper-left"
}, ie = { class: "m-input-wrapper-input" }, se = ["id", "type", "name", "inputmode", "placeholder", "aria-label", "disabled", "readonly", "required"], ue = { key: 0 }, de = {
  key: 1,
  class: "m-input-wrapper-right"
}, ce = /* @__PURE__ */ Q({
  inheritAttrs: !1,
  __name: "MazInput",
  props: {
    style: { default: void 0 },
    class: { default: void 0 },
    modelValue: { default: void 0 },
    placeholder: { default: void 0 },
    label: { default: void 0 },
    name: { default: void 0 },
    color: { default: "primary" },
    type: { default: "text" },
    required: { type: Boolean, default: !1 },
    disabled: { type: Boolean, default: !1 },
    readonly: { type: Boolean, default: !1 },
    id: { default: void 0 },
    error: { type: Boolean, default: !1 },
    success: { type: Boolean, default: !1 },
    warning: { type: Boolean, default: !1 },
    hint: { default: void 0 },
    inputClasses: { default: void 0 },
    noBorder: { type: Boolean, default: !1 },
    noRadius: { type: Boolean, default: !1 },
    inputmode: { default: "text" },
    size: { default: "md" },
    debounce: { type: [Boolean, Number], default: !1 },
    debounceDelay: { default: 500 },
    validButton: { type: Boolean, default: !1 },
    validButtonLoading: { type: Boolean, default: !1 },
    autoFocus: { type: Boolean, default: !1 },
    borderActive: { type: Boolean, default: !1 },
    leftIcon: { type: [String, Function, Object], default: void 0 },
    rightIcon: { type: [String, Function, Object], default: void 0 },
    roundedSize: { default: void 0 },
    block: { type: Boolean }
  },
  emits: ["update:model-value", "focus", "blur", "click", "change"],
  setup(i, { emit: u }) {
    const t = i, l = u, m = f(() => import("./MazBtn-CPkNymJj.mjs")), B = f(() => import("./MazIcon-BDGxCy9M.mjs")), M = f(() => import("./eye-slash-tjfxO0LK.mjs")), q = f(() => import("./eye-DqO_Jx56.mjs")), T = f(() => import("./check-CL-vqe79.mjs")), v = z(!1), c = z(!1), w = z(), D = ae({
      componentName: "MazInput",
      providedId: t.id
    });
    W(() => {
      var e;
      t.autoFocus && ((e = w.value) == null || e.focus());
    });
    const $ = r(() => t.type === "password"), N = r(() => v.value ? "text" : t.type), P = r(() => {
      if (!t.noBorder) {
        if (t.error)
          return "maz-border-danger";
        if (t.success)
          return "maz-border-success";
        if (t.warning)
          return "maz-border-warning";
        if (c.value || t.borderActive) {
          if (t.color === "black")
            return "maz-border-black";
          if (t.color === "danger")
            return "maz-border-danger";
          if (t.color === "info")
            return "maz-border-info";
          if (t.color === "primary")
            return "maz-border-primary";
          if (t.color === "secondary")
            return "maz-border-secondary";
          if (t.color === "success")
            return "maz-border-success";
          if (t.color === "warning")
            return "maz-border-warning";
          if (t.color === "white")
            return "maz-border-white";
        }
        return "--default-border";
      }
    }), h = X(), A = r(() => {
      const { required: e, placeholder: a } = t;
      if (a)
        return e ? `${a} *` : a;
    }), E = r(() => y.value !== void 0 && y.value !== ""), F = le(
      (e) => {
        l("update:model-value", e);
      },
      typeof t.debounce == "number" ? t.debounce : t.debounceDelay ?? 500
    );
    function L(e) {
      if (t.debounce)
        return F(e);
      l("update:model-value", e);
    }
    const y = r({
      get: () => t.modelValue,
      set: (e) => L(e)
    }), O = r(() => (!!t.label || !!t.hint) && (c.value || !!E.value || !!t.placeholder || ["date", "month", "week"].includes(t.type))), R = r(() => !!t.label || !!t.hint);
    function U() {
      return !!h["right-icon"] || $.value || !!h["valid-button"] || t.validButton || !!t.rightIcon;
    }
    function j() {
      return !!h["left-icon"] || !!t.leftIcon;
    }
    function H(e) {
      l("focus", e), c.value = !0;
    }
    function G(e) {
      l("blur", e), c.value = !1;
    }
    function J(e) {
      return l("change", e);
    }
    return (e, a) => (n(), p(
      "div",
      {
        class: g(["m-input", [
          {
            "--is-focused": c.value || e.borderActive,
            "--should-up": O.value,
            "--has-label": R.value,
            "--is-disabled": e.disabled,
            "--is-readonly": e.readonly,
            "--has-z-2": e.error || e.warning || e.success,
            "--has-state": e.error || e.warning || e.success,
            "--block": e.block
          },
          t.class,
          `--${e.color}`,
          `--${e.size}`
        ]]),
        style: Y(e.style)
      },
      [
        k(
          "div",
          {
            class: g(["m-input-wrapper", [
              e.inputClasses,
              P.value,
              e.roundedSize ? `--rounded-${e.roundedSize}` : { "maz-rounded": !e.noRadius }
            ]])
          },
          [
            j() ? (n(), p("div", re, [
              o(`
          @slot left-icon - The icon to display on the left of the input
        `),
              e.$slots["left-icon"] || e.leftIcon ? I(e.$slots, "left-icon", { key: 0 }, () => [
                typeof e.leftIcon == "string" ? (n(), d(s(B), {
                  key: 0,
                  name: e.leftIcon,
                  class: "maz-text-xl maz-text-muted"
                }, null, 8, ["name"])) : e.leftIcon ? (n(), d(C(e.leftIcon), {
                  key: 1,
                  class: "maz-text-xl maz-text-muted"
                })) : o("v-if", !0)
              ], !0) : o("v-if", !0)
            ])) : o("v-if", !0),
            k("div", ie, [
              Z(k("input", x({
                id: s(D),
                ref_key: "input",
                ref: w,
                "onUpdate:modelValue": a[0] || (a[0] = (b) => y.value = b),
                type: N.value,
                name: e.name
              }, e.$attrs, {
                inputmode: e.inputmode,
                placeholder: A.value,
                "aria-label": e.label || e.placeholder,
                disabled: e.disabled,
                readonly: e.readonly,
                required: e.required,
                class: "m-input-input"
              }, _({
                blur: G,
                focus: H,
                change: J
              }, !0), {
                onClick: a[1] || (a[1] = (b) => e.$emit("click", b))
              }), null, 16, se), [
                [ee, y.value]
              ]),
              e.label || e.hint ? (n(), p(
                "span",
                {
                  key: 0,
                  class: g(["m-input-label", [
                    {
                      "maz-text-danger-600": e.error,
                      "maz-text-success-600": e.success,
                      "maz-text-warning-600": e.warning
                    }
                  ]])
                },
                [
                  te(
                    oe(e.hint || e.label) + " ",
                    1
                    /* TEXT */
                  ),
                  e.required ? (n(), p("sup", ue, "*")) : o("v-if", !0)
                ],
                2
                /* CLASS */
              )) : o("v-if", !0)
            ]),
            U() ? (n(), p("div", de, [
              o(`
          @slot right-icon - The icon to display on the right of the input
        `),
              e.$slots["right-icon"] || e.rightIcon ? I(e.$slots, "right-icon", { key: 0 }, () => [
                typeof e.rightIcon == "string" ? (n(), d(s(B), {
                  key: 0,
                  name: e.rightIcon,
                  class: "maz-text-xl maz-text-muted"
                }, null, 8, ["name"])) : e.rightIcon ? (n(), d(C(e.rightIcon), {
                  key: 1,
                  class: "maz-text-xl maz-text-muted"
                })) : o("v-if", !0)
              ], !0) : o("v-if", !0),
              $.value ? (n(), d(s(m), {
                key: 1,
                color: "transparent",
                tabindex: "-1",
                size: "mini",
                onClick: a[2] || (a[2] = ne((b) => v.value = !v.value, ["stop"]))
              }, {
                default: V(() => [
                  v.value ? (n(), d(s(M), {
                    key: 0,
                    class: "maz-text-xl maz-text-muted"
                  })) : (n(), d(s(q), {
                    key: 1,
                    class: "maz-text-xl maz-text-muted"
                  }))
                ]),
                _: 1
                /* STABLE */
              })) : o("v-if", !0),
              o(`
          @slot valid-button - Replace the valid button by your own
        `),
              e.$slots["valid-button"] || e.validButton ? I(e.$slots, "valid-button", { key: 2 }, () => [
                S(s(m), {
                  color: "transparent",
                  disabled: e.disabled,
                  tabindex: "-1",
                  loading: e.validButtonLoading,
                  class: "m-input-valid-button",
                  size: "mini",
                  type: "submit"
                }, {
                  default: V(() => [
                    S(s(T), { class: "maz-text-2xl maz-text-normal" })
                  ]),
                  _: 1
                  /* STABLE */
                }, 8, ["disabled", "loading"])
              ], !0) : o("v-if", !0)
            ])) : o("v-if", !0)
          ],
          2
          /* CLASS */
        )
      ],
      6
      /* CLASS, STYLE */
    ));
  }
}), fe = (i, u) => {
  const t = i.__vccOpts || i;
  for (const [l, m] of u)
    t[l] = m;
  return t;
}, me = /* @__PURE__ */ fe(ce, [["__scopeId", "data-v-e9affc48"]]);
export {
  me as M,
  fe as _
};
