// maz-ui/components/chunks/MazPagination-NbHR41CQ.mjs
import { defineComponent as q, defineAsyncComponent as N, useAttrs as H, useSlots as J, onBeforeMount as K, computed as t, openBlock as i, createBlock as c, resolveDynamicComponent as D, normalizeClass as S, withCtx as V, createElementBlock as b, createCommentVNode as s, renderSlot as p, unref as m, createElementVNode as d, createVNode as v, mergeProps as $, createTextVNode as k, toDisplayString as A, Fragment as Q, renderList as W } from "vue";
import '../assets/MazPagination.css';const X = /* @__PURE__ */ q({
  __name: "MazBtn",
  props: {
    variant: { default: "button" },
    size: { default: "md" },
    color: { default: "primary" },
    type: { default: "button" },
    rounded: { type: Boolean },
    roundedSize: { default: "lg" },
    noRounded: { type: Boolean },
    outline: { type: Boolean },
    pastel: { type: Boolean },
    block: { type: Boolean },
    noUnderline: { type: Boolean },
    loading: { type: Boolean },
    disabled: { type: Boolean },
    fab: { type: Boolean },
    icon: { type: [String, Function, Object], default: void 0 },
    leftIcon: { type: [String, Function, Object], default: void 0 },
    rightIcon: { type: [String, Function, Object], default: void 0 },
    noPadding: { type: Boolean },
    noElevation: { type: Boolean },
    contentClass: { default: void 0 }
  },
  setup(P) {
    const l = P, e = N(() => import("./MazSpinner-C6T_enKo.mjs")), f = N(() => import("./MazIcon-BDGxCy9M.mjs")), { href: h, to: T } = H(), C = J();
    K(() => {
      l.icon && !l.fab && console.error('[maz-ui](MazBtn) the prop "icon" must be used only with "fab" props');
    });
    const B = t(() => h ? "a" : T ? "router-link" : "button"), r = t(
      () => l.pastel ? `--${l.color}-pastel` : l.outline ? `--${l.color}-outline` : `--${l.color}`
    ), g = t(
      () => (l.loading || l.disabled) && B.value === "button"
    ), z = t(() => g.value ? "--cursor-default" : "--cursor-pointer"), I = t(() => `--is-${l.variant}`), y = t(() => l.loading && l.variant === "button"), E = t(() => !!C["left-icon"] || l.leftIcon), F = t(() => !!C["right-icon"] || l.rightIcon), L = t(() => E.value || F.value), U = t(() => l.fab && (l.icon || !!C.icon)), j = t(() => B.value === "button" ? l.type : void 0);
    return (a, w) => (i(), c(D(B.value), {
      disabled: g.value || void 0,
      class: S(["m-btn", [
        `--${a.size}`,
        !a.fab && a.roundedSize && `--rounded-${a.roundedSize}`,
        r.value,
        z.value,
        I.value,
        {
          "--block": a.block,
          "--no-underline": a.noUnderline,
          "--fab": a.fab,
          "--loading": a.loading,
          "--disabled": g.value,
          "--icon": L.value,
          "--rounded": a.rounded,
          "--no-rounded": a.noRounded,
          "--no-padding": a.noPadding,
          "--no-elevation": a.noElevation
        }
      ]]),
      type: j.value
    }, {
      default: V(() => [
        E.value ? (i(), b(
          "div",
          {
            key: 0,
            class: S(["m-btn__icon-left maz-flex maz-flex-center", { "maz-invisible": y.value }])
          },
          [
            s(`
        @slot left-icon - The icon to display on the left of the button
      `),
            p(a.$slots, "left-icon", {}, () => [
              typeof a.leftIcon == "string" ? (i(), c(m(f), {
                key: 0,
                name: a.leftIcon
              }, null, 8, ["name"])) : a.leftIcon ? (i(), c(D(a.leftIcon), { key: 1 })) : s("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : s("v-if", !0),
        U.value ? (i(), b(
          "div",
          {
            key: 1,
            class: S(["m-btn__icon", { "maz-invisible": y.value }])
          },
          [
            s(`
        @slot icon - The icon to display on the fab button
      `),
            p(a.$slots, "icon", {}, () => [
              typeof a.icon == "string" ? (i(), c(m(f), {
                key: 0,
                name: a.icon
              }, null, 8, ["name"])) : a.icon ? (i(), c(D(a.icon), { key: 1 })) : s("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : s("v-if", !0),
        a.$slots.default ? (i(), b(
          "span",
          {
            key: 2,
            class: S([{ "maz-invisible": y.value }, a.contentClass])
          },
          [
            s(`
        @slot default - The content of the button
      `),
            p(a.$slots, "default", {}, void 0, !0)
          ],
          2
          /* CLASS */
        )) : s("v-if", !0),
        F.value ? (i(), b(
          "div",
          {
            key: 3,
            class: S(["m-btn__icon-right", { "maz-invisible": y.value }])
          },
          [
            s(`
        @slot right-icon - The icon to display on the right of the button
      `),
            p(a.$slots, "right-icon", {}, () => [
              typeof a.rightIcon == "string" ? (i(), c(m(f), {
                key: 0,
                name: a.rightIcon
              }, null, 8, ["name"])) : a.rightIcon ? (i(), c(D(a.rightIcon), { key: 1 })) : s("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : s("v-if", !0),
        y.value ? (i(), c(m(e), {
          key: 4,
          class: "m-btn-loader",
          size: "2em",
          color: a.color
        }, null, 8, ["color"])) : s("v-if", !0)
      ]),
      _: 3
      /* FORWARDED */
    }, 8, ["disabled", "class", "type"]));
  }
}), G = (P, l) => {
  const e = P.__vccOpts || P;
  for (const [f, h] of l)
    e[f] = h;
  return e;
}, M = /* @__PURE__ */ G(X, [["__scopeId", "data-v-48f14785"]]), Y = {
  class: "m-pagination",
  role: "navigation",
  "aria-label": "page navigation"
}, Z = { class: "maz-sr-only" }, _ = { class: "maz-sr-only" }, x = ["id"], ee = { class: "maz-sr-only" }, ae = { class: "maz-sr-only" }, oe = { class: "maz-sr-only" }, ne = /* @__PURE__ */ q({
  __name: "MazPagination",
  props: {
    modelValue: { default: 1 },
    buttonProps: { default: void 0 },
    resultsSize: { default: void 0 },
    activeColor: { default: "primary" },
    size: { default: "md" },
    totalPages: {},
    pageRange: { default: 2 },
    loading: { type: Boolean }
  },
  emits: ["update:model-value"],
  setup(P, { emit: l }) {
    const e = P, f = l, h = N(() => import("./chevron-left-CcZaP3Rc.mjs")), T = N(() => import("./chevron-double-left-zoBBzOlh.mjs")), C = N(() => import("./ellipsis-horizontal-VGAsnCwV.mjs")), B = {
      size: "md",
      color: "theme",
      outline: !0,
      noElevation: !0,
      fab: !0
    }, r = t(() => ({
      ...B,
      ...e.buttonProps
    })), g = t(() => e.modelValue > 1 ? e.modelValue - 1 : 1), z = t(
      () => e.modelValue < e.totalPages ? e.modelValue + 1 : e.totalPages
    ), I = t(
      () => Array.from({ length: e.totalPages }, (o, u) => {
        const n = u + 1;
        return {
          number: n,
          isActive: n === e.modelValue,
          loading: n === e.modelValue && e.loading
        };
      })
    ), y = t(
      () => e.modelValue - e.pageRange > 1 ? I.value.slice(0, 1) : []
    ), E = t(
      () => e.modelValue < e.totalPages - e.pageRange ? I.value.slice(-1) : []
    ), F = t(() => e.modelValue - e.pageRange - 1 < 0 ? 0 : e.modelValue - e.pageRange - 1 > e.totalPages - e.pageRange ? e.totalPages - e.pageRange : e.modelValue - e.pageRange - 1), L = t(
      () => e.modelValue + e.pageRange > e.totalPages ? e.totalPages : e.modelValue + e.pageRange < e.pageRange ? e.pageRange : e.modelValue + e.pageRange
    ), U = t(() => I.value.slice(F.value, L.value)), j = t(
      () => e.modelValue - e.pageRange > 2 ? [{ divider: !0 }] : []
    ), a = t(
      () => e.modelValue < e.totalPages - e.pageRange - 1 ? [{ divider: !0 }] : []
    ), w = t(() => [
      ...y.value,
      ...j.value,
      ...U.value,
      ...a.value,
      ...E.value
    ]);
    function R(o) {
      o !== e.modelValue && f("update:model-value", o);
    }
    return (o, u) => (i(), b("nav", Y, [
      d("ul", null, [
        d("li", null, [
          v(M, $(r.value, {
            disabled: o.modelValue === 1,
            "aria-label": "First Page, Page 1",
            "aria-setsize": o.resultsSize ?? void 0,
            "aria-posinset": "1",
            size: o.size,
            onClick: u[0] || (u[0] = (n) => R(1))
          }), {
            default: V(() => [
              d("span", Z, [
                s(`
              @slot Accessible text for screen reader of the previous page button
                @binding {number} page - first page number
            `),
                p(o.$slots, "first-page-sr", { page: 1 }, () => [
                  k("First Page, page 1")
                ], !0)
              ]),
              v(m(T))
            ]),
            _: 3
            /* FORWARDED */
          }, 16, ["disabled", "aria-setsize", "size"])
        ]),
        d("li", null, [
          v(M, $(r.value, {
            disabled: o.modelValue === 1,
            "aria-label": `Previous Page, Page ${g.value}`,
            "aria-setsize": o.resultsSize ?? void 0,
            "aria-posinset": g.value,
            size: o.size,
            onClick: u[1] || (u[1] = (n) => R(g.value))
          }), {
            default: V(() => [
              d("span", _, [
                s(`
              @slot Accessible text for screen reader of the first page button
                @binding {number} page - previous page number
            `),
                p(o.$slots, "previous-page-sr", { page: g.value }, () => [
                  k(
                    " Previous Page, page " + A(g.value),
                    1
                    /* TEXT */
                  )
                ], !0)
              ]),
              v(m(h))
            ]),
            _: 3
            /* FORWARDED */
          }, 16, ["disabled", "aria-label", "aria-setsize", "aria-posinset", "size"])
        ]),
        (i(!0), b(
          Q,
          null,
          W(w.value, (n, O) => (i(), b("li", {
            id: "number" in n ? `button-${O}-${n.number}` : `ellipsis-${O}`,
            key: "number" in n ? `button-${O}-${n.number}` : `ellipsis-${O}`
          }, [
            "number" in n ? (i(), c(M, $({
              key: 0,
              ref_for: !0
            }, {
              ...r.value,
              color: n.isActive ? o.activeColor : r.value.color,
              outline: n.isActive ? !1 : r.value.outline
            }, {
              size: o.size,
              "aria-label": `Page ${n.number}`,
              "aria-current": n.isActive ? "true" : "false",
              "aria-setsize": o.resultsSize ?? void 0,
              loading: n.loading,
              "aria-posinset": n.number,
              class: { active: n.isActive },
              onClick: (te) => n.isActive ? void 0 : R(n.number)
            }), {
              default: V(() => [
                d("span", ee, [
                  s(`
                @slot Accessible text for screen reader of the current page button
                  @binding {number} page - current page number
              `),
                  p(o.$slots, "page-sr", {
                    page: n.number
                  }, () => [
                    k(
                      "Page " + A(n.number),
                      1
                      /* TEXT */
                    )
                  ], !0)
                ]),
                k(
                  " " + A(n.number),
                  1
                  /* TEXT */
                )
              ]),
              _: 2
              /* DYNAMIC */
            }, 1040, ["size", "aria-label", "aria-current", "aria-setsize", "loading", "aria-posinset", "class", "onClick"])) : n.divider ? (i(), b(
              "div",
              $({
                key: 1,
                ref_for: !0
              }, r.value, { class: "flex p-2 flex-center" }),
              [
                v(m(C))
              ],
              16
              /* FULL_PROPS */
            )) : s("v-if", !0)
          ], 8, x))),
          128
          /* KEYED_FRAGMENT */
        )),
        d("li", null, [
          v(M, $(r.value, {
            disabled: o.modelValue === o.totalPages,
            "aria-label": `Next Page, Page ${z.value}`,
            "aria-setsize": o.resultsSize ?? void 0,
            "aria-posinset": z.value,
            size: o.size,
            onClick: u[2] || (u[2] = (n) => R(z.value))
          }), {
            default: V(() => [
              d("span", ae, [
                s(`
              @slot Accessible text for screen reader of the next page button
                @binding {number} page - next page number
            `),
                p(o.$slots, "next-page-sr", { page: z.value }, () => [
                  k(
                    "Next Page, page " + A(z.value),
                    1
                    /* TEXT */
                  )
                ], !0)
              ]),
              v(m(h), { class: "-maz-rotate-180" })
            ]),
            _: 3
            /* FORWARDED */
          }, 16, ["disabled", "aria-label", "aria-setsize", "aria-posinset", "size"])
        ]),
        d("li", null, [
          v(M, $(r.value, {
            disabled: o.modelValue === o.totalPages,
            "aria-label": `Next Page, Page ${o.totalPages}`,
            "aria-setsize": o.resultsSize ?? void 0,
            "aria-posinset": o.totalPages,
            size: o.size,
            onClick: u[3] || (u[3] = (n) => R(o.totalPages))
          }), {
            default: V(() => [
              d("span", oe, [
                s(`
              @slot Accessible text for screen reader of the last page button
                @binding {number} page - last page number
            `),
                p(o.$slots, "last-page-sr", { page: o.totalPages }, () => [
                  k(
                    "Last Page, page " + A(o.totalPages),
                    1
                    /* TEXT */
                  )
                ], !0)
              ]),
              v(m(T), { class: "-maz-rotate-180" })
            ]),
            _: 3
            /* FORWARDED */
          }, 16, ["disabled", "aria-label", "aria-setsize", "aria-posinset", "size"])
        ])
      ])
    ]));
  }
}), se = /* @__PURE__ */ G(ne, [["__scopeId", "data-v-8b9651c3"]]);
export {
  se as M,
  G as _
};
