// maz-ui/components/chunks/MazPickerMonthSwitcher-DZPHWTJt.mjs
import { defineComponent as g, computed as M, openBlock as l, createElementBlock as d, createElementVNode as m, createVNode as u, withModifiers as h, withCtx as p, unref as n, normalizeClass as f, Fragment as v, renderList as w, createBlock as x, createTextVNode as D, toDisplayString as S } from "vue";
import { d as C, c as s, a as c, M as b, i as k, _ as B } from "./MazPicker-CJBezF2t.mjs";
import { X as q } from "./x-mark-CKwYEQQZ.mjs";
import '../assets/MazPickerMonthSwitcher.css';const $ = { class: "maz-picker-month-switcher" }, N = { class: "maz-picker-month-switcher__header" }, V = /* @__PURE__ */ g({
  __name: "MazPickerMonthSwitcher",
  props: {
    calendarDate: { type: String, required: !0 },
    color: { type: String, required: !0 },
    locale: { type: String, required: !0 },
    double: { type: Boolean, required: !0 }
  },
  emits: ["update:calendar-date", "close"],
  setup(r, { emit: z }) {
    const a = r, i = z, _ = M(() => Array.from({ length: 12 }, (o, e) => e).map((o) => {
      const e = C(a.calendarDate).set("month", o);
      return a.double ? {
        label: `${s(
          c(e.format(), a.locale, {
            month: "short"
          })
        )} - ${s(
          c(e.add(1, "month").format(), a.locale, {
            month: "short"
          })
        )}`,
        date: e
      } : {
        label: s(
          c(e.format(), a.locale, {
            month: "long"
          })
        ),
        date: e
      };
    }));
    function y(o) {
      i("update:calendar-date", o.format()), i("close");
    }
    return (o, e) => (l(), d("div", $, [
      m("div", N, [
        u(b, {
          size: "xs",
          color: "transparent",
          type: "button",
          onClick: e[0] || (e[0] = h((t) => o.$emit("close", t), ["stop"]))
        }, {
          default: p(() => [
            u(n(q), { class: "maz-text-lg" })
          ]),
          _: 1
          /* STABLE */
        })
      ]),
      m(
        "div",
        {
          class: f(["maz-picker-month-switcher__main", { "--has-double": r.double }])
        },
        [
          (l(!0), d(
            v,
            null,
            w(_.value, (t) => (l(), x(b, {
              key: t.label,
              size: a.double ? "sm" : "xs",
              class: f({
                "--is-selected": n(k)(t.date, r.calendarDate, "month")
              }),
              color: n(k)(t.date, r.calendarDate, "month") ? r.color : "transparent",
              type: "button",
              onClick: h((E) => y(t.date), ["stop"])
            }, {
              default: p(() => [
                D(
                  S(t.label),
                  1
                  /* TEXT */
                )
              ]),
              _: 2
              /* DYNAMIC */
            }, 1032, ["size", "class", "color", "onClick"]))),
            128
            /* KEYED_FRAGMENT */
          ))
        ],
        2
        /* CLASS */
      )
    ]));
  }
}), j = /* @__PURE__ */ B(V, [["__scopeId", "data-v-74b3d289"]]);
export {
  j as default
};
