// maz-ui/modules/chunks/index-D7R1dvsf.mjs
import '../assets/index.css';var Ge = Object.defineProperty;
var Je = (s, e, t) => e in s ? Ge(s, e, { enumerable: !0, configurable: !0, writable: !0, value: t }) : s[e] = t;
var a = (s, e, t) => Je(s, typeof e != "symbol" ? e + "" : e, t);
import { inject as et, createVNode as U, render as re, nextTick as tt, defineComponent as De, defineAsyncComponent as k, ref as h, computed as f, watch as ke, onMounted as de, onBeforeUnmount as Me, openBlock as z, createElementBlock as W, withModifiers as H, withKeys as st, withDirectives as Pe, unref as O, vShow as $e, createCommentVNode as x, createElementVNode as D, normalizeClass as Ee, createBlock as q, Transition as nt, withCtx as ye, resolveDynamicComponent as it, toDisplayString as be, createTextVNode as ot, normalizeStyle as rt, getCurrentInstance as at, onUnmounted as lt } from "vue";
function Se(s, e) {
  const t = et(s, e);
  if (!t)
    throw new TypeError(`[maz-ui](injectStrict) Could not resolve ${s}`);
  return t;
}
function Es(s, e) {
  let t;
  return function(...n) {
    clearTimeout(t), t = setTimeout(() => {
      s.apply(this, n);
    }, e);
  };
}
function ct(s) {
  return !!s;
}
function _e(s, e) {
  let t = e == null ? void 0 : e.element;
  function n() {
    t && re(null, t);
  }
  const i = U(
    s,
    { ...e == null ? void 0 : e.props, ...e != null && e.addDestroyInProps ? { destroy: n } : {} },
    e == null ? void 0 : e.children
  );
  return e != null && e.app ? (i.appContext = e.app._context, t ? re(i, t) : typeof document < "u" && (t = document.createElement("div"), re(i, t))) : (t = t ?? document.body, re(i, t)), { vNode: i, destroy: n, el: t };
}
function E() {
  return typeof document < "u";
}
class ut {
  constructor(e, t) {
    a(this, "defaultOptions", {
      element: void 0,
      timeout: 60 * 1e3 * 5,
      // 5 minutes
      once: !1,
      immediate: !0
    });
    a(this, "options");
    a(this, "timeoutHandler");
    a(this, "isIdle", !1);
    a(this, "isDestroy", !1);
    a(this, "startTime", 0);
    a(this, "remainingTime", 0);
    a(this, "lastClientX", -1);
    a(this, "lastClientY", -1);
    a(this, "eventNames", [
      "DOMMouseScroll",
      "mousedown",
      "mousemove",
      "mousewheel",
      "MSPointerDown",
      "MSPointerMove",
      "keydown",
      "touchmove",
      "touchstart",
      "wheel",
      "click"
    ]);
    a(this, "handleEvent", (e) => {
      try {
        if (this.remainingTime > 0)
          return;
        if (e.type === "mousemove") {
          const { clientX: t, clientY: n } = e;
          if (t === void 0 && n === void 0 || t === this.lastClientX && n === this.lastClientY)
            return;
          this.lastClientX = t, this.lastClientY = n;
        }
        this.resetTimeout(), this.callback({ isIdle: this.isIdle, eventType: e.type });
      } catch (t) {
        throw new Error(`[IdleTimeout](handleEvent) ${t}`);
      }
    });
    this.callback = e, this.options = {
      ...this.defaultOptions,
      ...t
    }, E() && this.start();
  }
  get element() {
    return this.options.element ?? document.body;
  }
  start() {
    if (!E()) {
      console.warn("[IdleTimeout](start) you should run this method on client side");
      return;
    }
    for (const e of this.eventNames)
      this.element.addEventListener(e, this.handleEvent);
    this.resetTimeout(), this.options.immediate && this.callback({ isIdle: !1 });
  }
  pause() {
    const e = this.startTime + this.options.timeout - Date.now();
    e <= 0 || (this.remainingTime = e, this.timeoutHandler && (clearTimeout(this.timeoutHandler), this.timeoutHandler = void 0));
  }
  resume() {
    this.remainingTime <= 0 || (this.resetTimeout(), this.callback({ isIdle: this.isIdle }), this.remainingTime = 0);
  }
  reset() {
    this.isDestroy = !1, this.isIdle = !1, this.remainingTime = 0, this.resetTimeout(), this.callback({ isIdle: this.isIdle });
  }
  destroy() {
    if (!E()) {
      console.warn("[IdleTimeout](destroy) you should run this method on client side");
      return;
    }
    this.isDestroy = !0;
    for (const e of this.eventNames)
      this.element.removeEventListener(e, this.handleEvent);
    this.timeoutHandler && clearTimeout(this.timeoutHandler);
  }
  resetTimeout() {
    this.isIdle = !1, this.timeoutHandler && (clearTimeout(this.timeoutHandler), this.timeoutHandler = void 0), this.timeoutHandler = setTimeout(
      this.handleTimeout.bind(this),
      this.remainingTime || this.options.timeout
    ), this.startTime = Date.now();
  }
  handleTimeout() {
    this.isIdle = !0, this.callback({ isIdle: this.isIdle }), this.options.once && this.destroy();
  }
  get destroyed() {
    return this.isDestroy;
  }
  get timeout() {
    return this.options.timeout;
  }
  set timeout(e) {
    this.options.timeout = e;
  }
  get idle() {
    return this.isIdle;
  }
  set idle(e) {
    e ? this.handleTimeout() : this.reset(), this.callback({ isIdle: this.isIdle });
  }
}
class mt {
  constructor(e, t) {
    a(this, "eventHandlerFunction");
    a(this, "event", "visibilitychange");
    a(this, "timeoutHandler");
    a(this, "options");
    a(this, "defaultOptions", {
      timeout: 5e3,
      once: !1,
      immediate: !0
    });
    a(this, "isVisible", !1);
    this.callback = e, this.options = {
      ...this.defaultOptions,
      ...t
    }, this.eventHandlerFunction = this.eventHandler.bind(this), E() && this.start();
  }
  start() {
    if (!E()) {
      console.warn("[UserVisibility](start) you should run this method on client side");
      return;
    }
    this.options.immediate && this.emitCallback(), this.addEventListener();
  }
  emitCallback() {
    this.isVisible = document.visibilityState === "visible", this.callback({ isVisible: this.isVisible }), this.options.once && this.destroy();
  }
  eventHandler() {
    document.visibilityState === "visible" && !this.isVisible ? (this.clearTimeout(), this.emitCallback()) : document.visibilityState === "hidden" && this.initTimeout();
  }
  clearTimeout() {
    this.timeoutHandler && (clearTimeout(this.timeoutHandler), this.timeoutHandler = void 0);
  }
  initTimeout() {
    this.clearTimeout(), this.timeoutHandler = setTimeout(this.emitCallback.bind(this), this.options.timeout);
  }
  addEventListener() {
    document.addEventListener(this.event, this.eventHandlerFunction);
  }
  removeEventListener() {
    document.removeEventListener(this.event, this.eventHandlerFunction);
  }
  destroy() {
    this.removeEventListener(), this.timeoutHandler && clearTimeout(this.timeoutHandler);
  }
}
function dt(s) {
  return new Promise((e) => setTimeout(e, s));
}
class Ls {
  constructor({ src: e, identifier: t, once: n = !0, async: i = !0, defer: o = !0 }) {
    a(this, "src");
    a(this, "script");
    a(this, "once");
    a(this, "async");
    a(this, "defer");
    a(this, "identifier");
    if (typeof window > "u")
      throw new TypeError("[ScriptLoader]: Is supported only on browser side");
    if (!e)
      throw new Error('[ScriptLoader]: You should provide the attribut "src"');
    if (!t)
      throw new Error('[ScriptLoader]: You should provide the attribut "identifier"');
    this.src = e, this.identifier = t, this.once = n, this.async = i, this.defer = o;
  }
  removeTag(e) {
    var t;
    typeof e == "string" ? (t = document.head.querySelector(`[data-identifier="${e}"]`)) == null || t.remove() : e.remove();
  }
  load() {
    const e = window, t = document.head.querySelectorAll(`[data-identifier="${this.identifier}"]`);
    if (this.once && e[this.identifier] && t.length > 0)
      return this.script = e[this.identifier], Promise.resolve(this.script);
    if (!this.once && t.length > 0)
      for (const n of t)
        this.removeTag(n);
    return this.injectScript();
  }
  injectScript() {
    const e = window;
    return new Promise((t, n) => {
      try {
        const i = document.createElement("script");
        i.src = this.src, i.async = this.async, i.defer = this.defer, i.dataset.identifier = this.identifier, i.addEventListener("error", (o) => n(new Error(`[ScriptLoader](injectScript) ${o.message}`))), i.addEventListener("load", (o) => (this.script = o, e[this.identifier] = o, t(o))), document.head.append(i);
      } catch (i) {
        throw new Error(`[ScriptLoader](init) ${i}`);
      }
    });
  }
}
function Ts(s) {
  return [...s].map((e) => {
    const t = e.codePointAt(0);
    return t ? t % 32 + 127461 : void 0;
  }).filter(ct).map((e) => String.fromCodePoint(e)).join("");
}
class ht {
  constructor(e) {
    a(this, "defaultOptions", {
      preventDefaultOnTouchMove: !1,
      preventDefaultOnMouseWheel: !1,
      threshold: 50,
      immediate: !1,
      triggerOnEnd: !1
    });
    a(this, "element");
    a(this, "xStart");
    a(this, "yStart");
    a(this, "xEnd");
    a(this, "yEnd");
    a(this, "xDiff");
    a(this, "yDiff");
    a(this, "onToucheStartCallback");
    a(this, "onToucheMoveCallback");
    a(this, "onToucheEndCallback");
    a(this, "onMouseWheelCallback");
    a(this, "options");
    if (this.inputOption = e, this.options = { ...this.defaultOptions, ...e }, !this.options.element)
      throw new Error(
        "[SwipeHandler] Element should be provided. Its can be a string selector or an HTMLElement"
      );
    if (typeof this.options.element == "string") {
      const t = document.querySelector(this.options.element);
      if (!(t instanceof HTMLElement))
        throw new TypeError("[SwipeHandler] String selector for element is not found");
      this.element = t;
    } else
      this.element = this.options.element;
    this.onToucheStartCallback = this.toucheStartHandler.bind(this), this.onToucheMoveCallback = this.handleTouchMove.bind(this), this.onToucheEndCallback = this.handleTouchEnd.bind(this), this.onMouseWheelCallback = this.handleMouseWheel.bind(this), this.options.immediate && this.start();
  }
  start() {
    this.element.addEventListener("touchstart", this.onToucheStartCallback, { passive: !0 }), this.element.addEventListener("touchmove", this.onToucheMoveCallback, { passive: !0 }), this.options.triggerOnEnd && this.element.addEventListener("touchend", this.onToucheEndCallback, { passive: !0 }), this.options.preventDefaultOnMouseWheel && this.element.addEventListener("mousewheel", this.onMouseWheelCallback, { passive: !1 });
  }
  stop() {
    this.element.removeEventListener("touchstart", this.onToucheStartCallback), this.element.removeEventListener("touchmove", this.onToucheMoveCallback), this.element.removeEventListener("touchend", this.onToucheEndCallback), this.options.preventDefaultOnMouseWheel && this.element.removeEventListener("mousewheel", this.onMouseWheelCallback);
  }
  handleMouseWheel(e) {
    e.preventDefault();
  }
  toucheStartHandler(e) {
    this.xStart = e.touches[0].clientX, this.yStart = e.touches[0].clientY, this.emitValuesChanged();
  }
  emitValuesChanged() {
    var e, t;
    (t = (e = this.options).onValuesChanged) == null || t.call(e, {
      xStart: this.xStart,
      yStart: this.yStart,
      xEnd: this.xEnd,
      yEnd: this.yEnd,
      xDiff: this.xDiff,
      yDiff: this.yDiff
    });
  }
  handleTouchMove(e) {
    this.options.preventDefaultOnTouchMove && e.cancelable && e.preventDefault(), this.xEnd = e.touches[0].clientX, this.yEnd = e.touches[0].clientY, !(!this.xStart || !this.yStart) && (this.xDiff = this.xStart - this.xEnd, this.yDiff = this.yStart - this.yEnd, this.emitValuesChanged(), this.options.triggerOnEnd || this.runCallbacks(e));
  }
  handleTouchEnd(e) {
    this.runCallbacks(e), this.emitValuesChanged();
  }
  runCallbacks(e) {
    var t, n, i, o, r, c, m, v;
    typeof this.xDiff != "number" || typeof this.yDiff != "number" || Math.abs(this.xDiff) < this.options.threshold && Math.abs(this.yDiff) < this.options.threshold || (Math.abs(this.xDiff) > Math.abs(this.yDiff) ? this.xDiff > 0 ? (n = (t = this.options).onLeft) == null || n.call(t, e) : (o = (i = this.options).onRight) == null || o.call(i, e) : this.yDiff > 0 ? (c = (r = this.options).onUp) == null || c.call(r, e) : (v = (m = this.options).onDown) == null || v.call(m, e), this.resetValues());
  }
  resetValues() {
    this.xStart = void 0, this.yStart = void 0, this.xEnd = void 0, this.yEnd = void 0, this.xDiff = void 0, this.yDiff = void 0, this.emitValuesChanged();
  }
}
function Is(s, e) {
  let t = 0;
  return async function() {
    const n = Date.now();
    n - t >= e && (await s(), t = n);
  };
}
const ft = {
  removeAccents: !0,
  caseSensitive: !1,
  replaceSpaces: !0,
  removeSpecialCharacters: !1,
  trim: !0,
  normalizeSpaces: !0,
  removeNumbers: !1,
  customNormalizationForms: ["NFC", "NFKD"]
};
function xe(s, e) {
  const t = { ...ft, ...e }, n = {
    À: "A",
    Á: "A",
    Â: "A",
    Ã: "A",
    Ä: "A",
    Å: "A",
    à: "a",
    á: "a",
    â: "a",
    ã: "a",
    ä: "a",
    å: "a",
    È: "E",
    É: "E",
    Ê: "E",
    Ë: "E",
    è: "e",
    é: "e",
    ê: "e",
    ë: "e",
    Î: "I",
    Ï: "I",
    í: "I",
    î: "i",
    ï: "i",
    Ô: "O",
    Õ: "O",
    Ö: "O",
    Ø: "O",
    ô: "o",
    õ: "o",
    ö: "o",
    ø: "o",
    Ù: "U",
    Ú: "U",
    Û: "U",
    Ü: "U",
    ù: "u",
    ú: "u",
    û: "u",
    ü: "u",
    Ç: "C",
    ç: "c",
    ÿ: "y",
    Ñ: "N",
    ñ: "n",
    ó: "o"
  };
  let i = s.toString();
  if (t.trim && (i = i.trim()), t.normalizeSpaces && (i = i.replaceAll(/\s+/g, " ")), t.replaceSpaces && (i = i.replaceAll(" ", "-")), t.removeNumbers && (i = i.replaceAll(/\d/g, "")), t.removeAccents && (i = i.replaceAll(/[ÀÁÂÃÄÅÇÈÉÊËÎÏÑÔÕÖØÙÚÛÜàáâãäåçèéêëíîïñóôõöøùúûüÿ]/g, (o) => n[o] || o), i = i.replaceAll(/[\u0300-\u036F]/g, "")), t.caseSensitive === !1 && (i = i.toLowerCase()), t.removeSpecialCharacters && (i = i.replaceAll(/[^\dA-Z-]/gi, "")), t.trim && (i = i.trim()), t.customNormalizationForms)
    for (const o of t.customNormalizationForms)
      i = i.normalize(o);
  return i;
}
const me = "__maz-click-outside__";
function Fe() {
  return document.ontouchstart === null ? "touchstart" : "click";
}
async function Ne(s, e) {
  try {
    Ve(s);
    const t = e.instance, n = e.value, i = typeof n == "function";
    if (!i)
      throw new Error("[maz-ui](vClickOutside) the callback should be a function");
    await tt(), s[me] = (r) => {
      if ((!s || r.target && !s.contains(r.target)) && n && i)
        return n.call(t, r);
    };
    const o = Fe();
    document.addEventListener(o, s[me], { passive: !0 });
  } catch (t) {
    console.error("[maz-ui](vClickOutside)", t);
  }
}
function Ve(s) {
  try {
    const e = Fe();
    document.removeEventListener(e, s[me], !1), delete s[me];
  } catch (e) {
    console.error("[maz-ui](vClickOutside)", e);
  }
}
function vt(s, e) {
  try {
    if (e.value === e.oldValue)
      return;
    Ne(s, e);
  } catch (t) {
    console.error("[maz-ui](vClickOutside)", t);
  }
}
const pt = {
  mounted: Ne,
  updated: vt,
  unmounted: Ve
}, gt = {
  install: (s) => {
    s.directive("click-outside", pt);
  }
};
function Be(s, e, t) {
  var r;
  s.stopPropagation();
  const n = typeof t.value == "function" ? t.value : t.value.handler, i = typeof t.value == "object" ? t.value.exclude : void 0;
  let o = !1;
  if (i && i.length > 0) {
    for (const c of i)
      if (!o && s.target instanceof HTMLElement) {
        const m = (r = document.querySelector(c)) == null ? void 0 : r.getAttribute("id");
        o = s.target.getAttribute("id") === m;
      }
  }
  !e.contains(s.target) && !o && (n == null || n());
}
function qe() {
  return document.ontouchstart === null ? "touchstart" : "click";
}
function yt(s, e) {
  const t = qe();
  document.removeEventListener(t, (n) => Be(n, s, e));
}
function bt(s, e) {
  if (typeof e.value != "function" && typeof e.value == "object" && typeof e.value.handler != "function") {
    console.error("[maz-ui](vClosable) v-closable directive requires a handler function");
    return;
  }
  const t = qe();
  document.addEventListener(t, (n) => Be(n, s, e));
}
const wt = {
  mounted: bt,
  unmounted: yt
}, zt = {
  install: (s) => {
    s.directive("closable", wt);
  }
}, Le = {
  position: "top"
};
class Ue {
  constructor(e = {}) {
    a(this, "options");
    this.options = {
      ...Le,
      ...e
    };
  }
  getPosition({ modifiers: e, value: t }) {
    return e.top ? "top" : e.bottom ? "bottom" : e.left ? "left" : e.right ? "right" : typeof t == "string" ? "top" : t.position ?? this.options.position;
  }
  getText({ value: e }) {
    return typeof e == "string" ? e : e.text;
  }
  getOpen({ value: e }) {
    return typeof e == "string" ? !1 : e.open ?? !1;
  }
  getColor({ value: e }) {
    return typeof e == "string" ? "default" : e.color ?? "default";
  }
  create(e, t) {
    e.setAttribute("data-tooltip", this.getText(t)), e.classList.add("m-tooltip");
    const n = this.getPosition(t);
    e.classList.add(`m-tooltip--${n}`), e.classList.add(`m-tooltip--${this.getColor(t)}`), this.getOpen(t) && e.classList.add("m-tooltip--open");
  }
  update(e, t) {
    this.remove(e, t), this.create(e, t);
  }
  remove(e, t) {
    e.removeAttribute("data-tooltip"), e.classList.remove("m-tooltip"), e.classList.remove("m-tooltip--top"), e.classList.remove("m-tooltip--bottom"), e.classList.remove("m-tooltip--left"), e.classList.remove("m-tooltip--right"), e.classList.remove("m-tooltip--open"), e.classList.remove(`m-tooltip--${this.getColor(t)}`);
  }
}
let ae;
const ks = {
  beforeMount(s, e) {
    const t = typeof e.value == "object" ? e.value : {};
    return ae = new Ue(t), ae.create(s, e);
  },
  updated(s, e) {
    return ae.update(s, e);
  },
  unmounted(s, e) {
    return ae.remove(s, e);
  }
}, Et = {
  install: (s, e = Le) => {
    const t = {
      ...Le,
      ...e
    }, n = new Ue(t);
    s.directive("tooltip", {
      beforeMount: n.create.bind(n),
      updated: n.update.bind(n),
      unmounted: n.remove.bind(n)
    });
  }
}, Lt = `
.maz-zoom-img {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 1rem;
  z-index: 1050;
  background-color: hsla(238, 15%, 40%, 0.7);
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.maz-zoom-img,
.maz-zoom-img * {
  box-sizing: border-box;
}

.maz-zoom-img .maz-zoom-img__wrapper {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  max-width: 100%;
  max-height: 100%;
  transition: all 300ms ease-in-out;
  opacity: 0;
  transform: scale(0.5);
}

.maz-zoom-img.maz-animate .maz-zoom-img__wrapper {
  opacity: 1;
  transform: scale(1);
}

.maz-zoom-img.maz-animate .maz-zoom-img__loader {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: hsla(238, 15%, 40%, 0.7);
  border-radius: 1rem;
  z-index: 2;
  min-width: 60px;
  min-height: 60px;
}
.maz-zoom-img.maz-animate .maz-zoom-img__loader[hidden] {
  display: none;
}

@-webkit-keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}

.maz-zoom-img.maz-animate .maz-zoom-img__loader__svg {
  animation: spin .6s linear infinite;
}

.maz-zoom-img img {
  max-width: 100%;
  max-height: 100%;
  min-width: 0;
  border-radius: 1rem;
}

.maz-zoom-img .maz-zoom-btn {
  margin: 0 auto;
  border: none;
  background-color: hsla(0, 0%, 7%, 0.5);
  box-shadow: 0 0 0.5rem 0 hsla(0, 0%, 0%, 0.2);
  height: 2.2rem;
  min-height: 2.2rem;
  width: 2.2rem;
  min-width: 2.2rem;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 2.2rem;
  cursor: pointer;
  flex: 0 0 auto;
  outline: none;
}

.maz-zoom-img .maz-zoom-btn svg {
  fill: white;
}

.maz-zoom-img .maz-zoom-btn.maz-zoom-btn--close {
  position: absolute;
  top: 0.5rem;
  right: 0.5rem;
  z-index: 1;
}

.maz-zoom-img .maz-zoom-btn.maz-zoom-btn--previous {
  position: absolute;
  left: 0.5rem;
  z-index: 1;
}

.maz-zoom-img .maz-zoom-btn.maz-zoom-btn--next {
  position: absolute;
  right: 0.5rem;
  z-index: 1;
}

.maz-zoom-img .maz-zoom-btn:hover {
  background-color: hsl(0, 0%, 0%);
}`, Ae = {
  close: '<svg xmlns="http://www.w3.org/2000/svg" height="24" viewBox="0 0 24 24" width="24"><path d="M0 0h24v24H0z" fill="none"/><path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>',
  next: '<svg xmlns="http://www.w3.org/2000/svg" height="24" viewBox="0 0 24 24" width="24"><path d="M0 0h24v24H0z" fill="none"/><path d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"/></svg>',
  previous: '<svg xmlns="http://www.w3.org/2000/svg" height="24" viewBox="0 0 24 24" width="24"><path d="M0 0h24v24H0z" fill="none"/><path d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"/></svg>',
  spinner: '<svg width="40px" height="40px" version="1.1" xmlns="http://www.w3.org/2000/svg" fill="currentColor" x="0px" y="0px" viewBox="0 0 50 50" xml:space="preserve" class="maz-zoom-img__loader__svg" data-v-6d1cb50c=""><path d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"></path></svg>'
};
class Tt {
  constructor(e) {
    a(this, "options");
    a(this, "loader");
    a(this, "wrapper");
    a(this, "img");
    a(this, "keydownHandler");
    a(this, "onImgLoadedCallback");
    a(this, "buttonsAdded");
    a(this, "defaultOptions", {
      scale: !0,
      blur: !0,
      disabled: !1
    });
    a(this, "mouseEnterListener");
    a(this, "mouseLeaveListener");
    a(this, "renderPreviewListener");
    if (!e.value)
      throw new Error(
        '[MazUI](zoom-img) Image path must be defined. Ex: `v-zoom-img="<PATH_TO_IMAGE>"`'
      );
    if (typeof e.value == "object" && !e.value.src)
      throw new Error("[maz-ui](zoom-img) src of image must be provided");
    this.buttonsAdded = !1, this.options = this.buildOptions(e), this.keydownHandler = this.keydownLister.bind(this), this.loader = this.getLoader(), this.wrapper = document.createElement("div"), this.wrapper.classList.add("maz-zoom-img__wrapper"), this.wrapper.prepend(this.loader), this.img = document.createElement("img"), this.onImgLoadedCallback = this.onImgLoaded.bind(this), this.imgEventHandler(!0);
  }
  buildOptions(e) {
    return {
      ...this.defaultOptions,
      ...typeof e.value == "object" ? e.value : { src: e.value }
    };
  }
  get allInstances() {
    return [...document.querySelectorAll(".maz-zoom-img-instance")];
  }
  create(e) {
    this.options.disabled || (e.style.cursor = "pointer", setTimeout(() => e.classList.add("maz-zoom-img-instance")), e.setAttribute("data-zoom-src", this.options.src), this.options.alt && e.setAttribute("data-zoom-alt", this.options.alt), e.style.transition = "all 300ms ease-in-out", this.mouseEnterListener = () => this.mouseEnter(e), this.mouseLeaveListener = () => this.mouseLeave(e), this.renderPreviewListener = () => this.renderPreview(e, this.options), e.addEventListener("mouseenter", this.mouseEnterListener), e.addEventListener("mouseleave", this.mouseLeaveListener), e.addEventListener("click", this.renderPreviewListener));
  }
  update(e) {
    this.options = this.buildOptions(e);
  }
  remove(e) {
    this.imgEventHandler(!1), e.removeEventListener("mouseenter", this.mouseEnterListener), e.removeEventListener("mouseleave", this.mouseLeaveListener), e.removeEventListener("click", this.renderPreviewListener), e.classList.remove("maz-zoom-img-instance"), e.removeAttribute("data-zoom-src"), e.removeAttribute("data-zoom-alt"), e.style.cursor = "";
  }
  renderPreview(e, t) {
    e.classList.add(
      "maz-is-open"
      /* OPEN */
    ), this.addStyle(Lt);
    const n = document.createElement("div");
    n.classList.add("maz-zoom-img"), n.setAttribute("id", "MazImgPreviewFullsize"), n.addEventListener("click", (i) => {
      n.isEqualNode(i.target) && this.closePreview();
    }), typeof t == "object" && (this.img.setAttribute("src", t.src), t.alt && this.img.setAttribute("alt", t.alt), this.img.id = "MazImgElement"), this.wrapper.append(this.img), n.append(this.wrapper), document.body.append(n), this.keyboardEventHandler(!0), setTimeout(() => {
      n && n.classList.add("maz-animate");
    }, 100);
  }
  onImgLoaded() {
    this.wrapper.style.width = `${this.img.width}px`, this.wrapper.style.minWidth = "200px", this.loader.hidden = !0;
    const e = this.getButton(), t = [], n = this.allInstances.length > 1;
    if (!this.buttonsAdded) {
      if (this.buttonsAdded = !0, n) {
        const i = this.getButton("previous"), o = this.getButton("next");
        t.push(i, o);
      }
      this.wrapper.append(e), n && (this.wrapper.prepend(t[0]), this.wrapper.append(t[1]));
    }
  }
  getLoader() {
    const e = document.createElement("div");
    return e.classList.add("maz-zoom-img__loader"), e.innerHTML = Ae.spinner, e;
  }
  mouseLeave(e) {
    this.options.scale && (e.style.transform = ""), this.options.blur && (e.style.filter = ""), e.style.zIndex = "";
  }
  mouseEnter(e) {
    e.style.zIndex = "1", this.options.scale && (e.style.transform = "scale(1.1)"), this.options.blur && (e.style.filter = "blur(2px)");
  }
  keydownLister(e) {
    e.preventDefault(), (e.key === "Escape" || e.key === " ") && this.closePreview(), (e.key === "ArrowLeft" || e.key === "ArrowRight") && this.nextPreviousImage(e.key === "ArrowRight");
  }
  getButton(e = "close") {
    const t = document.createElement("button");
    t.innerHTML = Ae[e];
    const n = e === "close" ? this.closePreview() : this.allInstances ? this.nextPreviousImage(e === "next") : null;
    return t.addEventListener("click", () => n), t.classList.add("maz-zoom-btn"), t.classList.add(`maz-zoom-btn--${e}`), t;
  }
  closePreview() {
    const e = document.querySelector("#MazImgPreviewFullsize"), t = document.querySelector("#MazPreviewStyle"), n = document.querySelector(
      ".maz-zoom-img-instance.maz-is-open"
    );
    n && n.classList.remove(
      "maz-is-open"
      /* OPEN */
    ), e && e.classList.remove("maz-animate"), this.keyboardEventHandler(!1), setTimeout(() => {
      e && e.remove(), t && t.remove();
    }, 300);
  }
  getNewInstanceIndex(e) {
    return e < 0 ? this.allInstances.length - 1 : e >= this.allInstances.length ? 0 : e;
  }
  nextPreviousImage(e) {
    const t = e, n = document.querySelector(
      ".maz-zoom-img-instance.maz-is-open"
    );
    if (n) {
      const i = this.allInstances.indexOf(n), o = t ? i + 1 : i - 1, r = this.allInstances[this.getNewInstanceIndex(o)];
      r && this.useNextInstance(n, r);
    }
  }
  useNextInstance(e, t) {
    e.classList.remove(
      "maz-is-open"
      /* OPEN */
    ), t.classList.add(
      "maz-is-open"
      /* OPEN */
    );
    const n = t.getAttribute("data-zoom-src"), i = t.getAttribute("data-zoom-alt");
    this.wrapper.style.width = "", this.loader.hidden = !1, n && this.img.setAttribute("src", n), i && this.img.setAttribute("alt", i);
  }
  addStyle(e) {
    const t = document.createElement("style");
    t.id = "MazPreviewStyle", t.textContent = e, document.head.append(t);
  }
  keyboardEventHandler(e) {
    if (e)
      return document.addEventListener("keydown", this.keydownHandler);
    document.removeEventListener("keydown", this.keydownHandler);
  }
  imgEventHandler(e) {
    if (e)
      return this.img.addEventListener("load", this.onImgLoadedCallback);
    this.img.removeEventListener("load", this.onImgLoadedCallback);
  }
}
let le;
const It = {
  created(s, e) {
    le = new Tt(e), le.create(s);
  },
  updated(s, e) {
    le.update(e);
  },
  unmounted(s) {
    le.remove(s);
  }
}, kt = {
  install(s) {
    s.directive("zoom-img", It);
  }
}, St = "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7", Te = {
  baseClass: "m-lazy-img",
  loadedClass: "m-lazy-loaded",
  loadingClass: "m-lazy-loading",
  errorClass: "m-lazy-error",
  noPhotoClass: "m-lazy-no-photo",
  noPhoto: !1,
  observerOnce: !0,
  loadOnce: !1,
  noUseErrorPhoto: !1,
  observerOptions: {
    threshold: 0.1
  }
};
class We {
  constructor(e = {}) {
    a(this, "observers", []);
    a(this, "defaultOptions", Te);
    a(this, "options");
    a(this, "onImgLoadedCallback");
    a(this, "onImgErrorCallback");
    a(this, "hasImgLoaded", !1);
    this.options = this.buildOptions(e), this.onImgLoadedCallback = this.imageIsLoaded.bind(this), this.onImgErrorCallback = this.imageHasError.bind(this);
  }
  async loadErrorPhoto() {
    const { default: e } = await import("./no-photography-BJX8HSus.mjs");
    return e;
  }
  buildOptions(e) {
    return {
      ...this.defaultOptions,
      ...e,
      observerOptions: {
        ...this.defaultOptions.observerOptions,
        ...e.observerOptions
      }
    };
  }
  removeClass(e, t) {
    e.classList.remove(t);
  }
  addClass(e, t) {
    e.classList.add(t);
  }
  removeAllStateClasses(e) {
    this.removeClass(e, this.options.loadedClass), this.removeClass(e, this.options.loadingClass), this.removeClass(e, this.options.errorClass), this.removeClass(e, this.options.noPhotoClass);
  }
  setBaseClass(e) {
    this.addClass(e, this.options.baseClass);
  }
  imageIsLoading(e) {
    var t, n;
    this.addClass(e, this.options.loadingClass), (n = (t = this.options).onLoading) == null || n.call(t, e);
  }
  imageHasNoPhoto(e) {
    this.removeClass(e, this.options.loadingClass), this.addClass(e, this.options.noPhotoClass), this.setDefaultPhoto(e);
  }
  imageIsLoaded(e) {
    var t, n;
    this.hasImgLoaded = !0, this.removeClass(e, this.options.loadingClass), this.addClass(e, this.options.loadedClass), (n = (t = this.options).onLoaded) == null || n.call(t, e);
  }
  imageHasError(e) {
    var t, n;
    this.removeClass(e, this.options.loadingClass), this.addClass(e, this.options.errorClass), (n = (t = this.options).onError) == null || n.call(t, e), this.setDefaultPhoto(e);
  }
  getSrc(e) {
    return typeof e.value == "object" ? e.value.src : e.value;
  }
  getImageUrl(e, t) {
    const n = this.getImgElement(e).getAttribute("data-lazy-src");
    return n || this.getSrc(t);
  }
  async setPictureSourceUrls(e) {
    const t = e.querySelectorAll("source");
    if (t.length > 0)
      for await (const n of t) {
        const i = n.getAttribute("data-lazy-srcset");
        if (i)
          n.srcset = i;
        else
          return this.imageHasError(e);
      }
    else
      this.imageHasError(e);
  }
  hasBgImgMode(e) {
    return e.arg === "bg-image";
  }
  isPictureElement(e) {
    return e instanceof HTMLPictureElement;
  }
  getImgElement(e) {
    return this.isPictureElement(e) ? e.querySelector("img") : e;
  }
  async setDefaultPhoto(e) {
    if (this.options.noUseErrorPhoto)
      return;
    const t = this.options.fallbackSrc ?? this.options.errorPhoto;
    typeof t == "string" && this.addClass(e, this.options.noPhotoClass);
    const n = t ?? await this.loadErrorPhoto(), i = e.querySelectorAll("source");
    if (i.length > 0)
      for await (const o of i)
        o.srcset = n;
    else
      this.setImgSrc(e, n);
  }
  addEventListenerToImg(e) {
    const t = this.getImgElement(e);
    t.addEventListener("load", () => this.onImgLoadedCallback(e), {
      once: !0
    }), t.addEventListener("error", (n) => this.onImgErrorCallback(e, n), { once: !0 });
  }
  async loadImage(e, t) {
    if (this.imageIsLoading(e), this.isPictureElement(e))
      this.addEventListenerToImg(e), await this.setPictureSourceUrls(e);
    else {
      const n = this.getImageUrl(e, t);
      if (!n)
        return this.imageHasError(e);
      this.hasBgImgMode(t) ? (e.style.backgroundImage = `url('${n}')`, this.imageIsLoaded(e)) : (this.addEventListenerToImg(e), this.setImgSrc(e, n));
    }
  }
  setImgSrc(e, t) {
    const n = this.getImgElement(e);
    n.src = t;
  }
  handleIntersectionObserver(e, t, n, i) {
    var o, r;
    this.observers.push(i);
    for (const c of n)
      if (c.isIntersecting) {
        if ((r = (o = this.options).onIntersecting) == null || r.call(o, c.target), this.options.observerOnce && i.unobserve(e), this.options.loadOnce && this.hasImgLoaded)
          return;
        this.loadImage(e, t);
      }
  }
  createObserver(e, t) {
    const n = (r, c) => {
      this.handleIntersectionObserver(e, t, r, c);
    }, i = this.options.observerOptions;
    new IntersectionObserver(n, i).observe(e);
  }
  async imageHandler(e, t, n) {
    if (n === "update")
      for await (const i of this.observers) i.unobserve(e);
    window.IntersectionObserver ? this.createObserver(e, t) : this.loadImage(e, t);
  }
  async bindUpdateHandler(e, t, n) {
    if (this.options.noPhoto)
      return this.imageHasNoPhoto(e);
    await this.imageHandler(e, t, n);
  }
  async add(e, t) {
    if (this.hasBgImgMode(t) && this.isPictureElement(e))
      throw new Error(`[MazLazyImg] You can't use the "bg-image" mode with "<picture />" element`);
    setTimeout(() => this.setBaseClass(e), 0), e.getAttribute("src") || this.setImgSrc(e, St), await this.bindUpdateHandler(e, t, "bind");
  }
  async update(e, t) {
    t.value !== t.oldValue && (this.hasImgLoaded = !1, this.removeAllStateClasses(e), await this.bindUpdateHandler(e, t, "update"));
  }
  remove(e, t) {
    this.hasImgLoaded = !1, this.hasBgImgMode(t) && (e.style.backgroundImage = ""), this.removeAllStateClasses(e);
    for (const n of this.observers)
      n.unobserve(e);
    this.observers = [];
  }
}
let ce;
const Ss = {
  created(s, e) {
    const t = typeof e.value == "object" ? e.value : {};
    ce = new We(t), ce.add(s, e);
  },
  updated(s, e) {
    ce.update(s, e);
  },
  unmounted(s, e) {
    ce.remove(s, e);
  }
}, Ct = {
  install(s, e = {}) {
    const t = {
      ...Te,
      ...e,
      observerOptions: {
        ...Te.observerOptions,
        ...e.observerOptions
      }
    }, n = new We(t);
    s.directive("lazy-img", {
      created: n.add.bind(n),
      updated: n.update.bind(n),
      unmounted: n.remove.bind(n)
    });
  }
}, xt = ["onKeypress"], At = { class: "m-fullscreen-img-scroller" }, Ot = ["src", "alt"], Ht = /* @__PURE__ */ De({
  __name: "MazFullscreenImg",
  props: {
    src: {},
    clickedElementBounds: { default: void 0 },
    offset: { default: void 0 },
    animation: { default: () => ({
      duration: 300,
      easing: "ease-in-out"
    }) },
    openInstanceClass: { default: "m-fullscreen-img-instance" },
    clickedElement: {},
    destroy: { type: Function, default: void 0 },
    alt: { default: void 0 },
    zoom: { type: Boolean, default: !0 }
  },
  emits: ["close", "previous", "next", "before-close"],
  setup(s, { emit: e }) {
    const t = s, n = e, i = k(() => import("./MazSpinner--BVvGFIO.mjs")), o = k(() => import("./x-mark-BP8zWe_C.mjs")), r = k(() => import("./chevron-left-CcZaP3Rc.mjs")), c = h(!1), m = h(!1), v = h(!1), L = h(!1), b = h(!1), y = h(t.clickedElement), T = f(() => t.clickedElement.getBoundingClientRect()), ee = f(
      () => T.value.height < T.value.width
    ), _ = h(t.src), M = h(t.alt), te = h(), S = h(), R = f(() => b.value ? "--is-zoomed maz-cursor-zoom-out" : "maz-cursor-zoom-in");
    function Y() {
      c.value = !0, m.value = !1, v.value = !0;
    }
    ke(
      () => v.value,
      (l) => {
        l && ve();
      }
    );
    function P() {
      n("before-close"), u();
    }
    function $(l) {
      l.key === "Escape" && (l.preventDefault(), P()), (l.key === "ArrowLeft" || l.key === "ArrowRight") && (l.preventDefault(), K(l.key === "ArrowRight" ? "next" : "previous"));
    }
    function F() {
      document.documentElement.classList.add("--m-fullscreen-open");
    }
    function he() {
      document.documentElement.classList.remove("--m-fullscreen-open");
    }
    function se() {
      return [...document.querySelectorAll(".m-fullscreen-img-instance")];
    }
    function ne(l, d) {
      return d < 0 ? l.length - 1 : d >= l.length ? 0 : d;
    }
    async function ie(l, d) {
      l.classList.remove(t.openInstanceClass), d.classList.add(t.openInstanceClass);
      const g = d.getAttribute("data-src"), w = d.getAttribute("data-alt");
      M.value = w, g && (_.value = g), c.value = !1, c.value || (m.value = !0);
    }
    function K(l) {
      const d = document.querySelector(
        `.m-fullscreen-img-instance.${t.openInstanceClass}`
      );
      if (d) {
        const g = se(), w = g.indexOf(d), C = l === "next" ? w + 1 : w - 1, I = g[ne(g, C)];
        y.value = I, I && ie(d, I), n(l), b.value ? N(S.value) : B();
      }
    }
    function N(l) {
      ee.value ? (l.style.width = `${window.innerWidth}px`, l.style.removeProperty("height")) : (l.style.height = `${window.innerHeight}px`, l.style.removeProperty("width")), l.style.removeProperty("top"), l.style.removeProperty("left");
    }
    async function fe() {
      const l = S.value;
      b.value ? (b.value = !b.value, B()) : (b.value = !b.value, N(l));
    }
    function X(l) {
      var d;
      return (d = S.value) == null ? void 0 : d.animate(l, {
        duration: t.animation.duration,
        // Durée de l'animation en millisecondes
        easing: t.animation.easing
        // Fonction d'interpolation pour l'animation inverse
      });
    }
    function oe(l = t.offset ?? 0) {
      const d = (y.value instanceof HTMLImageElement ? y.value.naturalWidth : y.value.clientWidth) || 1, g = (y.value instanceof HTMLImageElement ? y.value.naturalHeight : y.value.clientHeight) || 1, w = window.innerWidth, C = window.innerHeight, I = Math.min(
        (w - 2 * l) / d,
        (C - 2 * l) / g
      ), pe = (w - d * I) / 2, ge = (C - g * I) / 2;
      return {
        centerX: pe,
        centerY: ge,
        originalWidth: d,
        originalHeight: g,
        scale: I
      };
    }
    function V({ trigger: l }) {
      const { originalWidth: d, originalHeight: g, scale: w, centerX: C, centerY: I } = oe(), { top: pe, left: ge, width: Ze, height: Qe } = T.value, Ce = [
        {
          top: `${pe}px`,
          left: `${ge}px`,
          width: `${Ze}px`,
          height: `${Qe}px`,
          opacity: 0
        },
        {
          top: `${I}px`,
          left: `${C}px`,
          width: `${d * w}px`,
          height: `${g * w}px`,
          opacity: 1
        }
      ];
      return {
        frames: l === "open" ? Ce : Ce.reverse()
      };
    }
    function B() {
      const { centerX: l, centerY: d, originalHeight: g, originalWidth: w, scale: C } = oe(), I = {
        top: `${d}px`,
        left: `${l}px`,
        width: `${w * C}px`,
        height: `${g * C}px`
      };
      if (!S.value)
        throw console.error("[maz-ui](vFullscreenImg) ImgElement is not defined");
      Object.assign(S.value.style, I);
    }
    function ve() {
      const { frames: l } = V({
        trigger: "open"
      }), d = X(l);
      if (!d)
        throw console.error("[maz-ui](vFullscreenImg) open animation is not defined");
      d.onfinish = () => {
        B();
      };
    }
    function u() {
      const { frames: l } = V({
        trigger: "close"
      }), d = X(l);
      if (!d)
        throw console.error("[maz-ui](vFullscreenImg) close animation is not defined");
      d.onfinish = () => {
        var g, w;
        n("close"), (g = te.value) == null || g.remove(), (w = t.destroy) == null || w.call(t);
      };
    }
    function p() {
      b.value || B();
    }
    return de(() => {
      m.value = !0, document.addEventListener("keydown", $), window.addEventListener("resize", p), F(), L.value = se().length > 1;
    }), Me(() => {
      document.removeEventListener("keydown", $), window.removeEventListener("resize", p), he();
    }), (l, d) => (z(), W("div", {
      ref_key: "FullscreenImgElement",
      ref: te,
      role: "button",
      class: "m-fullscreen-img",
      tabindex: "0",
      onClick: H(P, ["stop"]),
      onKeypress: st(H(P, ["prevent"]), ["esc"])
    }, [
      Pe(U(
        O(i),
        { class: "m-fullscreen-img-loader maz-h-16" },
        null,
        512
        /* NEED_PATCH */
      ), [
        [$e, m.value]
      ]),
      v.value && L.value ? (z(), W("button", {
        key: 0,
        type: "button",
        class: "m-fullscreen-btn --next",
        onClick: d[0] || (d[0] = H((g) => K("next"), ["stop"]))
      }, [
        U(O(r), { class: "maz-rotate-180 maz-text-3xl" })
      ])) : x("v-if", !0),
      v.value && L.value ? (z(), W("button", {
        key: 1,
        type: "button",
        class: "m-fullscreen-btn --previous",
        onClick: d[1] || (d[1] = H((g) => K("previous"), ["stop"]))
      }, [
        U(O(r), { class: "maz-text-3xl" })
      ])) : x("v-if", !0),
      D("button", {
        type: "button",
        class: "m-fullscreen-btn --close",
        onClick: P
      }, [
        U(O(o), { class: "maz-text-3xl" })
      ]),
      D("div", At, [
        x(" eslint-disable-next-line vuejs-accessibility/no-static-element-interactions, vuejs-accessibility/click-events-have-key-events "),
        D("img", {
          ref_key: "ImgElement",
          ref: S,
          src: _.value,
          alt: M.value ?? void 0,
          tabindex: "0",
          class: Ee([R.value]),
          onLoad: Y,
          onClick: d[2] || (d[2] = H((g) => l.zoom ? fe() : void 0, ["stop"]))
        }, null, 42, Ot)
      ])
    ], 40, xt));
  }
}), je = (s, e) => {
  const t = s.__vccOpts || s;
  for (const [n, i] of e)
    t[n] = i;
  return t;
}, Dt = /* @__PURE__ */ je(Ht, [["__scopeId", "data-v-884a2c57"]]), Oe = "m-fullscreen-is-open";
class Mt {
  constructor() {
    a(this, "options");
    a(this, "defaultOptions", {
      scaleOnHover: !1,
      blurOnHover: !1,
      disabled: !1,
      zoom: !0,
      offset: 80,
      animation: {
        duration: 300,
        easing: "ease-in-out"
      }
    });
    a(this, "mouseEnterListener");
    a(this, "mouseLeaveListener");
    a(this, "renderPreviewListener");
  }
  buildOptions(e, t) {
    const n = typeof t.value == "object" ? t.value : { src: t.value, alt: void 0 }, i = (n == null ? void 0 : n.src) ?? this.getImgSrc(e), o = (n == null ? void 0 : n.alt) ?? this.getImgAlt(e);
    return {
      ...this.defaultOptions,
      ...n,
      src: i,
      alt: o
    };
  }
  get allInstances() {
    return [...document.querySelectorAll(".m-fullscreen-img-instance")];
  }
  getImgSrc(e) {
    var n;
    const t = ((n = this.options) == null ? void 0 : n.src) || e.getAttribute("src") || e.getAttribute("data-src");
    if (!t)
      throw new Error(
        '[maz-ui](fullscreen-img) src of image must be provided by `v-fullscreen=""`, `v-fullscreen="{ src: "" }"`, `src=""` or `data-src=""` atributes'
      );
    return t;
  }
  getImgAlt(e) {
    var t;
    return ((t = this.options) == null ? void 0 : t.alt) || e.getAttribute("alt") || e.getAttribute("data-alt");
  }
  create(e, t) {
    if (this.options = this.buildOptions(e, t), this.options.disabled)
      return;
    e.style.cursor = "move", (this.options.scaleOnHover || this.options.blurOnHover) && (e.style.transition = "all 200ms ease-in-out"), e.classList.add("m-fullscreen-img-instance"), e.setAttribute("data-src", this.getImgSrc(e));
    const n = this.getImgAlt(e);
    n && e.setAttribute("data-alt", n), this.mouseEnterListener = () => this.mouseEnter(e), this.mouseLeaveListener = () => this.mouseLeave(e), this.renderPreviewListener = () => this.renderPreview(e), e.addEventListener("mouseenter", this.mouseEnterListener), e.addEventListener("mouseleave", this.mouseLeaveListener), e.addEventListener("click", this.renderPreviewListener);
  }
  update(e, t) {
    this.options = this.buildOptions(e, t);
  }
  remove(e) {
    e.removeEventListener("mouseenter", this.mouseEnterListener), e.removeEventListener("mouseleave", this.mouseLeaveListener), e.removeEventListener("click", this.renderPreviewListener), e.classList.remove("m-fullscreen-img-instance"), e.style.cursor = "";
  }
  renderPreview(e) {
    return e.classList.add(Oe), _e(Dt, {
      props: {
        ...this.options,
        openInstanceClass: Oe,
        clickedElement: e,
        clickedElementBounds: e.getBoundingClientRect()
      },
      addDestroyInProps: !0
    });
  }
  mouseLeave(e) {
    this.options.scaleOnHover && (e.style.transform = ""), this.options.blurOnHover && (e.style.filter = ""), e.style.zIndex = "";
  }
  mouseEnter(e) {
    e.style.zIndex = "1", this.options.scaleOnHover && (e.style.transform = "scale(1.04)"), this.options.blurOnHover && (e.style.filter = "blur(3px)");
  }
}
let ue;
const Pt = {
  mounted(s, e) {
    return ue = new Mt(), ue.create(s, e);
  },
  updated(s, e) {
    return ue.update(s, e);
  },
  unmounted(s) {
    return ue.remove(s);
  }
}, $t = {
  install(s) {
    s.directive("fullscreen-img", Pt);
  }
}, _t = [
  zt,
  kt,
  Ct,
  gt,
  $t,
  Et
], Cs = {
  install(s) {
    for (const e of _t)
      e.install(s, {});
  }
};
function Ft(s) {
  return s ? (s = s.toString(), s.charAt(0).toUpperCase() + s.slice(1)) : "";
}
const Nt = {
  style: "currency",
  minimumFractionDigits: 2,
  round: !1
};
function Vt(s, e, t) {
  let n = Number(s);
  return t.round && (n = Math.floor(n), t.minimumFractionDigits = 0), new Intl.NumberFormat(e, t).format(n);
}
function Bt(s, e, t) {
  if (s === void 0)
    throw new TypeError("[maz-ui](FilterCurrency) The `number` attribute is required.");
  if (e === void 0)
    throw new TypeError("[maz-ui](FilterCurrency) The `locale` attribute is required.");
  if (typeof e != "string")
    throw new TypeError("[maz-ui](FilterCurrency) The `locale` attribute must be a string.");
  if (t.currency === void 0)
    throw new TypeError("[maz-ui](FilterCurrency) The `options.currency` attribute is required.");
}
function qt(s, e, t) {
  const n = {
    ...Nt,
    ...t
  };
  Bt(s, e, n);
  try {
    return Vt(s, e, n);
  } catch (i) {
    throw new Error(`[maz-ui](FilterCurrency) ${i}`);
  }
}
const Ut = {
  month: "short",
  day: "numeric",
  year: "numeric"
};
function Wt(s, e, t) {
  if (e === void 0)
    throw new TypeError("[maz-ui](FilterDate) The `locale` attribute is required.");
  if (typeof e != "string")
    throw new TypeError("[maz-ui](FilterDate) The `locale` attribute must be a string.");
  const n = t ?? Ut;
  try {
    const i = s instanceof Date ? s : new Date(s);
    return new Intl.DateTimeFormat(e, n).format(i);
  } catch (i) {
    throw new Error(`[maz-ui](FilterDate) ${i}`);
  }
}
const jt = {
  minimumFractionDigits: 2
};
function Rt(s, e, t) {
  const n = {
    ...jt,
    ...t
  };
  if (s === void 0)
    throw new TypeError("[maz-ui](FilterNumber) The `number` attribute is required.");
  if (e === void 0)
    throw new TypeError("[maz-ui](FilterNumber) The `locale` attribute is required.");
  if (typeof e != "string")
    throw new TypeError("[maz-ui](FilterNumber) The `locale` attribute must be a string.");
  try {
    return new Intl.NumberFormat(e, n).format(Number(s));
  } catch (i) {
    throw new Error(`[maz-ui](FilterNumber) ${i}`);
  }
}
const Yt = {
  capitalize: Ft,
  currency: qt,
  date: Wt,
  number: Rt
}, xs = {
  install(s) {
    s.provide("filters", Yt);
  }
};
function Kt({ timeout: s = 1e3, callback: e, remainingTimeUpdate: t = 200, callbackOffsetTime: n = 0 }) {
  const i = h(s), o = h(s);
  let r;
  function c(y) {
    typeof y == "number" && (o.value = y), typeof y == "number" && (i.value = y), m();
  }
  function m() {
    r || (r = setInterval(() => {
      o.value -= t, o.value <= 0 && (b(), setTimeout(() => e == null ? void 0 : e(), n));
    }, t));
  }
  function v() {
    r && (clearInterval(r), r = void 0);
  }
  function L() {
    !r && o.value > 0 && m();
  }
  function b() {
    setTimeout(() => o.value = i.value, n * 2), v();
  }
  return {
    remainingTime: o,
    start: c,
    pause: v,
    resume: L,
    stop: b
  };
}
const Xt = { class: "m-toast__message-wrapper" }, Zt = { class: "m-toast__message" }, Qt = { class: "maz-flex maz-items-center maz-gap-2" }, Gt = { key: 0 }, Jt = {
  key: 4,
  class: "progress-bar maz-absolute maz-inset-x-0 maz-bottom-0 maz-h-1"
}, es = /* @__PURE__ */ De({
  __name: "MazToast",
  props: {
    message: { default: void 0 },
    position: { default: "bottom-right" },
    maxToasts: { type: [Number, Boolean], default: !1 },
    timeout: { type: [Number, Boolean], default: 1e4 },
    queue: { type: Boolean },
    noPauseOnHover: { type: Boolean },
    type: { default: "info" },
    link: { default: void 0 },
    action: { default: void 0 },
    persistent: { type: Boolean },
    icon: { type: Boolean, default: !0 }
  },
  emits: ["close", "click", "open"],
  setup(s, { expose: e, emit: t }) {
    const n = s, i = t, o = k(() => import("./MazBtn-1g7UvzQ_.mjs")), r = k(() => import("./x-mark-BP8zWe_C.mjs")), c = k(
      () => import("./arrow-top-right-on-square-CkcBq4qD.mjs")
    ), m = k(
      () => import("./exclamation-triangle-BfZLmRbP.mjs")
    ), v = k(
      () => import("./exclamation-circle-DzBJ8vGa.mjs")
    ), L = k(
      () => import("./information-circle-C0pE_7kr.mjs")
    ), b = k(() => import("./check-circle-CtCMBYyp.mjs")), y = k(() => import("./link-D7FkvsK4.mjs")), T = h(), ee = f(() => {
      if (n.icon)
        switch (n.type) {
          case "danger":
            return m;
          case "info":
            return L;
          case "success":
            return b;
          case "warning":
            return v;
          default:
            return;
        }
    }), _ = f(() => n.position.includes("top") ? "top" : "bottom"), M = f(() => n.position.includes("left") ? "left" : n.position.includes("right") ? "right" : "center"), te = f(() => M.value !== "center" ? M.value === "right" ? "m-slide-right" : "m-slide-left" : _.value === "top" ? "m-slide-top" : "m-slide-bottom"), S = h(!1), R = h(!1), Y = h(), P = `m-toast-container --${_.value} --${M.value}`, $ = `.${P.replaceAll(" ", ".")}`, F = Kt({
      callback: V,
      timeout: typeof n.timeout == "number" ? n.timeout : 0,
      callbackOffsetTime: 200
    });
    function he() {
      const u = document.querySelector($);
      if (!u && !u) {
        const p = document.body, l = document.createElement("div");
        l.className = P, p.append(l);
      }
    }
    function se() {
      const u = document.querySelector($);
      return !n.queue && n.maxToasts === !1 ? !1 : typeof n.maxToasts == "number" && u ? n.maxToasts <= u.childElementCount : u && u.childElementCount > 0;
    }
    function ne() {
      if (se()) {
        Y.value = setTimeout(ne, 250);
        return;
      }
      const u = document.querySelector($);
      T.value && u && u.prepend(T.value), R.value = !0, typeof n.timeout == "number" && n.timeout > 0 && F.start();
    }
    const ie = h("100%");
    function K() {
      switch (n.type) {
        case "danger":
          return "maz-bg-danger-700";
        case "info":
          return "maz-bg-info-700";
        case "success":
          return "maz-bg-success-700";
        case "warning":
          return "maz-bg-warning-700";
        default:
          return "maz-bg-secondary";
      }
    }
    ke(
      F.remainingTime,
      (u) => {
        if (typeof n.timeout == "number") {
          const p = 100 * u / n.timeout;
          ie.value = `${p}%`;
        }
      }
    );
    function N(u) {
      i("click", u), n.persistent || V();
    }
    async function fe(u, p) {
      var l;
      S.value = !0, await u(), S.value = !1, (l = n.action) != null && l.closeToast && N(p);
    }
    function X(u) {
      n.noPauseOnHover || (u ? F.pause() : F.resume());
    }
    function oe() {
      stop(), Y.value && clearTimeout(Y.value);
    }
    function V() {
      oe(), R.value = !1;
    }
    e({ closeToast: V });
    function B() {
      i("open");
    }
    function ve() {
      var p;
      i("close"), (p = T.value) == null || p.remove();
      const u = document.querySelector($);
      u && !(u != null && u.hasChildNodes()) && u.remove();
    }
    return de(() => {
      he(), ne();
    }), (u, p) => (z(), q(nt, {
      name: te.value,
      onAfterLeave: ve,
      onAfterEnter: B,
      persisted: ""
    }, {
      default: ye(() => [
        Pe(D(
          "button",
          {
            ref_key: "Toaster",
            ref: T,
            class: Ee(["m-toast", [
              `--${u.type}`,
              `--${_.value}`,
              `--${M.value}`,
              { "maz-pb-1": typeof u.timeout == "number" && u.timeout > 0, "--persistent": u.persistent }
            ]]),
            role: "alert",
            onMouseover: p[2] || (p[2] = (l) => X(!0)),
            onMouseleave: p[3] || (p[3] = (l) => X(!1)),
            onClick: p[4] || (p[4] = H((l) => {
              var d;
              return u.link && !((d = u.link) != null && d.closeToast) ? void 0 : N(l);
            }, ["stop"]))
          },
          [
            ee.value ? (z(), q(it(ee.value), {
              key: 0,
              class: "maz-text-2xl"
            })) : x("v-if", !0),
            D("div", Xt, [
              D(
                "p",
                Zt,
                be(u.message),
                1
                /* TEXT */
              )
            ]),
            u.action ? (z(), q(O(o), {
              key: 1,
              "data-test": "action-btn",
              color: u.type,
              pastel: "",
              loading: S.value,
              size: "sm",
              onClick: p[0] || (p[0] = H((l) => u.action ? fe(u.action.func, l) : void 0, ["stop"]))
            }, {
              default: ye(() => [
                ot(
                  be(u.action.text),
                  1
                  /* TEXT */
                )
              ]),
              _: 1
              /* STABLE */
            }, 8, ["color", "loading"])) : x("v-if", !0),
            u.link ? (z(), q(O(o), {
              key: 2,
              "data-test": "link-btn",
              color: u.type,
              pastel: "",
              size: "xs",
              href: u.link.href,
              target: u.link.target ?? "_self"
            }, {
              default: ye(() => {
                var l;
                return [
                  D("div", Qt, [
                    u.link.text ? (z(), W(
                      "span",
                      Gt,
                      be(u.link.text),
                      1
                      /* TEXT */
                    )) : x("v-if", !0),
                    ((l = u.link) == null ? void 0 : l.target) === "_blank" ? (z(), q(O(c), {
                      key: 1,
                      class: "maz-text-xl"
                    })) : (z(), q(O(y), {
                      key: 2,
                      class: "maz-text-xl"
                    }))
                  ])
                ];
              }),
              _: 1
              /* STABLE */
            }, 8, ["color", "href", "target"])) : x("v-if", !0),
            u.persistent ? x("v-if", !0) : (z(), W("button", {
              key: 3,
              class: "--close",
              onClick: p[1] || (p[1] = H((l) => N(l), ["stop"]))
            }, [
              U(O(r), { class: "--icon maz-text-xl" })
            ])),
            typeof u.timeout == "number" && u.timeout > 0 ? (z(), W("div", Jt, [
              D(
                "div",
                {
                  style: rt({
                    width: ie.value
                  }),
                  class: Ee(["maz-h-full !maz-transition-all !maz-duration-200 !maz-ease-linear", K()])
                },
                null,
                6
                /* CLASS, STYLE */
              )
            ])) : x("v-if", !0)
          ],
          34
          /* CLASS, NEED_HYDRATION */
        ), [
          [$e, R.value]
        ])
      ]),
      _: 1
      /* STABLE */
    }, 8, ["name"]));
  }
}), ts = /* @__PURE__ */ je(es, [["__scopeId", "data-v-cdb80f1f"]]), ss = {
  position: "bottom-right",
  timeout: 1e4,
  persistent: !1,
  icon: !0
};
class ns {
  constructor(e, t) {
    this.app = e, this.globalOptions = t;
  }
  show(e, t) {
    const n = {
      ...ss,
      ...this.globalOptions,
      ...t,
      message: e
    }, { destroy: i, vNode: o } = _e(ts, {
      props: n,
      app: this.app
    });
    return {
      destroy: i,
      close: () => {
        var r, c;
        return (c = (r = o.component) == null ? void 0 : r.exposed) == null ? void 0 : c.closeToast();
      }
    };
  }
  getLocalOptions(e, t) {
    return {
      type: e,
      ...t
    };
  }
  message(e, t) {
    return this.show(e, this.getLocalOptions("theme", t));
  }
  success(e, t) {
    return this.show(e, this.getLocalOptions("success", t));
  }
  error(e, t) {
    return this.show(e, this.getLocalOptions("danger", t));
  }
  info(e, t) {
    return this.show(e, this.getLocalOptions("info", t));
  }
  warning(e, t) {
    return this.show(e, this.getLocalOptions("warning", t));
  }
}
function is(s, e) {
  return new ns(s, e);
}
const As = {
  install(s, e) {
    s.provide("toast", is(s, e));
  }
}, j = "";
function os(s) {
  return s.filter((e, t, n) => t === n.indexOf(e));
}
function rs(s) {
  return (e = j) => typeof e == "function" ? s.findIndex((...t) => e(...t)) > -1 : s.includes(e);
}
const as = (s) => s.length > 0;
function ls(s) {
  return (e = j) => os([...s, e]);
}
function cs(s) {
  return (e = j) => s.filter((t) => t !== e);
}
class us {
  constructor() {
    a(this, "_loaders", h([]));
  }
  get loaders() {
    return f(() => this._loaders.value);
  }
  stop(e = j) {
    this._loaders.value = cs(this._loaders.value)(e);
  }
  start(e = j) {
    return this._loaders.value = ls(this._loaders.value)(e), () => this.stop(e);
  }
  get anyLoading() {
    return f(() => as(this._loaders.value));
  }
  isLoading(e = j) {
    return f(() => rs(this._loaders.value)(e)).value;
  }
}
const ms = new us(), Os = {
  install: (s) => {
    s.provide("wait", ms);
  }
}, we = {
  delay: 100,
  observer: {
    root: void 0,
    rootMargin: "0px",
    threshold: 0.2
  },
  animation: {
    once: !0,
    duration: 300,
    delay: 0
  }
};
class ds {
  constructor(e) {
    a(this, "options");
    this.options = {
      delay: (e == null ? void 0 : e.delay) ?? we.delay,
      observer: {
        ...we.observer,
        ...e == null ? void 0 : e.observer
      },
      animation: {
        ...we.animation,
        ...e == null ? void 0 : e.animation
      }
    };
  }
  handleIntersect(e, t) {
    for (const n of e) {
      const o = n.target.getAttribute("data-maz-aos-children") === "true", r = n.target.getAttribute("data-maz-aos") ? [n.target] : [];
      if (o) {
        const c = [...document.querySelectorAll("[data-maz-aos-anchor]")].map(
          (m) => m.getAttribute("data-maz-aos-anchor") === `#${n.target.id}` ? m : void 0
        );
        for (const m of c)
          m && r.push(m);
      }
      for (const c of r) {
        const m = c.getAttribute("data-maz-aos-once"), v = typeof m == "string" ? m === "true" : this.options.animation.once;
        if (typeof this.options.observer.threshold == "number" && n.intersectionRatio > this.options.observer.threshold) {
          const L = c.getAttribute("data-maz-aos-duration"), b = c.getAttribute("data-maz-aos-delay");
          if (L || (c.style.transitionDuration = `${this.options.animation.duration}ms`, setTimeout(() => {
            c.style.transitionDuration = "0";
          }, 1e3)), b || (c.style.transitionDelay = `${this.options.animation.delay}ms`, setTimeout(() => {
            c.style.transitionDelay = "0";
          }, 1e3)), c.classList.add("maz-aos-animate"), v) {
            const y = c.getAttribute("data-maz-aos-anchor");
            if (y) {
              const T = document.querySelector(y);
              T && t.unobserve(T);
            }
            t.unobserve(c);
          }
        } else
          c.classList.remove("maz-aos-animate");
      }
    }
  }
  async handleObserver() {
    await dt(this.options.delay);
    const e = new IntersectionObserver(this.handleIntersect.bind(this), {
      ...this.options.observer
    });
    for (const t of document.querySelectorAll("[data-maz-aos]")) {
      const n = t.getAttribute("data-maz-aos-anchor");
      if (n) {
        const i = document.querySelector(n);
        i ? (i.setAttribute("data-maz-aos-children", "true"), e.observe(i)) : console.warn(`[maz-ui](aos) no element found with selector "${n}"`);
      } else
        e.observe(t);
    }
  }
  runAnimations() {
    if (E())
      return this.handleObserver();
    console.warn("[MazAos](runAnimations) should be executed on client side");
  }
}
let Z;
const Hs = {
  install: (s, e) => {
    Z = new ds(e), s.provide("aos", Z), E() && (e != null && e.router ? e.router.afterEach(async () => {
      Z.runAnimations();
    }) : Z.runAnimations());
  }
};
function Ds() {
  return Z;
}
const hs = {
  darkClass: "dark",
  lightClass: "light",
  storageThemeKey: "theme",
  storageThemeValueDark: "dark",
  storageThemeValueLight: "light",
  storageThemeValueSystem: "system",
  watchChanges: !0
}, Q = h("system"), A = h("system");
function He() {
  return window.matchMedia("(prefers-color-scheme: dark)").matches;
}
function G({
  darkClass: s,
  lightClass: e,
  storageThemeKey: t,
  storageThemeValueDark: n,
  setLocalStorageValue: i = !0,
  setSelectedTheme: o = !0
}) {
  E() && (document.documentElement.classList.remove(e), document.documentElement.classList.add(s), Q.value = n, o && (A.value = n), i && (localStorage[t] = n));
}
function J({
  darkClass: s,
  lightClass: e,
  storageThemeKey: t,
  storageThemeValueLight: n,
  setLocalStorageValue: i = !0,
  setSelectedTheme: o = !0
}) {
  E() && (document.documentElement.classList.remove(s), document.documentElement.classList.add(e), Q.value = n, o && (A.value = n), i && (localStorage[t] = n));
}
function Re({
  setLocalStorageValue: s = !0,
  ...e
}) {
  if (E())
    return document.documentElement.classList.remove(e.darkClass), document.documentElement.classList.remove(e.lightClass), Q.value = e.storageThemeValueSystem, A.value = e.storageThemeValueSystem, s && (localStorage[e.storageThemeKey] = e.storageThemeValueSystem), Ie(e);
}
function Ye(s) {
  if (!s.defaultTheme)
    return console.error("[maz-ui](useThemeHandler) No default theme set");
  if (!["light", "dark"].includes(s.defaultTheme))
    return console.error('[maz-ui](useThemeHandler) Default theme must be "light" or "dark"');
  switch (s.defaultTheme) {
    case "dark":
      return G({
        ...s,
        setLocalStorageValue: !1,
        setSelectedTheme: !0
      });
    case "light":
      return J({
        ...s,
        setLocalStorageValue: !1,
        setSelectedTheme: !0
      });
  }
}
function Ie(s) {
  return E() ? !localStorage[s.storageThemeKey] && s.defaultTheme ? Ye(s) : localStorage[s.storageThemeKey] === s.storageThemeValueDark || !localStorage[s.storageThemeKey] && He() || localStorage[s.storageThemeKey] === s.storageThemeValueSystem && He() ? G({
    ...s,
    setLocalStorageValue: !1,
    setSelectedTheme: !1
  }) : J({
    ...s,
    setLocalStorageValue: !1,
    setSelectedTheme: !1
  }) : void 0;
}
function ze({
  theme: s,
  ...e
}) {
  return s === "system" ? Re(e) : s === "dark" ? G(e) : J(e);
}
function fs(s) {
  return Q.value === s.storageThemeValueDark ? J(s) : G(s);
}
function Ms(s) {
  const e = {
    ...hs,
    ...s
  };
  function t() {
    Ie(e);
  }
  return de(() => {
    localStorage[e.storageThemeKey] && (A.value = localStorage[e.storageThemeKey]), e.watchChanges && window.matchMedia("(prefers-color-scheme: dark)").addEventListener("change", t);
  }), Me(() => {
    e.watchChanges && window.matchMedia("(prefers-color-scheme: dark)").removeEventListener("change", t);
  }), ke(A, (n) => {
    switch (localStorage[e.storageThemeKey] = n, n) {
      case e.storageThemeValueDark: {
        G(e);
        break;
      }
      case e.storageThemeValueLight: {
        J(e);
        break;
      }
      case e.storageThemeValueSystem: {
        Re(e);
        break;
      }
    }
  }), {
    autoSetTheme: () => Ie(e),
    toggleTheme: () => fs(e),
    setSystemTheme: () => ze({ ...e, theme: "system" }),
    setDarkTheme: () => ze({ ...e, theme: "dark" }),
    setLightTheme: () => ze({ ...e, theme: "light" }),
    setDefaultTheme: () => Ye(e),
    hasDarkTheme: f(() => A.value === e.storageThemeValueDark),
    hasLightTheme: f(() => A.value === e.storageThemeValueLight),
    hasSystemTheme: f(() => A.value === e.storageThemeValueSystem),
    theme: Q,
    selectedTheme: A
  };
}
function Ps({
  componentName: s,
  providedId: e
}) {
  const t = at();
  return f(() => e ?? `${s}-${t == null ? void 0 : t.uid}`);
}
function $s() {
  return Se("aos");
}
function _s() {
  return Se("wait");
}
function Fs() {
  return Se("toast");
}
function Ns({
  callback: s,
  options: e
}) {
  return new ut(s, e);
}
function Vs({
  callback: s,
  options: e
}) {
  return new mt(s, e);
}
function vs(s) {
  const e = {};
  for (const [t, n] of Object.entries(s))
    e[t] = Number.parseInt(n, 10);
  return e;
}
function ps(s = {}) {
  const {
    internalWindow: e = E() ? window : void 0,
    initialWidth: t = Number.POSITIVE_INFINITY,
    initialHeight: n = Number.POSITIVE_INFINITY,
    includeScrollbar: i = !0
  } = s, o = h(t), r = h(n);
  function c() {
    e && (i ? (o.value = e.innerWidth, r.value = e.innerHeight) : (o.value = e.document.documentElement.clientWidth, r.value = e.document.documentElement.clientHeight));
  }
  return c(), de(() => {
    e && window.addEventListener("resize", c, { passive: !0 });
  }), lt(() => {
    e && window.removeEventListener("resize", c);
  }), { width: o, height: r };
}
function Bs({
  initialWidth: s = 0,
  initialHeight: e,
  includeScrollbar: t,
  internalWindow: n,
  listenOrientation: i,
  breakpoints: o,
  mediumBreakPoint: r = "md",
  largeBreakPoint: c = "lg"
}) {
  const { width: m } = ps({
    initialWidth: s,
    // (SSR) mobile first
    initialHeight: e,
    includeScrollbar: t,
    internalWindow: n,
    listenOrientation: i
  }), v = vs(o), L = f(() => m.value >= v[c]), b = f(
    () => m.value >= v[r] && m.value < v[c]
  ), y = f(() => m.value >= 0 && m.value < v[r]);
  return {
    width: m,
    numericBreakpoints: v,
    isSmallScreen: y,
    isLargeScreen: L,
    isMediumScreen: b,
    breakpoints: o
  };
}
function qs(s) {
  const e = h(), t = h(), n = h(), i = h(), o = h(), r = h(), c = new ht({
    ...s,
    element: s.element,
    onValuesChanged(m) {
      e.value = m.xDiff, t.value = m.yDiff, n.value = m.xStart, i.value = m.xEnd, o.value = m.yStart, r.value = m.yEnd;
    }
  });
  return {
    xDiff: e,
    yDiff: t,
    xStart: n,
    xEnd: i,
    yStart: o,
    yEnd: r,
    start: c.start,
    stop: c.stop
  };
}
function Ke(s) {
  const e = s.join(", ").match(/\b\w+\b/g);
  return e ? e.length : 0;
}
function gs(s, e = 150) {
  const t = Ke([s]);
  return Math.ceil(t / e);
}
function Us(s) {
  const e = f(
    () => {
      var r;
      return typeof s.velocity == "number" ? s.velocity : ((r = s.velocity) == null ? void 0 : r.value) ?? 150;
    }
  ), t = f(() => {
    var r;
    return typeof s.contentSelector == "string" ? s.contentSelector : (r = s.contentSelector) == null ? void 0 : r.value;
  }), n = f(() => {
    var r, c, m;
    if (typeof ((r = s.contentRef) == null ? void 0 : r.value) == "object")
      return (c = s.contentRef.value) == null ? void 0 : c.textContent;
    if (t.value && typeof document < "u") {
      const v = document.querySelector(t.value);
      if (v)
        return v.textContent;
    }
    return typeof s.content == "string" ? s.content : (m = s.content) == null ? void 0 : m.value;
  }), i = f(() => gs(n.value, e.value)), o = f(() => Ke([n.value]));
  return {
    content: n,
    wordCount: o,
    velocity: e,
    duration: i
  };
}
function ys(s, e) {
  const t = s.length, n = e.length, i = [];
  for (let o = 0; o <= t; o++)
    i[o] = [o];
  for (let o = 0; o <= n; o++)
    i[0][o] = o;
  for (let o = 1; o <= t; o++)
    for (let r = 1; r <= n; r++) {
      const c = s[o - 1] === e[r - 1] ? 0 : 1;
      i[o][r] = Math.min(
        i[o - 1][r] + 1,
        i[o][r - 1] + 1,
        i[o - 1][r - 1] + c
      );
    }
  return i[t][n];
}
function Xe(s, e) {
  const t = ys(s, e), n = Math.max(s.length, e.length);
  return 1 - t / n;
}
function bs(s, e, t = 0.75) {
  return Xe(s, e) >= t;
}
function Ws(s, e, t = 0.75) {
  const n = f(
    () => xe(typeof s == "string" ? s : s.value)
  ), i = f(
    () => xe(typeof e == "string" ? e : e.value)
  ), o = f(() => typeof t == "number" ? t : t.value), r = f(() => Xe(n.value, i.value));
  return {
    isMatching: f(
      () => bs(n.value, i.value, o.value)
    ),
    score: r
  };
}
export {
  Wt as A,
  Rt as B,
  As as C,
  Os as D,
  ms as E,
  ds as F,
  Hs as G,
  Ds as H,
  ut as I,
  Ms as J,
  Ps as K,
  $s as L,
  _s as M,
  Fs as N,
  Ns as O,
  Vs as P,
  Kt as Q,
  Bs as R,
  Ls as S,
  ns as T,
  mt as U,
  ps as V,
  us as W,
  qs as X,
  Us as Y,
  Ws as Z,
  je as _,
  E as a,
  ht as b,
  Ts as c,
  Es as d,
  Is as e,
  pt as f,
  gt as g,
  Ct as h,
  Se as i,
  Ss as j,
  kt as k,
  $t as l,
  _e as m,
  xe as n,
  Pt as o,
  Cs as p,
  wt as q,
  zt as r,
  dt as s,
  ct as t,
  Et as u,
  It as v,
  ks as w,
  xs as x,
  Ft as y,
  qt as z
};
