// maz-ui/components/chunks/MazBtn-Dfz-5FeI.mjs
import { defineComponent as F, defineAsyncComponent as z, useAttrs as O, useSlots as j, onBeforeMount as E, computed as l, openBlock as n, createBlock as a, resolveDynamicComponent as c, normalizeClass as r, withCtx as R, createElementBlock as f, createCommentVNode as t, renderSlot as p, unref as v } from "vue";
import '../assets/MazBtn.css';const w = /* @__PURE__ */ F({
  __name: "MazBtn",
  props: {
    variant: { default: "button" },
    size: { default: "md" },
    color: { default: "primary" },
    type: { default: "button" },
    rounded: { type: Boolean },
    roundedSize: { default: "lg" },
    noRounded: { type: Boolean },
    outline: { type: Boolean },
    pastel: { type: Boolean },
    block: { type: Boolean },
    noUnderline: { type: Boolean },
    loading: { type: Boolean },
    disabled: { type: Boolean },
    fab: { type: Boolean },
    icon: { type: [String, Function, Object], default: void 0 },
    leftIcon: { type: [String, Function, Object], default: void 0 },
    rightIcon: { type: [String, Function, Object], default: void 0 },
    noPadding: { type: Boolean },
    noElevation: { type: Boolean },
    contentClass: { default: void 0 }
  },
  setup(u) {
    const o = u, d = z(() => import("./MazSpinner-rAqBu92m.mjs")), i = z(() => import("./MazIcon-BDGxCy9M.mjs")), { href: b, to: B } = O(), y = j();
    E(() => {
      o.icon && !o.fab && console.error('[maz-ui](MazBtn) the prop "icon" must be used only with "fab" props');
    });
    const m = l(() => b ? "a" : B ? "router-link" : "button"), I = l(
      () => o.pastel ? `--${o.color}-pastel` : o.outline ? `--${o.color}-outline` : `--${o.color}`
    ), h = l(
      () => (o.loading || o.disabled) && m.value === "button"
    ), C = l(() => h.value ? "--cursor-default" : "--cursor-pointer"), $ = l(() => `--is-${o.variant}`), s = l(() => o.loading && o.variant === "button"), g = l(() => !!y["left-icon"] || o.leftIcon), k = l(() => !!y["right-icon"] || o.rightIcon), S = l(() => g.value || k.value), M = l(() => o.fab && (o.icon || !!y.icon)), T = l(() => m.value === "button" ? o.type : void 0);
    return (e, D) => (n(), a(c(m.value), {
      disabled: h.value || void 0,
      class: r(["m-btn", [
        `--${e.size}`,
        !e.fab && e.roundedSize && `--rounded-${e.roundedSize}`,
        I.value,
        C.value,
        $.value,
        {
          "--block": e.block,
          "--no-underline": e.noUnderline,
          "--fab": e.fab,
          "--loading": e.loading,
          "--disabled": h.value,
          "--icon": S.value,
          "--rounded": e.rounded,
          "--no-rounded": e.noRounded,
          "--no-padding": e.noPadding,
          "--no-elevation": e.noElevation
        }
      ]]),
      type: T.value
    }, {
      default: R(() => [
        g.value ? (n(), f(
          "div",
          {
            key: 0,
            class: r(["m-btn__icon-left maz-flex maz-flex-center", { "maz-invisible": s.value }])
          },
          [
            t(`
        @slot left-icon - The icon to display on the left of the button
      `),
            p(e.$slots, "left-icon", {}, () => [
              typeof e.leftIcon == "string" ? (n(), a(v(i), {
                key: 0,
                name: e.leftIcon
              }, null, 8, ["name"])) : e.leftIcon ? (n(), a(c(e.leftIcon), { key: 1 })) : t("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : t("v-if", !0),
        M.value ? (n(), f(
          "div",
          {
            key: 1,
            class: r(["m-btn__icon", { "maz-invisible": s.value }])
          },
          [
            t(`
        @slot icon - The icon to display on the fab button
      `),
            p(e.$slots, "icon", {}, () => [
              typeof e.icon == "string" ? (n(), a(v(i), {
                key: 0,
                name: e.icon
              }, null, 8, ["name"])) : e.icon ? (n(), a(c(e.icon), { key: 1 })) : t("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : t("v-if", !0),
        e.$slots.default ? (n(), f(
          "span",
          {
            key: 2,
            class: r([{ "maz-invisible": s.value }, e.contentClass])
          },
          [
            t(`
        @slot default - The content of the button
      `),
            p(e.$slots, "default", {}, void 0, !0)
          ],
          2
          /* CLASS */
        )) : t("v-if", !0),
        k.value ? (n(), f(
          "div",
          {
            key: 3,
            class: r(["m-btn__icon-right", { "maz-invisible": s.value }])
          },
          [
            t(`
        @slot right-icon - The icon to display on the right of the button
      `),
            p(e.$slots, "right-icon", {}, () => [
              typeof e.rightIcon == "string" ? (n(), a(v(i), {
                key: 0,
                name: e.rightIcon
              }, null, 8, ["name"])) : e.rightIcon ? (n(), a(c(e.rightIcon), { key: 1 })) : t("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : t("v-if", !0),
        s.value ? (n(), a(v(d), {
          key: 4,
          class: "m-btn-loader",
          size: "2em",
          color: e.color
        }, null, 8, ["color"])) : t("v-if", !0)
      ]),
      _: 3
      /* FORWARDED */
    }, 8, ["disabled", "class", "type"]));
  }
}), A = (u, o) => {
  const d = u.__vccOpts || u;
  for (const [i, b] of o)
    d[i] = b;
  return d;
}, P = /* @__PURE__ */ A(w, [["__scopeId", "data-v-48f14785"]]);
export {
  P as M,
  A as _
};
