// maz-ui/components/chunks/MazCarousel-DJ_nQfy4.mjs
import { defineComponent as g, defineAsyncComponent as p, useSlots as N, ref as v, openBlock as s, createElementBlock as a, normalizeClass as r, renderSlot as n, createCommentVNode as c, createVNode as i, unref as u, withCtx as B, createElementVNode as k, pushScopeId as W, popScopeId as x } from "vue";
import '../assets/MazCarousel.css';const $ = (t) => (W("data-v-c70bd7f6"), t = t(), x(), t), T = { key: 0 }, V = {
  key: 1,
  class: "m-carousel__header__actions"
}, w = /* @__PURE__ */ $(() => /* @__PURE__ */ k(
  "div",
  { class: "m-carousel__items__spacer" },
  null,
  -1
  /* HOISTED */
)), E = /* @__PURE__ */ g({
  __name: "MazCarousel",
  props: {
    noScrollBtn: { type: Boolean },
    ariaLabelPreviousButton: { default: "Scroll to previous items" },
    ariaLabelNextButton: { default: "Scroll to next items" },
    hideScrollbar: { type: Boolean }
  },
  setup(t) {
    const d = t, l = p(() => import("./MazBtn-BIVlttu0.mjs")), f = p(() => import("./chevron-left-CcZaP3Rc.mjs")), m = p(() => import("./chevron-right-BKWCeY9O.mjs")), h = N(), S = v(!1), b = v(!1), _ = v();
    function y() {
      return !d.noScrollBtn || h.title;
    }
    function C() {
      return !!h.title;
    }
    function I() {
      const e = _.value;
      e == null || e.scrollTo(e.scrollLeft + e.clientWidth, 0);
    }
    function L() {
      const e = _.value;
      e == null || e.scrollTo(e.scrollLeft - e.clientWidth, 0);
    }
    function z(e) {
      const o = e.target;
      if (!o)
        return;
      S.value = o.scrollLeft >= 20;
      const M = o.scrollWidth - o.clientWidth;
      b.value = o.scrollLeft >= M - 20;
    }
    return (e, o) => (s(), a(
      "div",
      {
        class: r(["m-carousel", {
          "--hide-scrollbar": e.hideScrollbar
        }])
      },
      [
        y() ? (s(), a(
          "div",
          {
            key: 0,
            class: r(["m-carousel__header", { "--has-title": C() }])
          },
          [
            C() ? (s(), a("div", T, [
              n(e.$slots, "title", {}, void 0, !0)
            ])) : c("v-if", !0),
            e.noScrollBtn ? c("v-if", !0) : (s(), a("div", V, [
              i(u(l), {
                color: "transparent",
                class: r(["m-carousel__btn", { "--muted": !S.value }]),
                "no-elevation": "",
                fab: "",
                "aria-label": e.ariaLabelPreviousButton,
                onClick: L
              }, {
                default: B(() => [
                  n(e.$slots, "previous-icon", {}, () => [
                    i(u(f), { class: "maz-text-lg" })
                  ], !0)
                ]),
                _: 3
                /* FORWARDED */
              }, 8, ["class", "aria-label"]),
              i(u(l), {
                color: "transparent",
                class: r(["m-carousel__btn", { "--muted": b.value }]),
                fab: "",
                "no-elevation": "",
                "aria-label": e.ariaLabelNextButton,
                onClick: I
              }, {
                default: B(() => [
                  n(e.$slots, "next-icon", {}, () => [
                    i(u(m), { class: "maz-text-lg" })
                  ], !0)
                ]),
                _: 3
                /* FORWARDED */
              }, 8, ["class", "aria-label"])
            ]))
          ],
          2
          /* CLASS */
        )) : c("v-if", !0),
        k(
          "div",
          {
            ref_key: "MazCarouselItems",
            ref: _,
            class: "m-carousel__items",
            onScroll: z
          },
          [
            c(" Insert your items "),
            n(e.$slots, "default", {}, void 0, !0),
            w
          ],
          544
          /* NEED_HYDRATION, NEED_PATCH */
        )
      ],
      2
      /* CLASS */
    ));
  }
}), P = (t, d) => {
  const l = t.__vccOpts || t;
  for (const [f, m] of d)
    l[f] = m;
  return l;
}, H = /* @__PURE__ */ P(E, [["__scopeId", "data-v-c70bd7f6"]]);
export {
  H as M,
  P as _
};
