// maz-ui/components/chunks/MazDialogPromise-Bgxjg9ss.mjs
import { ref as S, defineComponent as j, onMounted as W, watch as K, openBlock as p, createBlock as h, Teleport as I, createVNode as x, Transition as U, withCtx as B, createElementBlock as T, normalizeClass as M, createElementVNode as E, withModifiers as q, mergeProps as F, renderSlot as g, createCommentVNode as d, defineAsyncComponent as P, useAttrs as X, computed as b, createTextVNode as D, toDisplayString as z, unref as c, Fragment as L, renderList as G } from "vue";
import '../assets/MazDialogPromise.css';const A = {
  cancelText: "Cancel",
  confirmText: "Confirm",
  cancelButton: {
    text: "Cancel",
    color: "danger",
    outline: !0
  },
  confirmButton: {
    text: "Confirm",
    color: "success"
  }
}, J = S(A), w = S([]);
function Q(i, t) {
  return new Promise((u, n) => {
    w.value = [
      ...w.value,
      {
        id: i,
        isActive: !0,
        resolve: async (f) => {
          await (t == null ? void 0 : t()), u(f);
        },
        reject: n
      }
    ];
  });
}
function H(i) {
  return w.value = w.value.filter(({ id: t }) => t !== i), w.value;
}
function N(i, t, u = !1) {
  var n;
  t && ((n = t[i]) == null || n.call(t, u), t.isActive = !1, setTimeout(() => {
    H(t.id);
  }, 500));
}
function R() {
  return {
    data: J,
    dialogState: w,
    showDialogAndWaitChoice: Q,
    removeDialogFromState: H,
    rejectDialog: (i, t = new Error("cancel")) => N("reject", i, t),
    resolveDialog: (i, t = "accept") => N("resolve", i, t)
  };
}
const O = "--backdrop-present", Y = /* @__PURE__ */ j({
  inheritAttrs: !1,
  __name: "MazBackdrop",
  props: {
    modelValue: { type: Boolean, default: !1 },
    teleportSelector: { default: "body" },
    beforeClose: { type: Function, default: void 0 },
    persistent: { type: Boolean, default: !1 },
    noCloseOnEscKey: { type: Boolean, default: !1 },
    transitionName: { default: "backdrop-anim" },
    backdropClass: { default: void 0 },
    backdropContentClass: { default: void 0 }
  },
  emits: ["open", "close", "update:model-value", "before-close"],
  setup(i, { expose: t, emit: u }) {
    const n = i, f = u;
    function k() {
      document.documentElement.classList.add(O);
    }
    async function o() {
      document.querySelector(".m-backdrop.--present") || document.documentElement.classList.remove(O);
    }
    const s = S(n.modelValue);
    function a() {
      $(!1);
    }
    async function $(l) {
      var V;
      l || (f("before-close"), await ((V = n.beforeClose) == null ? void 0 : V.call(n))), s.value = l;
    }
    function C() {
      f("open");
    }
    function m() {
      f("update:model-value", !1), f("close"), e();
    }
    function v() {
      n.persistent || a();
    }
    function r(l) {
      !n.noCloseOnEscKey && l.key === "Escape" && !n.persistent && a();
    }
    function y() {
      k(), document.addEventListener("keyup", r, !1);
    }
    function e() {
      document.removeEventListener("keyup", r), o();
    }
    return W(() => {
      n.modelValue ? y() : e();
    }), K(
      () => n.modelValue,
      (l) => {
        s.value = l, l ? y() : e();
      }
    ), t({
      onBackdropAnimationEnter: C,
      onBackdropAnimationLeave: m,
      onBackdropClicked: v,
      close: a,
      present: s,
      toggleModal: $,
      onKeyPress: r
    }), (l, V) => (p(), h(I, { to: l.teleportSelector }, [
      x(U, {
        appear: "",
        name: l.transitionName,
        onAfterEnter: C,
        onAfterLeave: m
      }, {
        default: B(() => [
          s.value ? (p(), T(
            "div",
            {
              key: 0,
              class: M(["m-backdrop --present", [l.backdropClass]]),
              tabindex: "-1",
              role: "dialog"
            },
            [
              E(
                "button",
                {
                  class: M(["m-backdrop-overlay", { "--disabled": l.persistent }]),
                  tabindex: "-1",
                  onClick: q(v, ["self"])
                },
                null,
                2
                /* CLASS */
              ),
              E(
                "div",
                F({
                  class: ["m-backdrop-content", l.backdropContentClass]
                }, l.$attrs, {
                  role: "document",
                  tabindex: "0"
                }),
                [
                  g(l.$slots, "default", { close: a })
                ],
                16
                /* FULL_PROPS */
              )
            ],
            2
            /* CLASS */
          )) : d("v-if", !0)
        ]),
        _: 3
        /* FORWARDED */
      }, 8, ["name"])
    ], 8, ["to"]));
  }
}), Z = {
  key: 0,
  id: "dialogTitle",
  class: "maz-my-0 maz-text-xl maz-font-semibold"
}, _ = {
  id: "dialogDesc",
  class: "m-dialog-content"
}, ee = {
  key: 0,
  class: "m-dialog-footer"
}, te = /* @__PURE__ */ j({
  __name: "MazDialog",
  props: {
    modelValue: { type: Boolean },
    title: { default: void 0 },
    noClose: { type: Boolean, default: !1 },
    width: { default: "500px" },
    maxWidth: { default: "95vw" },
    maxHeight: { default: "95vh" },
    scrollable: { type: Boolean, default: !1 },
    persistent: { type: Boolean, default: !1 },
    teleportSelector: {},
    beforeClose: {},
    noCloseOnEscKey: { type: Boolean },
    transitionName: {},
    backdropClass: {},
    backdropContentClass: {}
  },
  emits: ["open", "close", "update:model-value"],
  setup(i) {
    const t = P(() => import("./MazBtn-C7RDTnn-.mjs")), u = P(() => import("./x-mark-BP8zWe_C.mjs")), n = X(), f = b(() => ({
      ...n,
      class: void 0,
      style: void 0
    })), k = b(() => ({
      class: n.class,
      style: n.style
    }));
    return (o, s) => (p(), h(Y, F(f.value, {
      persistent: o.persistent,
      "model-value": o.modelValue,
      "transition-name": "modal-anim",
      "aria-labelledby": "dialogTitle",
      "aria-describedby": "dialogDesc",
      onClose: s[0] || (s[0] = (a) => o.$emit("close", a)),
      onOpen: s[1] || (s[1] = (a) => o.$emit("open", a)),
      "onUpdate:modelValue": s[2] || (s[2] = (a) => o.$emit("update:model-value", a))
    }), {
      default: B(({ close: a }) => [
        E(
          "div",
          F({
            class: ["m-dialog", { "--scrollable": o.scrollable }],
            role: "dialog",
            "aria-modal": "true",
            style: [{ width: o.width, maxWidth: o.maxWidth, maxHeight: o.maxHeight }]
          }, k.value),
          [
            d(`
          @slot Header slot
            @binding {Function} close close function
        `),
            g(o.$slots, "header", { close: a }, () => [
              E(
                "div",
                {
                  class: M(["m-dialog-header", { "--has-title": o.$slots.title || o.title }])
                },
                [
                  o.$slots.title || o.title ? (p(), T("h2", Z, [
                    d(`
                @slot Title slot in the header
              `),
                    g(o.$slots, "title", {}, () => [
                      D(
                        z(o.title),
                        1
                        /* TEXT */
                      )
                    ], !0)
                  ])) : d("v-if", !0),
                  !o.noClose && !o.persistent ? (p(), h(c(t), {
                    key: 1,
                    class: "m-dialog-closebtn",
                    color: "transparent",
                    onClick: a
                  }, {
                    default: B(() => [
                      x(c(u), { class: "maz-text-lg" })
                    ]),
                    _: 2
                    /* DYNAMIC */
                  }, 1032, ["onClick"])) : d("v-if", !0)
                ],
                2
                /* CLASS */
              )
            ], !0),
            E("div", _, [
              d(`
            @slot Default content
              @binding {Function} close close function
          `),
              g(o.$slots, "default", { close: a }, void 0, !0)
            ]),
            o.$slots.footer ? (p(), T("div", ee, [
              d(`
            @slot Footer slot
              @binding {Function} close close function
          `),
              g(o.$slots, "footer", { close: a }, void 0, !0)
            ])) : d("v-if", !0)
          ],
          16
          /* FULL_PROPS */
        )
      ]),
      _: 3
      /* FORWARDED */
    }, 16, ["persistent", "model-value"]));
  }
}), oe = (i, t) => {
  const u = i.__vccOpts || i;
  for (const [n, f] of t)
    u[n] = f;
  return u;
}, ne = /* @__PURE__ */ oe(te, [["__scopeId", "data-v-20ca3bcd"]]), le = { class: "maz-flex maz-items-center maz-gap-2" }, se = /* @__PURE__ */ j({
  __name: "MazDialogPromise",
  props: {
    data: { default: void 0 },
    identifier: {},
    buttons: { default: void 0 },
    modelValue: { type: Boolean },
    title: {},
    noClose: { type: Boolean },
    width: {},
    maxWidth: {},
    maxHeight: {},
    scrollable: { type: Boolean },
    persistent: { type: Boolean }
  },
  setup(i) {
    const t = i, u = P(() => import("./MazBtn-C7RDTnn-.mjs")), n = b(() => t.buttons ?? s.value.buttons), { dialogState: f, rejectDialog: k, resolveDialog: o, data: s } = R(), a = b(() => {
      var r, y, e, l;
      return ((r = s.value) == null ? void 0 : r.cancelButton) ?? ((y = t.data) == null ? void 0 : y.cancelButton) ?? A.cancelButton ? {
        ...A.cancelButton,
        ...(e = s.value) == null ? void 0 : e.cancelButton,
        ...(l = t.data) == null ? void 0 : l.cancelButton
      } : !1;
    }), $ = b(() => {
      var r, y, e, l;
      return ((r = s.value) == null ? void 0 : r.confirmButton) ?? ((y = t.data) == null ? void 0 : y.confirmButton) ?? A.confirmButton ? {
        ...A.confirmButton,
        ...(e = s.value) == null ? void 0 : e.confirmButton,
        ...(l = t.data) == null ? void 0 : l.confirmButton
      } : !1;
    }), C = b(() => ({
      ...A,
      ...s.value,
      ...t.data
    })), m = b(
      () => f.value.find(({ id: v }) => v === t.identifier)
    );
    return (v, r) => {
      var y;
      return p(), h(ne, {
        "model-value": ((y = m.value) == null ? void 0 : y.isActive) ?? !1,
        "onUpdate:modelValue": r[2] || (r[2] = (e) => c(k)(m.value))
      }, {
        title: B(() => [
          d(`
        @slot title slot - Place your title
      `),
          g(v.$slots, "title", {}, () => {
            var e;
            return [
              D(
                z((e = C.value) == null ? void 0 : e.title),
                1
                /* TEXT */
              )
            ];
          })
        ]),
        default: B(() => [
          d(`
        @slot Default slot - Place your content
          @binding {Function} resolve resolve function
          @binding {Function} reject reject function
      `),
          g(v.$slots, "default", {
            resolve: (e) => c(o)(m.value, e),
            reject: (e) => c(k)(m.value, e)
          }, () => {
            var e;
            return [
              D(
                z((e = C.value) == null ? void 0 : e.message),
                1
                /* TEXT */
              )
            ];
          })
        ]),
        footer: B(() => [
          d(`
        @slot Footer slot
          @binding {Function} resolve resolve function
          @binding {Function} reject reject function
      `),
          g(v.$slots, "footer-button", {
            resolve: (e) => c(o)(m.value, e),
            reject: (e) => c(k)(m.value, e)
          }, () => [
            E("div", le, [
              n.value && n.value.length > 0 ? (p(!0), T(
                L,
                { key: 0 },
                G(n.value, (e, l) => (p(), h(c(u), {
                  key: l,
                  color: e.color,
                  size: e.size,
                  outline: e.outline,
                  rounded: e.rounded,
                  disabled: e.disabled,
                  block: e.block,
                  loading: e.loading,
                  onClick: (V) => e.type === "resolve" ? c(o)(m.value, e.response) : c(k)(m.value, e.response)
                }, {
                  default: B(() => [
                    D(
                      z(e.text),
                      1
                      /* TEXT */
                    )
                  ]),
                  _: 2
                  /* DYNAMIC */
                }, 1032, ["color", "size", "outline", "rounded", "disabled", "block", "loading", "onClick"]))),
                128
                /* KEYED_FRAGMENT */
              )) : (p(), T(
                L,
                { key: 1 },
                [
                  a.value ? (p(), h(
                    c(u),
                    F({ key: 0 }, a.value, {
                      outline: "",
                      onClick: r[0] || (r[0] = (e) => c(k)(m.value))
                    }),
                    {
                      default: B(() => [
                        d(`
                @slot cancel-text slot - Place your cancel text
              `),
                        g(v.$slots, "cancel-text", {}, () => {
                          var e;
                          return [
                            D(
                              z(a.value.text ?? ((e = C.value) == null ? void 0 : e.cancelText)),
                              1
                              /* TEXT */
                            )
                          ];
                        })
                      ]),
                      _: 3
                      /* FORWARDED */
                    },
                    16
                    /* FULL_PROPS */
                  )) : d("v-if", !0),
                  $.value ? (p(), h(
                    c(u),
                    F({ key: 1 }, $.value, {
                      onClick: r[1] || (r[1] = (e) => c(o)(m.value))
                    }),
                    {
                      default: B(() => [
                        d(`
                @slot confirm-text slot - Place your confirm text
              `),
                        g(v.$slots, "confirm-text", {}, () => [
                          D(
                            z($.value.text ?? C.value.confirmText),
                            1
                            /* TEXT */
                          )
                        ])
                      ]),
                      _: 3
                      /* FORWARDED */
                    },
                    16
                    /* FULL_PROPS */
                  )) : d("v-if", !0)
                ],
                64
                /* STABLE_FRAGMENT */
              ))
            ])
          ])
        ]),
        _: 3
        /* FORWARDED */
      }, 8, ["model-value"]);
    };
  }
});
export {
  oe as _,
  se as a,
  R as u
};
