// maz-ui/components/chunks/MazGallery-Din0Z37d.mjs
import '../assets/MazGallery.css';var ye = Object.defineProperty;
var be = (o, e, t) => e in o ? ye(o, e, { enumerable: !0, configurable: !0, writable: !0, value: t }) : o[e] = t;
var f = (o, e, t) => be(o, typeof e != "symbol" ? e + "" : e, t);
import { defineComponent as oe, defineAsyncComponent as _, ref as y, computed as E, watch as Ae, onMounted as Ie, onBeforeUnmount as Ee, openBlock as b, createElementBlock as A, withModifiers as S, withKeys as Le, withDirectives as R, createVNode as x, unref as L, vShow as we, createCommentVNode as O, createElementVNode as P, normalizeClass as F, render as B, useCssVars as Ce, onBeforeMount as ze, normalizeStyle as ee, Fragment as te, renderList as se, toDisplayString as xe } from "vue";
import { _ as ae } from "./MazCard-BBwohtCJ.mjs";
const Oe = ["onKeypress"], Pe = { class: "m-fullscreen-img-scroller" }, Se = ["src", "alt"], ke = /* @__PURE__ */ oe({
  __name: "MazFullscreenImg",
  props: {
    src: {},
    clickedElementBounds: { default: void 0 },
    offset: { default: void 0 },
    animation: { default: () => ({
      duration: 300,
      easing: "ease-in-out"
    }) },
    openInstanceClass: { default: "m-fullscreen-img-instance" },
    clickedElement: {},
    destroy: { type: Function, default: void 0 },
    alt: { default: void 0 },
    zoom: { type: Boolean, default: !0 }
  },
  emits: ["close", "previous", "next", "before-close"],
  setup(o, { emit: e }) {
    const t = o, s = e, r = _(() => import("./MazSpinner-BVd7-OTu.mjs")), h = _(() => import("./x-mark-BP8zWe_C.mjs")), p = _(() => import("./chevron-left-CcZaP3Rc.mjs")), g = y(!1), C = y(!1), w = y(!1), u = y(!1), m = y(!1), l = y(t.clickedElement), c = E(() => t.clickedElement.getBoundingClientRect()), re = E(
      () => c.value.height < c.value.width
    ), W = y(t.src), Z = y(t.alt), j = y(), z = y(), ie = E(() => m.value ? "--is-zoomed maz-cursor-zoom-out" : "maz-cursor-zoom-in");
    function le() {
      g.value = !0, C.value = !1, w.value = !0;
    }
    Ae(
      () => w.value,
      (n) => {
        n && fe();
      }
    );
    function k() {
      s("before-close"), ge();
    }
    function q(n) {
      n.key === "Escape" && (n.preventDefault(), k()), (n.key === "ArrowLeft" || n.key === "ArrowRight") && (n.preventDefault(), N(n.key === "ArrowRight" ? "next" : "previous"));
    }
    function ce() {
      document.documentElement.classList.add("--m-fullscreen-open");
    }
    function ue() {
      document.documentElement.classList.remove("--m-fullscreen-open");
    }
    function V() {
      return [...document.querySelectorAll(".m-fullscreen-img-instance")];
    }
    function me(n, a) {
      return a < 0 ? n.length - 1 : a >= n.length ? 0 : a;
    }
    async function de(n, a) {
      n.classList.remove(t.openInstanceClass), a.classList.add(t.openInstanceClass);
      const i = a.getAttribute("data-src"), d = a.getAttribute("data-alt");
      Z.value = d, i && (W.value = i), g.value = !1, g.value || (C.value = !0);
    }
    function N(n) {
      const a = document.querySelector(
        `.m-fullscreen-img-instance.${t.openInstanceClass}`
      );
      if (a) {
        const i = V(), d = i.indexOf(a), I = n === "next" ? d + 1 : d - 1, v = i[me(i, I)];
        l.value = v, v && de(a, v), s(n), m.value ? Y(z.value) : H();
      }
    }
    function Y(n) {
      re.value ? (n.style.width = `${window.innerWidth}px`, n.style.removeProperty("height")) : (n.style.height = `${window.innerHeight}px`, n.style.removeProperty("width")), n.style.removeProperty("top"), n.style.removeProperty("left");
    }
    async function he() {
      const n = z.value;
      m.value ? (m.value = !m.value, H()) : (m.value = !m.value, Y(n));
    }
    function G(n) {
      var a;
      return (a = z.value) == null ? void 0 : a.animate(n, {
        duration: t.animation.duration,
        // Durée de l'animation en millisecondes
        easing: t.animation.easing
        // Fonction d'interpolation pour l'animation inverse
      });
    }
    function X(n = t.offset ?? 0) {
      const a = (l.value instanceof HTMLImageElement ? l.value.naturalWidth : l.value.clientWidth) || 1, i = (l.value instanceof HTMLImageElement ? l.value.naturalHeight : l.value.clientHeight) || 1, d = window.innerWidth, I = window.innerHeight, v = Math.min(
        (d - 2 * n) / a,
        (I - 2 * n) / i
      ), T = (d - a * v) / 2, D = (I - i * v) / 2;
      return {
        centerX: T,
        centerY: D,
        originalWidth: a,
        originalHeight: i,
        scale: v
      };
    }
    function K({ trigger: n }) {
      const { originalWidth: a, originalHeight: i, scale: d, centerX: I, centerY: v } = X(), { top: T, left: D, width: ve, height: pe } = c.value, J = [
        {
          top: `${T}px`,
          left: `${D}px`,
          width: `${ve}px`,
          height: `${pe}px`,
          opacity: 0
        },
        {
          top: `${v}px`,
          left: `${I}px`,
          width: `${a * d}px`,
          height: `${i * d}px`,
          opacity: 1
        }
      ];
      return {
        frames: n === "open" ? J : J.reverse()
      };
    }
    function H() {
      const { centerX: n, centerY: a, originalHeight: i, originalWidth: d, scale: I } = X(), v = {
        top: `${a}px`,
        left: `${n}px`,
        width: `${d * I}px`,
        height: `${i * I}px`
      };
      if (!z.value)
        throw console.error("[maz-ui](vFullscreenImg) ImgElement is not defined");
      Object.assign(z.value.style, v);
    }
    function fe() {
      const { frames: n } = K({
        trigger: "open"
      }), a = G(n);
      if (!a)
        throw console.error("[maz-ui](vFullscreenImg) open animation is not defined");
      a.onfinish = () => {
        H();
      };
    }
    function ge() {
      const { frames: n } = K({
        trigger: "close"
      }), a = G(n);
      if (!a)
        throw console.error("[maz-ui](vFullscreenImg) close animation is not defined");
      a.onfinish = () => {
        var i, d;
        s("close"), (i = j.value) == null || i.remove(), (d = t.destroy) == null || d.call(t);
      };
    }
    function Q() {
      m.value || H();
    }
    return Ie(() => {
      C.value = !0, document.addEventListener("keydown", q), window.addEventListener("resize", Q), ce(), u.value = V().length > 1;
    }), Ee(() => {
      document.removeEventListener("keydown", q), window.removeEventListener("resize", Q), ue();
    }), (n, a) => (b(), A("div", {
      ref_key: "FullscreenImgElement",
      ref: j,
      role: "button",
      class: "m-fullscreen-img",
      tabindex: "0",
      onClick: S(k, ["stop"]),
      onKeypress: Le(S(k, ["prevent"]), ["esc"])
    }, [
      R(x(
        L(r),
        { class: "m-fullscreen-img-loader maz-h-16" },
        null,
        512
        /* NEED_PATCH */
      ), [
        [we, C.value]
      ]),
      w.value && u.value ? (b(), A("button", {
        key: 0,
        type: "button",
        class: "m-fullscreen-btn --next",
        onClick: a[0] || (a[0] = S((i) => N("next"), ["stop"]))
      }, [
        x(L(p), { class: "maz-rotate-180 maz-text-3xl" })
      ])) : O("v-if", !0),
      w.value && u.value ? (b(), A("button", {
        key: 1,
        type: "button",
        class: "m-fullscreen-btn --previous",
        onClick: a[1] || (a[1] = S((i) => N("previous"), ["stop"]))
      }, [
        x(L(p), { class: "maz-text-3xl" })
      ])) : O("v-if", !0),
      P("button", {
        type: "button",
        class: "m-fullscreen-btn --close",
        onClick: k
      }, [
        x(L(h), { class: "maz-text-3xl" })
      ]),
      P("div", Pe, [
        O(" eslint-disable-next-line vuejs-accessibility/no-static-element-interactions, vuejs-accessibility/click-events-have-key-events "),
        P("img", {
          ref_key: "ImgElement",
          ref: z,
          src: W.value,
          alt: Z.value ?? void 0,
          tabindex: "0",
          class: F([ie.value]),
          onLoad: le,
          onClick: a[2] || (a[2] = S((i) => n.zoom ? he() : void 0, ["stop"]))
        }, null, 42, Se)
      ])
    ], 40, Oe));
  }
}), He = /* @__PURE__ */ ae(ke, [["__scopeId", "data-v-884a2c57"]]);
function Be(o, e) {
  let t = e == null ? void 0 : e.element;
  function s() {
    t && B(null, t);
  }
  const r = x(
    o,
    { ...e == null ? void 0 : e.props, destroy: s },
    e == null ? void 0 : e.children
  );
  return e != null && e.app ? (r.appContext = e.app._context, t ? B(r, t) : typeof document < "u" && (t = document.createElement("div"), B(r, t))) : (t = t ?? document.body, B(r, t)), { vNode: r, destroy: s, el: t };
}
const ne = "m-fullscreen-is-open";
class $e {
  constructor() {
    f(this, "options");
    f(this, "defaultOptions", {
      scaleOnHover: !1,
      blurOnHover: !1,
      disabled: !1,
      zoom: !0,
      offset: 80,
      animation: {
        duration: 300,
        easing: "ease-in-out"
      }
    });
    f(this, "mouseEnterListener");
    f(this, "mouseLeaveListener");
    f(this, "renderPreviewListener");
  }
  buildOptions(e, t) {
    const s = typeof t.value == "object" ? t.value : { src: t.value, alt: void 0 }, r = (s == null ? void 0 : s.src) ?? this.getImgSrc(e), h = (s == null ? void 0 : s.alt) ?? this.getImgAlt(e);
    return {
      ...this.defaultOptions,
      ...s,
      src: r,
      alt: h
    };
  }
  get allInstances() {
    return [...document.querySelectorAll(".m-fullscreen-img-instance")];
  }
  getImgSrc(e) {
    var s;
    const t = ((s = this.options) == null ? void 0 : s.src) || e.getAttribute("src") || e.getAttribute("data-src");
    if (!t)
      throw new Error(
        '[maz-ui](fullscreen-img) src of image must be provided by `v-fullscreen=""`, `v-fullscreen="{ src: "" }"`, `src=""` or `data-src=""` atributes'
      );
    return t;
  }
  getImgAlt(e) {
    var t;
    return ((t = this.options) == null ? void 0 : t.alt) || e.getAttribute("alt") || e.getAttribute("data-alt");
  }
  create(e, t) {
    if (this.options = this.buildOptions(e, t), this.options.disabled)
      return;
    e.style.cursor = "move", (this.options.scaleOnHover || this.options.blurOnHover) && (e.style.transition = "all 200ms ease-in-out"), e.classList.add("m-fullscreen-img-instance"), e.setAttribute("data-src", this.getImgSrc(e));
    const s = this.getImgAlt(e);
    s && e.setAttribute("data-alt", s), this.mouseEnterListener = () => this.mouseEnter(e), this.mouseLeaveListener = () => this.mouseLeave(e), this.renderPreviewListener = () => this.renderPreview(e), e.addEventListener("mouseenter", this.mouseEnterListener), e.addEventListener("mouseleave", this.mouseLeaveListener), e.addEventListener("click", this.renderPreviewListener);
  }
  update(e, t) {
    this.options = this.buildOptions(e, t);
  }
  remove(e) {
    e.removeEventListener("mouseenter", this.mouseEnterListener), e.removeEventListener("mouseleave", this.mouseLeaveListener), e.removeEventListener("click", this.renderPreviewListener), e.classList.remove("m-fullscreen-img-instance"), e.style.cursor = "";
  }
  renderPreview(e) {
    return e.classList.add(ne), Be(He, {
      props: {
        ...this.options,
        openInstanceClass: ne,
        clickedElement: e,
        clickedElementBounds: e.getBoundingClientRect()
      },
      addDestroyInProps: !0
    });
  }
  mouseLeave(e) {
    this.options.scaleOnHover && (e.style.transform = ""), this.options.blurOnHover && (e.style.filter = ""), e.style.zIndex = "";
  }
  mouseEnter(e) {
    e.style.zIndex = "1", this.options.scaleOnHover && (e.style.transform = "scale(1.04)"), this.options.blurOnHover && (e.style.filter = "blur(3px)");
  }
}
let $;
const U = {
  mounted(o, e) {
    return $ = new $e(), $.create(o, e);
  },
  updated(o, e) {
    return $.update(o, e);
  },
  unmounted(o) {
    return $.remove(o);
  }
}, Me = "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7", _e = {
  baseClass: "m-lazy-img",
  loadedClass: "m-lazy-loaded",
  loadingClass: "m-lazy-loading",
  errorClass: "m-lazy-error",
  noPhotoClass: "m-lazy-no-photo",
  noPhoto: !1,
  observerOnce: !0,
  loadOnce: !1,
  noUseErrorPhoto: !1,
  observerOptions: {
    threshold: 0.1
  }
};
class Re {
  constructor(e = {}) {
    f(this, "observers", []);
    f(this, "defaultOptions", _e);
    f(this, "options");
    f(this, "onImgLoadedCallback");
    f(this, "onImgErrorCallback");
    f(this, "hasImgLoaded", !1);
    this.options = this.buildOptions(e), this.onImgLoadedCallback = this.imageIsLoaded.bind(this), this.onImgErrorCallback = this.imageHasError.bind(this);
  }
  async loadErrorPhoto() {
    const { default: e } = await import("./no-photography-BJX8HSus.mjs");
    return e;
  }
  buildOptions(e) {
    return {
      ...this.defaultOptions,
      ...e,
      observerOptions: {
        ...this.defaultOptions.observerOptions,
        ...e.observerOptions
      }
    };
  }
  removeClass(e, t) {
    e.classList.remove(t);
  }
  addClass(e, t) {
    e.classList.add(t);
  }
  removeAllStateClasses(e) {
    this.removeClass(e, this.options.loadedClass), this.removeClass(e, this.options.loadingClass), this.removeClass(e, this.options.errorClass), this.removeClass(e, this.options.noPhotoClass);
  }
  setBaseClass(e) {
    this.addClass(e, this.options.baseClass);
  }
  imageIsLoading(e) {
    var t, s;
    this.addClass(e, this.options.loadingClass), (s = (t = this.options).onLoading) == null || s.call(t, e);
  }
  imageHasNoPhoto(e) {
    this.removeClass(e, this.options.loadingClass), this.addClass(e, this.options.noPhotoClass), this.setDefaultPhoto(e);
  }
  imageIsLoaded(e) {
    var t, s;
    this.hasImgLoaded = !0, this.removeClass(e, this.options.loadingClass), this.addClass(e, this.options.loadedClass), (s = (t = this.options).onLoaded) == null || s.call(t, e);
  }
  imageHasError(e) {
    var t, s;
    this.removeClass(e, this.options.loadingClass), this.addClass(e, this.options.errorClass), (s = (t = this.options).onError) == null || s.call(t, e), this.setDefaultPhoto(e);
  }
  getSrc(e) {
    return typeof e.value == "object" ? e.value.src : e.value;
  }
  getImageUrl(e, t) {
    const s = this.getImgElement(e).getAttribute("data-lazy-src");
    return s || this.getSrc(t);
  }
  async setPictureSourceUrls(e) {
    const t = e.querySelectorAll("source");
    if (t.length > 0)
      for await (const s of t) {
        const r = s.getAttribute("data-lazy-srcset");
        if (r)
          s.srcset = r;
        else
          return this.imageHasError(e);
      }
    else
      this.imageHasError(e);
  }
  hasBgImgMode(e) {
    return e.arg === "bg-image";
  }
  isPictureElement(e) {
    return e instanceof HTMLPictureElement;
  }
  getImgElement(e) {
    return this.isPictureElement(e) ? e.querySelector("img") : e;
  }
  async setDefaultPhoto(e) {
    if (this.options.noUseErrorPhoto)
      return;
    const t = this.options.fallbackSrc ?? this.options.errorPhoto;
    typeof t == "string" && this.addClass(e, this.options.noPhotoClass);
    const s = t ?? await this.loadErrorPhoto(), r = e.querySelectorAll("source");
    if (r.length > 0)
      for await (const h of r)
        h.srcset = s;
    else
      this.setImgSrc(e, s);
  }
  addEventListenerToImg(e) {
    const t = this.getImgElement(e);
    t.addEventListener("load", () => this.onImgLoadedCallback(e), {
      once: !0
    }), t.addEventListener("error", (s) => this.onImgErrorCallback(e, s), { once: !0 });
  }
  async loadImage(e, t) {
    if (this.imageIsLoading(e), this.isPictureElement(e))
      this.addEventListenerToImg(e), await this.setPictureSourceUrls(e);
    else {
      const s = this.getImageUrl(e, t);
      if (!s)
        return this.imageHasError(e);
      this.hasBgImgMode(t) ? (e.style.backgroundImage = `url('${s}')`, this.imageIsLoaded(e)) : (this.addEventListenerToImg(e), this.setImgSrc(e, s));
    }
  }
  setImgSrc(e, t) {
    const s = this.getImgElement(e);
    s.src = t;
  }
  handleIntersectionObserver(e, t, s, r) {
    var h, p;
    this.observers.push(r);
    for (const g of s)
      if (g.isIntersecting) {
        if ((p = (h = this.options).onIntersecting) == null || p.call(h, g.target), this.options.observerOnce && r.unobserve(e), this.options.loadOnce && this.hasImgLoaded)
          return;
        this.loadImage(e, t);
      }
  }
  createObserver(e, t) {
    const s = (p, g) => {
      this.handleIntersectionObserver(e, t, p, g);
    }, r = this.options.observerOptions;
    new IntersectionObserver(s, r).observe(e);
  }
  async imageHandler(e, t, s) {
    if (s === "update")
      for await (const r of this.observers) r.unobserve(e);
    window.IntersectionObserver ? this.createObserver(e, t) : this.loadImage(e, t);
  }
  async bindUpdateHandler(e, t, s) {
    if (this.options.noPhoto)
      return this.imageHasNoPhoto(e);
    await this.imageHandler(e, t, s);
  }
  async add(e, t) {
    if (this.hasBgImgMode(t) && this.isPictureElement(e))
      throw new Error(`[MazLazyImg] You can't use the "bg-image" mode with "<picture />" element`);
    setTimeout(() => this.setBaseClass(e), 0), e.getAttribute("src") || this.setImgSrc(e, Me), await this.bindUpdateHandler(e, t, "bind");
  }
  async update(e, t) {
    t.value !== t.oldValue && (this.hasImgLoaded = !1, this.removeAllStateClasses(e), await this.bindUpdateHandler(e, t, "update"));
  }
  remove(e, t) {
    this.hasImgLoaded = !1, this.hasBgImgMode(t) && (e.style.backgroundImage = ""), this.removeAllStateClasses(e);
    for (const s of this.observers)
      s.unobserve(e);
    this.observers = [];
  }
}
let M;
const Fe = {
  created(o, e) {
    const t = typeof e.value == "object" ? e.value : {};
    M = new Re(t), M.add(o, e);
  },
  updated(o, e) {
    M.update(o, e);
  },
  unmounted(o, e) {
    M.remove(o, e);
  }
}, Ne = { class: "m-gallery__wrapper maz-flex maz-flex-1" }, Te = ["alt"], De = {
  key: 0,
  class: "m-gallery__remaining-layer maz-flex maz-bg-overlay maz-flex-center"
}, Ue = { class: "maz-text-2xl maz-text-white" }, We = /* @__PURE__ */ oe({
  __name: "MazGallery",
  props: {
    /**
     *  Array of string or object: `['https://via.placeholder.com/500', 'https://via.placeholder.com/600']` or `[{ slug: 'https://via.placeholder.com/500', alt: 'image descripton' }, { slug: 'https://via.placeholder.com/600', alt: 'image descripton' }]`
     */
    images: {
      type: Array,
      default: () => []
    },
    /** Images count shown (max: 5) */
    imagesShownCount: { type: Number, default: 5 },
    /** Remove transparent layer with the remain count (ex: +2) */
    noRemaining: { type: Boolean, default: !1 },
    /** Height of gallery */
    height: { type: [Number, String], default: 150 },
    /** Remove default height - useful to set height with css */
    noHeight: { type: Boolean, default: !1 },
    /** Width of gallery */
    width: { type: [Number, String], default: "100%" },
    /** Remove default width */
    noWidth: { type: Boolean, default: !1 },
    /** Disable the border radius of the gallery */
    noRadius: { type: Boolean, default: !1 },
    /** Disable full size display when clicking on image */
    noZoom: { type: Boolean, default: !1 },
    /** Layer with photography icon when no images is provided */
    hasEmptyLayer: { type: Boolean, default: !0 },
    /** Lazy load image - if false, images are directly loaded */
    lazy: { type: Boolean, default: !0 },
    /** Disable blur effect on image hover */
    blur: { type: Boolean, default: !0 },
    /** Disable scale animation effect on image hover */
    scale: { type: Boolean, default: !0 },
    /** Choose color of borders between images - Should be a CSS color or CSS variable - Ex: `#000` or `var(--maz-color-bg-light)` */
    separatorColor: { type: String, default: "transparent" }
  },
  setup(o) {
    Ce((u) => ({
      a1931c44: o.separatorColor
    }));
    const e = o, t = _(() => import("./no-photography-CFaiDNTH.mjs"));
    ze(() => {
      e.imagesShownCount > 5 && console.warn('[MazUI](m-gallery) The maximum of "images-shown-count" is 5');
    });
    const s = E(() => {
      const u = e.noWidth, m = e.width, l = e.noHeight, c = e.height;
      return {
        ...u ? {} : {
          flex: `0 0 ${typeof m}` == "number" ? `${m}px` : m,
          width: typeof m == "number" ? `${m}px` : m
        },
        ...l ? {} : {
          height: typeof c == "number" ? `${c}px` : `${c}`,
          minHeight: typeof c == "number" ? `${c}px` : `${c}`
        }
      };
    }), r = E(() => e.imagesShownCount <= 5 ? e.imagesShownCount : 5), h = E(() => e.images.length - (e.images.length < r.value ? e.images.length : r.value)), p = E(() => e.images.map(
      (u) => typeof u == "object" ? { ...u, thumbnail: u.thumbnail ?? u.src } : { src: u, thumbnail: u, alt: void 0 }
    )), g = E(() => p.value.slice(0, r.value)), C = E(() => p.value.slice(r.value, e.images.length));
    function w(u) {
      return e.noRemaining ? !1 : (h.value && u + 1) === g.value.length;
    }
    return (u, m) => o.images.length > 0 || o.hasEmptyLayer ? (b(), A(
      "div",
      {
        key: 0,
        class: F(["m-gallery maz-flex", { "maz-rounded": !o.noRadius }]),
        style: ee([s.value])
      },
      [
        P("section", Ne, [
          (b(!0), A(
            te,
            null,
            se(g.value, (l, c) => (b(), A(
              "figure",
              {
                key: c,
                class: F(["m-gallery__item !maz-my-0 maz-flex maz-flex-center", [`m-gallery__item--${c + 1}`]])
              },
              [
                R(P("img", {
                  class: "m-gallery__item__image maz-flex-1",
                  src: "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7",
                  alt: l.alt,
                  loading: "lazy"
                }, null, 8, Te), [
                  [L(Fe), { src: l.thumbnail, disabled: !o.lazy }, "bg-image"],
                  [L(U), {
                    src: l.src,
                    alt: l.alt,
                    disabled: o.noZoom || w(c),
                    blurOnHover: o.blur,
                    scaleOnHover: o.scale
                  }]
                ]),
                w(c) ? R((b(), A("div", De, [
                  P(
                    "span",
                    Ue,
                    "+" + xe(h.value),
                    1
                    /* TEXT */
                  )
                ])), [
                  [L(U), {
                    src: l.src,
                    alt: l.alt,
                    disabled: o.noZoom,
                    blurOnHover: !1,
                    scaleOnHover: o.scale
                  }]
                ]) : O("v-if", !0)
              ],
              2
              /* CLASS */
            ))),
            128
            /* KEYED_FRAGMENT */
          )),
          o.hasEmptyLayer && o.images.length === 0 ? (b(), A(
            "div",
            {
              key: 0,
              class: F(["empty-layer maz-flex maz-w-full maz-bg-color-light maz-text-normal maz-flex-center", { "maz-rounded-xl": !o.noRadius }]),
              style: ee([s.value])
            },
            [
              x(L(t), { class: "maz-h-8 maz-w-8" })
            ],
            6
            /* CLASS, STYLE */
          )) : O("v-if", !0)
        ]),
        (b(!0), A(
          te,
          null,
          se(C.value, (l, c) => R((b(), A("div", {
            key: c,
            class: "m-gallery__hidden"
          })), [
            [L(U), { src: l.src, disabled: o.noZoom }]
          ])),
          128
          /* KEYED_FRAGMENT */
        ))
      ],
      6
      /* CLASS, STYLE */
    )) : O("v-if", !0);
  }
}), Ve = /* @__PURE__ */ ae(We, [["__scopeId", "data-v-4811cb18"]]);
export {
  Ve as default
};
