// maz-ui/components/chunks/MazPickerShortcuts-CBEmYjPG.mjs
import { defineComponent as h, ref as f, watch as n, openBlock as l, createElementBlock as i, Fragment as v, renderList as y, createBlock as _, normalizeClass as k, withModifiers as S, withCtx as b, createTextVNode as z, toDisplayString as B } from "vue";
import { M as g, _ as x } from "./MazPicker-CJBezF2t.mjs";
import '../assets/MazPickerShortcuts.css';const C = { class: "maz-picker-shortcuts" }, M = /* @__PURE__ */ h({
  __name: "MazPickerShortcuts",
  props: {
    color: { type: String, required: !0 },
    modelValue: {
      type: [String, Object],
      default: void 0
    },
    shortcuts: {
      type: Array,
      required: !0
    },
    double: { type: Boolean, required: !0 },
    shortcut: { type: String, default: void 0 },
    disabled: { type: Boolean, required: !0 }
  },
  emits: ["update:model-value"],
  setup(c, { emit: d }) {
    const r = c, m = d, s = f(r.shortcut);
    function u(o, e) {
      s.value = e, m("update:model-value", o);
    }
    return n(
      () => r.modelValue,
      (o) => {
        const e = o;
        e != null && e.end || (s.value = void 0);
      }
    ), n(
      () => r.shortcut,
      (o) => {
        const e = r.shortcuts.find(({ identifier: t }) => o === t);
        if (e) {
          const { value: t, identifier: a } = e;
          u(t, a);
        }
      },
      { immediate: !0 }
    ), (o, e) => (l(), i("div", C, [
      (l(!0), i(
        v,
        null,
        y(c.shortcuts, ({ identifier: t, label: a, value: p }) => (l(), _(g, {
          key: t,
          type: "button",
          size: "sm",
          disabled: c.disabled,
          color: t === s.value ? r.color : "transparent",
          class: k({ "--is-selected": t === s.value }),
          onClick: S((q) => u(p, t), ["stop"])
        }, {
          default: b(() => [
            z(
              B(a),
              1
              /* TEXT */
            )
          ]),
          _: 2
          /* DYNAMIC */
        }, 1032, ["disabled", "color", "class", "onClick"]))),
        128
        /* KEYED_FRAGMENT */
      ))
    ]));
  }
}), P = /* @__PURE__ */ x(M, [["__scopeId", "data-v-2c9090e3"]]);
export {
  P as default
};
