// maz-ui/components/chunks/MazPicker-CJBezF2t.mjs
import { nextTick as ge, defineComponent as le, defineAsyncComponent as ue, useAttrs as et, useSlots as je, onBeforeMount as We, computed as y, openBlock as g, createBlock as x, resolveDynamicComponent as ke, normalizeClass as X, withCtx as _, createElementBlock as A, createCommentVNode as W, renderSlot as ye, unref as G, ref as Q, watch as be, Fragment as de, renderList as ce, createElementVNode as ne, normalizeStyle as Se, withModifiers as Ee, createTextVNode as ze, toDisplayString as me, createVNode as R, TransitionGroup as Me, useCssVars as tt, Transition as Te, getCurrentInstance as Ue, onMounted as Fe, withDirectives as qe, mergeProps as xe, toHandlers as at, vModelDynamic as rt, onUnmounted as nt, vShow as ot } from "vue";
import '../assets/MazPicker.css';var Ye = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function Oe(t) {
  return t && t.__esModule && Object.prototype.hasOwnProperty.call(t, "default") ? t.default : t;
}
var Ne = { exports: {} };
(function(t, r) {
  (function(e, u) {
    t.exports = u();
  })(Ye, function() {
    var e = 1e3, u = 6e4, d = 36e5, V = "millisecond", Y = "second", m = "minute", T = "hour", M = "day", b = "week", C = "month", P = "quarter", H = "year", j = "date", S = "Invalid Date", v = /^(\d{4})[-/]?(\d{1,2})?[-/]?(\d{0,2})[Tt\s]*(\d{1,2})?:?(\d{1,2})?:?(\d{1,2})?[.:]?(\d+)?$/, B = /\[([^\]]+)]|Y{1,4}|M{1,4}|D{1,2}|d{1,4}|H{1,2}|h{1,2}|a|A|m{1,2}|s{1,2}|Z{1,2}|SSS/g, f = { name: "en", weekdays: "Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday".split("_"), months: "January_February_March_April_May_June_July_August_September_October_November_December".split("_"), ordinal: function(h) {
      var n = ["th", "st", "nd", "rd"], o = h % 100;
      return "[" + h + (n[(o - 20) % 10] || n[o] || n[0]) + "]";
    } }, F = function(h, n, o) {
      var c = String(h);
      return !c || c.length >= n ? h : "" + Array(n + 1 - c.length).join(o) + h;
    }, U = { s: F, z: function(h) {
      var n = -h.utcOffset(), o = Math.abs(n), c = Math.floor(o / 60), s = o % 60;
      return (n <= 0 ? "+" : "-") + F(c, 2, "0") + ":" + F(s, 2, "0");
    }, m: function h(n, o) {
      if (n.date() < o.date()) return -h(o, n);
      var c = 12 * (o.year() - n.year()) + (o.month() - n.month()), s = n.clone().add(c, C), $ = o - s < 0, a = n.clone().add(c + ($ ? -1 : 1), C);
      return +(-(c + (o - s) / ($ ? s - a : a - s)) || 0);
    }, a: function(h) {
      return h < 0 ? Math.ceil(h) || 0 : Math.floor(h);
    }, p: function(h) {
      return { M: C, y: H, w: b, d: M, D: j, h: T, m, s: Y, ms: V, Q: P }[h] || String(h || "").toLowerCase().replace(/s$/, "");
    }, u: function(h) {
      return h === void 0;
    } }, p = "en", w = {};
    w[p] = f;
    var z = "$isDayjsObject", I = function(h) {
      return h instanceof N || !(!h || !h[z]);
    }, E = function h(n, o, c) {
      var s;
      if (!n) return p;
      if (typeof n == "string") {
        var $ = n.toLowerCase();
        w[$] && (s = $), o && (w[$] = o, s = $);
        var a = n.split("-");
        if (!s && a.length > 1) return h(a[0]);
      } else {
        var i = n.name;
        w[i] = n, s = i;
      }
      return !c && s && (p = s), s || !c && p;
    }, L = function(h, n) {
      if (I(h)) return h.clone();
      var o = typeof n == "object" ? n : {};
      return o.date = h, o.args = arguments, new N(o);
    }, O = U;
    O.l = E, O.i = I, O.w = function(h, n) {
      return L(h, { locale: n.$L, utc: n.$u, x: n.$x, $offset: n.$offset });
    };
    var N = function() {
      function h(o) {
        this.$L = E(o.locale, null, !0), this.parse(o), this.$x = this.$x || o.x || {}, this[z] = !0;
      }
      var n = h.prototype;
      return n.parse = function(o) {
        this.$d = function(c) {
          var s = c.date, $ = c.utc;
          if (s === null) return /* @__PURE__ */ new Date(NaN);
          if (O.u(s)) return /* @__PURE__ */ new Date();
          if (s instanceof Date) return new Date(s);
          if (typeof s == "string" && !/Z$/i.test(s)) {
            var a = s.match(v);
            if (a) {
              var i = a[2] - 1 || 0, k = (a[7] || "0").substring(0, 3);
              return $ ? new Date(Date.UTC(a[1], i, a[3] || 1, a[4] || 0, a[5] || 0, a[6] || 0, k)) : new Date(a[1], i, a[3] || 1, a[4] || 0, a[5] || 0, a[6] || 0, k);
            }
          }
          return new Date(s);
        }(o), this.init();
      }, n.init = function() {
        var o = this.$d;
        this.$y = o.getFullYear(), this.$M = o.getMonth(), this.$D = o.getDate(), this.$W = o.getDay(), this.$H = o.getHours(), this.$m = o.getMinutes(), this.$s = o.getSeconds(), this.$ms = o.getMilliseconds();
      }, n.$utils = function() {
        return O;
      }, n.isValid = function() {
        return this.$d.toString() !== S;
      }, n.isSame = function(o, c) {
        var s = L(o);
        return this.startOf(c) <= s && s <= this.endOf(c);
      }, n.isAfter = function(o, c) {
        return L(o) < this.startOf(c);
      }, n.isBefore = function(o, c) {
        return this.endOf(c) < L(o);
      }, n.$g = function(o, c, s) {
        return O.u(o) ? this[c] : this.set(s, o);
      }, n.unix = function() {
        return Math.floor(this.valueOf() / 1e3);
      }, n.valueOf = function() {
        return this.$d.getTime();
      }, n.startOf = function(o, c) {
        var s = this, $ = !!O.u(c) || c, a = O.p(o), i = function(ae, J) {
          var re = O.w(s.$u ? Date.UTC(s.$y, J, ae) : new Date(s.$y, J, ae), s);
          return $ ? re : re.endOf(M);
        }, k = function(ae, J) {
          return O.w(s.toDate()[ae].apply(s.toDate("s"), ($ ? [0, 0, 0, 0] : [23, 59, 59, 999]).slice(J)), s);
        }, q = this.$W, Z = this.$M, K = this.$D, oe = "set" + (this.$u ? "UTC" : "");
        switch (a) {
          case H:
            return $ ? i(1, 0) : i(31, 11);
          case C:
            return $ ? i(1, Z) : i(0, Z + 1);
          case b:
            var ee = this.$locale().weekStart || 0, ie = (q < ee ? q + 7 : q) - ee;
            return i($ ? K - ie : K + (6 - ie), Z);
          case M:
          case j:
            return k(oe + "Hours", 0);
          case T:
            return k(oe + "Minutes", 1);
          case m:
            return k(oe + "Seconds", 2);
          case Y:
            return k(oe + "Milliseconds", 3);
          default:
            return this.clone();
        }
      }, n.endOf = function(o) {
        return this.startOf(o, !1);
      }, n.$set = function(o, c) {
        var s, $ = O.p(o), a = "set" + (this.$u ? "UTC" : ""), i = (s = {}, s[M] = a + "Date", s[j] = a + "Date", s[C] = a + "Month", s[H] = a + "FullYear", s[T] = a + "Hours", s[m] = a + "Minutes", s[Y] = a + "Seconds", s[V] = a + "Milliseconds", s)[$], k = $ === M ? this.$D + (c - this.$W) : c;
        if ($ === C || $ === H) {
          var q = this.clone().set(j, 1);
          q.$d[i](k), q.init(), this.$d = q.set(j, Math.min(this.$D, q.daysInMonth())).$d;
        } else i && this.$d[i](k);
        return this.init(), this;
      }, n.set = function(o, c) {
        return this.clone().$set(o, c);
      }, n.get = function(o) {
        return this[O.p(o)]();
      }, n.add = function(o, c) {
        var s, $ = this;
        o = Number(o);
        var a = O.p(c), i = function(Z) {
          var K = L($);
          return O.w(K.date(K.date() + Math.round(Z * o)), $);
        };
        if (a === C) return this.set(C, this.$M + o);
        if (a === H) return this.set(H, this.$y + o);
        if (a === M) return i(1);
        if (a === b) return i(7);
        var k = (s = {}, s[m] = u, s[T] = d, s[Y] = e, s)[a] || 1, q = this.$d.getTime() + o * k;
        return O.w(q, this);
      }, n.subtract = function(o, c) {
        return this.add(-1 * o, c);
      }, n.format = function(o) {
        var c = this, s = this.$locale();
        if (!this.isValid()) return s.invalidDate || S;
        var $ = o || "YYYY-MM-DDTHH:mm:ssZ", a = O.z(this), i = this.$H, k = this.$m, q = this.$M, Z = s.weekdays, K = s.months, oe = s.meridiem, ee = function(J, re, fe, ve) {
          return J && (J[re] || J(c, $)) || fe[re].slice(0, ve);
        }, ie = function(J) {
          return O.s(i % 12 || 12, J, "0");
        }, ae = oe || function(J, re, fe) {
          var ve = J < 12 ? "AM" : "PM";
          return fe ? ve.toLowerCase() : ve;
        };
        return $.replace(B, function(J, re) {
          return re || function(fe) {
            switch (fe) {
              case "YY":
                return String(c.$y).slice(-2);
              case "YYYY":
                return O.s(c.$y, 4, "0");
              case "M":
                return q + 1;
              case "MM":
                return O.s(q + 1, 2, "0");
              case "MMM":
                return ee(s.monthsShort, q, K, 3);
              case "MMMM":
                return ee(K, q);
              case "D":
                return c.$D;
              case "DD":
                return O.s(c.$D, 2, "0");
              case "d":
                return String(c.$W);
              case "dd":
                return ee(s.weekdaysMin, c.$W, Z, 2);
              case "ddd":
                return ee(s.weekdaysShort, c.$W, Z, 3);
              case "dddd":
                return Z[c.$W];
              case "H":
                return String(i);
              case "HH":
                return O.s(i, 2, "0");
              case "h":
                return ie(1);
              case "hh":
                return ie(2);
              case "a":
                return ae(i, k, !0);
              case "A":
                return ae(i, k, !1);
              case "m":
                return String(k);
              case "mm":
                return O.s(k, 2, "0");
              case "s":
                return String(c.$s);
              case "ss":
                return O.s(c.$s, 2, "0");
              case "SSS":
                return O.s(c.$ms, 3, "0");
              case "Z":
                return a;
            }
            return null;
          }(J) || a.replace(":", "");
        });
      }, n.utcOffset = function() {
        return 15 * -Math.round(this.$d.getTimezoneOffset() / 15);
      }, n.diff = function(o, c, s) {
        var $, a = this, i = O.p(c), k = L(o), q = (k.utcOffset() - this.utcOffset()) * u, Z = this - k, K = function() {
          return O.m(a, k);
        };
        switch (i) {
          case H:
            $ = K() / 12;
            break;
          case C:
            $ = K();
            break;
          case P:
            $ = K() / 3;
            break;
          case b:
            $ = (Z - q) / 6048e5;
            break;
          case M:
            $ = (Z - q) / 864e5;
            break;
          case T:
            $ = Z / d;
            break;
          case m:
            $ = Z / u;
            break;
          case Y:
            $ = Z / e;
            break;
          default:
            $ = Z;
        }
        return s ? $ : O.a($);
      }, n.daysInMonth = function() {
        return this.endOf(C).$D;
      }, n.$locale = function() {
        return w[this.$L];
      }, n.locale = function(o, c) {
        if (!o) return this.$L;
        var s = this.clone(), $ = E(o, c, !0);
        return $ && (s.$L = $), s;
      }, n.clone = function() {
        return O.w(this.$d, this);
      }, n.toDate = function() {
        return new Date(this.valueOf());
      }, n.toJSON = function() {
        return this.isValid() ? this.toISOString() : null;
      }, n.toISOString = function() {
        return this.$d.toISOString();
      }, n.toString = function() {
        return this.$d.toUTCString();
      }, h;
    }(), l = N.prototype;
    return L.prototype = l, [["$ms", V], ["$s", Y], ["$m", m], ["$H", T], ["$W", M], ["$M", C], ["$y", H], ["$D", j]].forEach(function(h) {
      l[h[1]] = function(n) {
        return this.$g(n, h[0], h[1]);
      };
    }), L.extend = function(h, n) {
      return h.$i || (h(n, N, L), h.$i = !0), L;
    }, L.locale = E, L.isDayjs = I, L.unix = function(h) {
      return L(1e3 * h);
    }, L.en = w[p], L.Ls = w, L.p = {}, L;
  });
})(Ne);
var it = Ne.exports;
const D = /* @__PURE__ */ Oe(it);
var Ze = { exports: {} };
(function(t, r) {
  (function(e, u) {
    t.exports = u();
  })(Ye, function() {
    var e = { LTS: "h:mm:ss A", LT: "h:mm A", L: "MM/DD/YYYY", LL: "MMMM D, YYYY", LLL: "MMMM D, YYYY h:mm A", LLLL: "dddd, MMMM D, YYYY h:mm A" }, u = /(\[[^[]*\])|([-_:/.,()\s]+)|(A|a|YYYY|YY?|MM?M?M?|Do|DD?|hh?|HH?|mm?|ss?|S{1,3}|z|ZZ?)/g, d = /\d\d/, V = /\d\d?/, Y = /\d*[^-_:/,()\s\d]+/, m = {}, T = function(S) {
      return (S = +S) + (S > 68 ? 1900 : 2e3);
    }, M = function(S) {
      return function(v) {
        this[S] = +v;
      };
    }, b = [/[+-]\d\d:?(\d\d)?|Z/, function(S) {
      (this.zone || (this.zone = {})).offset = function(v) {
        if (!v || v === "Z") return 0;
        var B = v.match(/([+-]|\d\d)/g), f = 60 * B[1] + (+B[2] || 0);
        return f === 0 ? 0 : B[0] === "+" ? -f : f;
      }(S);
    }], C = function(S) {
      var v = m[S];
      return v && (v.indexOf ? v : v.s.concat(v.f));
    }, P = function(S, v) {
      var B, f = m.meridiem;
      if (f) {
        for (var F = 1; F <= 24; F += 1) if (S.indexOf(f(F, 0, v)) > -1) {
          B = F > 12;
          break;
        }
      } else B = S === (v ? "pm" : "PM");
      return B;
    }, H = { A: [Y, function(S) {
      this.afternoon = P(S, !1);
    }], a: [Y, function(S) {
      this.afternoon = P(S, !0);
    }], S: [/\d/, function(S) {
      this.milliseconds = 100 * +S;
    }], SS: [d, function(S) {
      this.milliseconds = 10 * +S;
    }], SSS: [/\d{3}/, function(S) {
      this.milliseconds = +S;
    }], s: [V, M("seconds")], ss: [V, M("seconds")], m: [V, M("minutes")], mm: [V, M("minutes")], H: [V, M("hours")], h: [V, M("hours")], HH: [V, M("hours")], hh: [V, M("hours")], D: [V, M("day")], DD: [d, M("day")], Do: [Y, function(S) {
      var v = m.ordinal, B = S.match(/\d+/);
      if (this.day = B[0], v) for (var f = 1; f <= 31; f += 1) v(f).replace(/\[|\]/g, "") === S && (this.day = f);
    }], M: [V, M("month")], MM: [d, M("month")], MMM: [Y, function(S) {
      var v = C("months"), B = (C("monthsShort") || v.map(function(f) {
        return f.slice(0, 3);
      })).indexOf(S) + 1;
      if (B < 1) throw new Error();
      this.month = B % 12 || B;
    }], MMMM: [Y, function(S) {
      var v = C("months").indexOf(S) + 1;
      if (v < 1) throw new Error();
      this.month = v % 12 || v;
    }], Y: [/[+-]?\d+/, M("year")], YY: [d, function(S) {
      this.year = T(S);
    }], YYYY: [/\d{4}/, M("year")], Z: b, ZZ: b };
    function j(S) {
      var v, B;
      v = S, B = m && m.formats;
      for (var f = (S = v.replace(/(\[[^\]]+])|(LTS?|l{1,4}|L{1,4})/g, function(E, L, O) {
        var N = O && O.toUpperCase();
        return L || B[O] || e[O] || B[N].replace(/(\[[^\]]+])|(MMMM|MM|DD|dddd)/g, function(l, h, n) {
          return h || n.slice(1);
        });
      })).match(u), F = f.length, U = 0; U < F; U += 1) {
        var p = f[U], w = H[p], z = w && w[0], I = w && w[1];
        f[U] = I ? { regex: z, parser: I } : p.replace(/^\[|\]$/g, "");
      }
      return function(E) {
        for (var L = {}, O = 0, N = 0; O < F; O += 1) {
          var l = f[O];
          if (typeof l == "string") N += l.length;
          else {
            var h = l.regex, n = l.parser, o = E.slice(N), c = h.exec(o)[0];
            n.call(L, c), E = E.replace(c, "");
          }
        }
        return function(s) {
          var $ = s.afternoon;
          if ($ !== void 0) {
            var a = s.hours;
            $ ? a < 12 && (s.hours += 12) : a === 12 && (s.hours = 0), delete s.afternoon;
          }
        }(L), L;
      };
    }
    return function(S, v, B) {
      B.p.customParseFormat = !0, S && S.parseTwoDigitYear && (T = S.parseTwoDigitYear);
      var f = v.prototype, F = f.parse;
      f.parse = function(U) {
        var p = U.date, w = U.utc, z = U.args;
        this.$u = w;
        var I = z[1];
        if (typeof I == "string") {
          var E = z[2] === !0, L = z[3] === !0, O = E || L, N = z[2];
          L && (N = z[2]), m = this.$locale(), !E && N && (m = B.Ls[N]), this.$d = function(o, c, s) {
            try {
              if (["x", "X"].indexOf(c) > -1) return new Date((c === "X" ? 1e3 : 1) * o);
              var $ = j(c)(o), a = $.year, i = $.month, k = $.day, q = $.hours, Z = $.minutes, K = $.seconds, oe = $.milliseconds, ee = $.zone, ie = /* @__PURE__ */ new Date(), ae = k || (a || i ? 1 : ie.getDate()), J = a || ie.getFullYear(), re = 0;
              a && !i || (re = i > 0 ? i - 1 : ie.getMonth());
              var fe = q || 0, ve = Z || 0, Be = K || 0, Ce = oe || 0;
              return ee ? new Date(Date.UTC(J, re, ae, fe, ve, Be, Ce + 60 * ee.offset * 1e3)) : s ? new Date(Date.UTC(J, re, ae, fe, ve, Be, Ce)) : new Date(J, re, ae, fe, ve, Be, Ce);
            } catch {
              return /* @__PURE__ */ new Date("");
            }
          }(p, I, w), this.init(), N && N !== !0 && (this.$L = this.locale(N).$L), O && p != this.format(I) && (this.$d = /* @__PURE__ */ new Date("")), m = {};
        } else if (I instanceof Array) for (var l = I.length, h = 1; h <= l; h += 1) {
          z[1] = I[h - 1];
          var n = B.apply(this, z);
          if (n.isValid()) {
            this.$d = n.$d, this.$L = n.$L, this.init();
            break;
          }
          h === l && (this.$d = /* @__PURE__ */ new Date(""));
        }
        else F.call(this, U);
      };
    };
  });
})(Ze);
var lt = Ze.exports;
const st = /* @__PURE__ */ Oe(lt);
var Re = { exports: {} };
(function(t, r) {
  (function(e, u) {
    t.exports = u();
  })(Ye, function() {
    return function(e, u, d) {
      u.prototype.isBetween = function(V, Y, m, T) {
        var M = d(V), b = d(Y), C = (T = T || "()")[0] === "(", P = T[1] === ")";
        return (C ? this.isAfter(M, m) : !this.isBefore(M, m)) && (P ? this.isBefore(b, m) : !this.isAfter(b, m)) || (C ? this.isBefore(M, m) : !this.isAfter(M, m)) && (P ? this.isAfter(b, m) : !this.isBefore(b, m));
      };
    };
  });
})(Re);
var ut = Re.exports;
const dt = /* @__PURE__ */ Oe(ut), Ve = "__maz-click-outside__";
function Ge() {
  return document.ontouchstart === null ? "touchstart" : "click";
}
async function Je(t, r) {
  try {
    Qe(t);
    const e = r.instance, u = r.value, d = typeof u == "function";
    if (!d)
      throw new Error("[maz-ui](vClickOutside) the callback should be a function");
    await ge(), t[Ve] = (Y) => {
      if ((!t || Y.target && !t.contains(Y.target)) && u && d)
        return u.call(e, Y);
    };
    const V = Ge();
    document.addEventListener(V, t[Ve], { passive: !0 });
  } catch (e) {
    console.error("[maz-ui](vClickOutside)", e);
  }
}
function Qe(t) {
  try {
    const r = Ge();
    document.removeEventListener(r, t[Ve], !1), delete t[Ve];
  } catch (r) {
    console.error("[maz-ui](vClickOutside)", r);
  }
}
function ct(t, r) {
  try {
    if (r.value === r.oldValue)
      return;
    Je(t, r);
  } catch (e) {
    console.error("[maz-ui](vClickOutside)", e);
  }
}
const ft = {
  mounted: Je,
  updated: ct,
  unmounted: Qe
}, mt = {
  month: "short",
  day: "numeric",
  year: "numeric"
};
function te(t, r, e) {
  if (r === void 0)
    throw new TypeError("[maz-ui](FilterDate) The `locale` attribute is required.");
  if (typeof r != "string")
    throw new TypeError("[maz-ui](FilterDate) The `locale` attribute must be a string.");
  const u = e ?? mt;
  try {
    const d = t instanceof Date ? t : new Date(t);
    return new Intl.DateTimeFormat(r, u).format(d);
  } catch (d) {
    throw new Error(`[maz-ui](FilterDate) ${d}`);
  }
}
var Xe = { exports: {} };
(function(t, r) {
  (function(e, u) {
    t.exports = u();
  })(Ye, function() {
    return function(e, u) {
      u.prototype.weekday = function(d) {
        var V = this.$locale().weekStart || 0, Y = this.$W, m = (Y < V ? Y + 7 : Y) - V;
        return this.$utils().u(d) ? m : this.subtract(m, "day").add(d, "day");
      };
    };
  });
})(Xe);
var vt = Xe.exports;
const ht = /* @__PURE__ */ Oe(vt);
function De(t) {
  return t ? (t = t.toString(), t.charAt(0).toUpperCase() + t.slice(1)) : "";
}
D.extend(ht);
function He({
  value: t,
  locale: r,
  options: e
}) {
  return t ? De(te(t, r, e)) : void 0;
}
function yt({
  value: t,
  locale: r,
  options: e
}) {
  const u = He({
    value: t.start,
    locale: r,
    options: e
  }), d = He({
    value: t.end,
    locale: r,
    options: e
  });
  return u || d ? `${u || "..."} - ${d || "..."}` : void 0;
}
function pt(t) {
  return D(t).startOf("month").day();
}
function bt(t, r) {
  const e = D().locale(t), u = [];
  for (let d = 0; d < 7; d++) {
    const V = e.day(d + r).toDate(), Y = te(V, t, { weekday: "short" });
    u.push(Y);
  }
  return u;
}
function Dt(t) {
  return D(t).daysInMonth();
}
function gt(t) {
  return D().isSame(t, "date");
}
function he(t, r, e) {
  return D(t).isSame(r, e);
}
function Ke(t, r) {
  return D(t).day() === r;
}
function Pe(t, r = "YYYY-MM-DD") {
  if (t)
    return D(t).format(r);
}
function kt(t, r = "YYYY-MM-DD") {
  return {
    start: Pe(t.start, r),
    end: Pe(t.end, r)
  };
}
function Ie({
  value: t,
  minDate: r,
  maxDate: e,
  format: u
}) {
  return r && D(t).isBefore(r) ? {
    newValue: r,
    newCurrentDate: D(r, u).format()
  } : e && D(t).isAfter(e) ? {
    newValue: e,
    newCurrentDate: D(e, u).format()
  } : {
    newValue: void 0,
    newCurrentDate: void 0
  };
}
function we({
  value: t,
  disabledWeekly: r
}) {
  return r.some((e) => Ke(t, e));
}
function $e({
  value: t,
  disabledDates: r
}) {
  return r.some((e) => D(t).isSame(D(e), "date"));
}
const wt = function(t, r, e = 0, u = !0) {
  t.scrollTo({
    top: r.offsetTop - e,
    behavior: u ? "smooth" : "auto"
  });
};
function Le(t, r) {
  return t.reduce((e, u) => Math.abs(u - r) < Math.abs(e - r) ? u : e);
}
function $t() {
  try {
    return typeof window > "u" ? void 0 : window.navigator.language;
  } catch (t) {
    throw new Error(`[MazPhoneNumberInput] (browserLocale) ${t}`);
  }
}
async function Mt() {
  try {
    const t = await fetch("https://ipwho.is"), { country_code: r } = await t.json();
    return r;
  } catch (t) {
    console.error(`[maz-ui](MazPicker)(fetchCountryCode) ${t}`);
  }
}
const St = /* @__PURE__ */ le({
  __name: "MazBtn",
  props: {
    variant: { default: "button" },
    size: { default: "md" },
    color: { default: "primary" },
    type: { default: "button" },
    rounded: { type: Boolean },
    roundedSize: { default: "lg" },
    noRounded: { type: Boolean },
    outline: { type: Boolean },
    pastel: { type: Boolean },
    block: { type: Boolean },
    noUnderline: { type: Boolean },
    loading: { type: Boolean },
    disabled: { type: Boolean },
    fab: { type: Boolean },
    icon: { type: [String, Function, Object], default: void 0 },
    leftIcon: { type: [String, Function, Object], default: void 0 },
    rightIcon: { type: [String, Function, Object], default: void 0 },
    noPadding: { type: Boolean },
    noElevation: { type: Boolean },
    contentClass: { default: void 0 }
  },
  setup(t) {
    const r = t, e = ue(() => import("./MazSpinner-DnVHWljy.mjs")), u = ue(() => import("./MazIcon-BDGxCy9M.mjs")), { href: d, to: V } = et(), Y = je();
    We(() => {
      r.icon && !r.fab && console.error('[maz-ui](MazBtn) the prop "icon" must be used only with "fab" props');
    });
    const m = y(() => d ? "a" : V ? "router-link" : "button"), T = y(
      () => r.pastel ? `--${r.color}-pastel` : r.outline ? `--${r.color}-outline` : `--${r.color}`
    ), M = y(
      () => (r.loading || r.disabled) && m.value === "button"
    ), b = y(() => M.value ? "--cursor-default" : "--cursor-pointer"), C = y(() => `--is-${r.variant}`), P = y(() => r.loading && r.variant === "button"), H = y(() => !!Y["left-icon"] || r.leftIcon), j = y(() => !!Y["right-icon"] || r.rightIcon), S = y(() => H.value || j.value), v = y(() => r.fab && (r.icon || !!Y.icon)), B = y(() => m.value === "button" ? r.type : void 0);
    return (f, F) => (g(), x(ke(m.value), {
      disabled: M.value || void 0,
      class: X(["m-btn", [
        `--${f.size}`,
        !f.fab && f.roundedSize && `--rounded-${f.roundedSize}`,
        T.value,
        b.value,
        C.value,
        {
          "--block": f.block,
          "--no-underline": f.noUnderline,
          "--fab": f.fab,
          "--loading": f.loading,
          "--disabled": M.value,
          "--icon": S.value,
          "--rounded": f.rounded,
          "--no-rounded": f.noRounded,
          "--no-padding": f.noPadding,
          "--no-elevation": f.noElevation
        }
      ]]),
      type: B.value
    }, {
      default: _(() => [
        H.value ? (g(), A(
          "div",
          {
            key: 0,
            class: X(["m-btn__icon-left maz-flex maz-flex-center", { "maz-invisible": P.value }])
          },
          [
            W(`
        @slot left-icon - The icon to display on the left of the button
      `),
            ye(f.$slots, "left-icon", {}, () => [
              typeof f.leftIcon == "string" ? (g(), x(G(u), {
                key: 0,
                name: f.leftIcon
              }, null, 8, ["name"])) : f.leftIcon ? (g(), x(ke(f.leftIcon), { key: 1 })) : W("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : W("v-if", !0),
        v.value ? (g(), A(
          "div",
          {
            key: 1,
            class: X(["m-btn__icon", { "maz-invisible": P.value }])
          },
          [
            W(`
        @slot icon - The icon to display on the fab button
      `),
            ye(f.$slots, "icon", {}, () => [
              typeof f.icon == "string" ? (g(), x(G(u), {
                key: 0,
                name: f.icon
              }, null, 8, ["name"])) : f.icon ? (g(), x(ke(f.icon), { key: 1 })) : W("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : W("v-if", !0),
        f.$slots.default ? (g(), A(
          "span",
          {
            key: 2,
            class: X([{ "maz-invisible": P.value }, f.contentClass])
          },
          [
            W(`
        @slot default - The content of the button
      `),
            ye(f.$slots, "default", {}, void 0, !0)
          ],
          2
          /* CLASS */
        )) : W("v-if", !0),
        j.value ? (g(), A(
          "div",
          {
            key: 3,
            class: X(["m-btn__icon-right", { "maz-invisible": P.value }])
          },
          [
            W(`
        @slot right-icon - The icon to display on the right of the button
      `),
            ye(f.$slots, "right-icon", {}, () => [
              typeof f.rightIcon == "string" ? (g(), x(G(u), {
                key: 0,
                name: f.rightIcon
              }, null, 8, ["name"])) : f.rightIcon ? (g(), x(ke(f.rightIcon), { key: 1 })) : W("v-if", !0)
            ], !0)
          ],
          2
          /* CLASS */
        )) : W("v-if", !0),
        P.value ? (g(), x(G(e), {
          key: 4,
          class: "m-btn-loader",
          size: "2em",
          color: f.color
        }, null, 8, ["color"])) : W("v-if", !0)
      ]),
      _: 3
      /* FORWARDED */
    }, 8, ["disabled", "class", "type"]));
  }
}), se = (t, r) => {
  const e = t.__vccOpts || t;
  for (const [u, d] of r)
    e[u] = d;
  return e;
}, pe = /* @__PURE__ */ se(St, [["__scopeId", "data-v-48f14785"]]), zt = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  default: pe
}, Symbol.toStringTag, { value: "Module" })), Vt = { class: "m-picker-time__column__items" }, Yt = /* @__PURE__ */ le({
  __name: "MazPickerTime",
  props: {
    modelValue: {
      type: [String, Object],
      default: void 0
    },
    calendarDate: { type: String, required: !0 },
    formatterOptions: {
      type: Object,
      required: !0
    },
    locale: { type: String, required: !0 },
    color: { type: String, required: !0 },
    isOpen: { type: Boolean, required: !0 },
    hasDate: { type: Boolean, required: !0 },
    minuteInterval: { type: Number, required: !0 },
    disabledHours: { type: Array, default: void 0 },
    format: { type: String, required: !0 },
    isHour12: { type: Boolean, required: !0 },
    minDate: { type: String, default: void 0 },
    maxDate: { type: String, default: void 0 },
    disabled: { type: Boolean, required: !0 }
  },
  emits: ["update:model-value"],
  setup(t, { emit: r }) {
    const e = t, u = r, d = y({
      get: () => e.modelValue,
      set: (p) => {
        u("update:model-value", p);
      }
    }), V = y(() => Array.from({ length: e.isHour12 ? 12 : 24 }, (p, w) => w).map((p) => {
      const w = p + (e.isHour12 ? 1 : 0), z = S(w), I = D(d.value).set("hour", z), E = B(z) || (e.minDate && d.value ? D(e.minDate).isAfter(I, "hour") : !1) || (e.maxDate && d.value ? D(e.maxDate).isBefore(I, "hour") : !1);
      return {
        label: `${w < 10 ? "0" : ""}${w}`,
        value: D(d.value).set("hour", z),
        isDisabled: E
      };
    })), Y = y(() => {
      const p = Math.floor(60 / e.minuteInterval) - 0;
      return Array.from({ length: p }, (w, z) => z * e.minuteInterval).map((w) => {
        const z = D(d.value).set("minute", w), I = (e.minDate && d.value ? D(e.minDate).isAfter(z, "minute") : !1) || (e.maxDate && d.value ? D(e.maxDate).isBefore(z, "minute") : !1);
        return {
          label: `${w < 10 ? "0" : ""}${w}`,
          value: z,
          isDisabled: I
        };
      });
    }), m = y(() => typeof d.value == "string" ? P(D(d.value).get("hour")) : void 0), T = y(
      () => typeof d.value == "string" ? Le(
        Y.value.map(({ value: p }) => p.get("minute")),
        D(d.value).get("minute")
      ) : void 0
    ), M = y(() => typeof m.value == "number" ? m.value >= 12 ? "pm" : "am" : void 0), b = y(
      () => e.isHour12 ? [
        { label: "AM", value: "am" },
        { label: "PM", value: "pm" }
      ] : []
    ), C = y(() => {
      const p = [
        {
          identifier: "hour",
          values: V.value
        },
        {
          identifier: "minute",
          values: Y.value
        }
      ];
      return e.isHour12 && p.push({
        identifier: "ampm",
        values: b.value
      }), p;
    });
    function P(p) {
      if (!e.disabledHours)
        return p;
      const w = Array.from({ length: 24 }, (I, E) => E).filter(
        (I) => {
          var E;
          return !((E = e.disabledHours) != null && E.includes(I)) || !0;
        }
      ), z = Le(w, p);
      return z !== p && U("hour", D(d.value).set("hour", z)), z;
    }
    const H = Q(), j = Q();
    function S(p) {
      if (e.isHour12) {
        const z = M.value === "pm" ? p + 12 : p;
        return z === 12 ? 0 : z === 24 ? 12 : z;
      } else
        return p;
    }
    be(
      () => e.isOpen,
      async (p) => {
        var w;
        if (p) {
          if (await ge(), H.value) {
            const z = H.value.querySelector(
              ".m-picker-time__column .m-btn"
            ), I = z == null ? void 0 : z.offsetHeight, L = ((w = H.value) == null ? void 0 : w.offsetHeight) / 2 - I / 2;
            j.value = L / 16;
          }
          v(!1);
        }
      },
      { immediate: !0 }
    ), be(
      () => e.modelValue,
      async (p, w) => {
        p !== w && (await ge(), v(!0));
      },
      { immediate: !0 }
    );
    function v(p) {
      F("hour", p), F("minute", p), e.isHour12 && F("ampm", p);
    }
    function B(p) {
      return (e.disabledHours && e.disabledHours.includes(p)) ?? !1;
    }
    function f(p, w) {
      return w === "am" || w === "pm" ? M.value === w : p === "hour" ? m.value === w.get(p) : p === "minute" ? T.value === w.get(p) : !1;
    }
    async function F(p, w = !0) {
      if (H.value) {
        const z = H.value.querySelector(`.m-picker-time__column__${p}`), I = H.value.querySelector(
          `.m-picker-time__column__${p} .--is-selected`
        );
        j.value && z && I && H.value && (await ge(), wt(z, I, j.value * 16, w));
      }
    }
    async function U(p, w) {
      const z = D(d.value);
      if (p === "hour" && typeof w == "object" && (d.value = z.set("hour", w.get("hour")).format()), p === "minute" && typeof w == "object") {
        const I = z.set("minute", w.get("minute"));
        d.value = I.format();
      }
      if (p === "ampm" && (M.value !== w || !m.value) && (w === "am" && (d.value = z.set("hour", z.get("hour")).subtract(12, "hour").format()), w === "pm")) {
        const I = z.get("hour"), E = I + 12 > 12 && I + 12 < 24 ? I + 12 : I === 0 ? 12 : I;
        d.value = z.set("hour", E).format();
      }
      await ge(), F(p);
    }
    return (p, w) => (g(), A(
      "div",
      {
        ref_key: "MazPickerTime",
        ref: H,
        class: X(["m-picker-time", { "--has-date": t.hasDate }])
      },
      [
        (g(!0), A(
          de,
          null,
          ce(C.value, ({ values: z, identifier: I }, E) => (g(), A(
            "div",
            {
              key: E,
              class: X(["m-picker-time__column", [`m-picker-time__column__${I}`]])
            },
            [
              ne(
                "div",
                {
                  style: Se({ height: `${j.value}rem` }),
                  class: "m-picker-time__column__divider"
                },
                null,
                4
                /* STYLE */
              ),
              ne("div", Vt, [
                (g(!0), A(
                  de,
                  null,
                  ce(z, ({ value: L, label: O, isDisabled: N }, l) => (g(), x(pe, {
                    key: l,
                    size: "xs",
                    color: f(I, L) ? t.color : "transparent",
                    class: X({ "--is-selected": f(I, L) }),
                    disabled: N || t.disabled,
                    type: "button",
                    onClick: Ee((h) => U(I, L), ["stop"])
                  }, {
                    default: _(() => [
                      ze(
                        me(O),
                        1
                        /* TEXT */
                      )
                    ]),
                    _: 2
                    /* DYNAMIC */
                  }, 1032, ["color", "class", "disabled", "onClick"]))),
                  128
                  /* KEYED_FRAGMENT */
                ))
              ]),
              ne(
                "div",
                {
                  style: Se({ height: `${j.value}rem` }),
                  class: "m-picker-time__column__divider"
                },
                null,
                4
                /* STYLE */
              )
            ],
            2
            /* CLASS */
          ))),
          128
          /* KEYED_FRAGMENT */
        ))
      ],
      2
      /* CLASS */
    ));
  }
}), Ot = /* @__PURE__ */ se(Yt, [["__scopeId", "data-v-ded59006"]]), Bt = {
  key: 0,
  class: "m-picker-header__date"
}, Ct = /* @__PURE__ */ le({
  __name: "MazPickerHeader",
  props: {
    modelValue: {
      type: [String, Object],
      default: void 0
    },
    color: { type: String, required: !0 },
    locale: { type: String, required: !0 },
    noShortcuts: { type: Boolean, required: !0 },
    double: { type: Boolean, required: !0 },
    hasDate: { type: Boolean, required: !0 },
    hasTime: { type: Boolean, required: !0 },
    formatterOptions: {
      type: Object,
      required: !0
    },
    calendarDate: { type: String, required: !0 }
  },
  setup(t) {
    const r = t, e = y(
      () => {
        var b;
        return typeof r.modelValue == "string" ? r.modelValue : (b = r.modelValue) == null ? void 0 : b.start;
      }
    ), u = Q("maz-slidevnext"), d = y(() => r.modelValue && typeof r.modelValue == "object" ? `${r.modelValue.start ? te(r.modelValue.start, r.locale, {
      year: "numeric",
      timeZone: r.formatterOptions.timeZone
    }) : "..."} - ${r.modelValue.end ? te(r.modelValue.end, r.locale, {
      year: "numeric",
      timeZone: r.formatterOptions.timeZone
    }) : "..."}` : typeof r.modelValue == "string" ? te(r.modelValue, r.locale, {
      year: "numeric",
      timeZone: r.formatterOptions.timeZone
    }) : "-"), V = y(() => r.hasDate ? [d.value] : void 0), Y = y(() => {
      if (r.modelValue && typeof r.modelValue == "object" && (r.modelValue.start || r.modelValue.end)) {
        const b = r.noShortcuts && !r.double ? "short" : "long";
        return `${r.modelValue.start ? De(
          te(r.modelValue.start, r.locale, {
            weekday: b,
            month: b,
            day: "numeric",
            timeZone: r.formatterOptions.timeZone
          })
        ) : "..."} - ${r.modelValue.end ? De(
          te(r.modelValue.end, r.locale, {
            weekday: b,
            month: b,
            day: "numeric",
            timeZone: r.formatterOptions.timeZone
          })
        ) : "..."}`;
      } else if (typeof r.modelValue == "string")
        return De(
          te(r.modelValue, r.locale, {
            weekday: "long",
            month: "long",
            day: "numeric",
            timeZone: r.formatterOptions.timeZone
          })
        );
      return "-";
    }), m = y(() => r.hasDate ? [Y.value] : void 0), T = y(() => e.value ? te(e.value, r.locale, {
      timeStyle: "short",
      timeZone: r.formatterOptions.timeZone,
      hour12: r.formatterOptions.hour12
    }) : void 0), M = y(() => r.hasTime ? [T.value] : void 0);
    return be(
      () => r.modelValue,
      (b, C) => {
        const P = typeof b == "object" ? b.start : b, H = typeof C == "object" ? C.start : C;
        u.value = D(P).isAfter(H, "date") || D(P).isSame(H, "date") ? "maz-slidevnext" : "maz-slidevprev";
      }
    ), (b, C) => (g(), A(
      "div",
      {
        class: X(["m-picker-header", [`--${t.color}`]])
      },
      [
        t.hasDate ? (g(), A("div", Bt, [
          R(Me, {
            name: u.value,
            tag: "div",
            class: "m-picker-header__year-transition"
          }, {
            default: _(() => [
              (g(!0), A(
                de,
                null,
                ce(V.value, (P, H) => (g(), A(
                  "span",
                  {
                    key: `${P}-${H}`,
                    class: "m-picker-header__year-text"
                  },
                  me(P),
                  1
                  /* TEXT */
                ))),
                128
                /* KEYED_FRAGMENT */
              ))
            ]),
            _: 1
            /* STABLE */
          }, 8, ["name"]),
          R(Me, {
            name: u.value,
            tag: "div",
            class: "m-picker-header__date-transition"
          }, {
            default: _(() => [
              (g(!0), A(
                de,
                null,
                ce(m.value, (P, H) => (g(), A(
                  "span",
                  {
                    key: `${P}-${H}`,
                    class: "m-picker-header__date-text"
                  },
                  me(P),
                  1
                  /* TEXT */
                ))),
                128
                /* KEYED_FRAGMENT */
              ))
            ]),
            _: 1
            /* STABLE */
          }, 8, ["name"])
        ])) : W("v-if", !0),
        M.value ? (g(), A(
          "div",
          {
            key: 1,
            class: X(["m-picker-header__time", { "--has-date": t.hasDate }])
          },
          [
            R(Me, {
              name: u.value,
              tag: "div",
              class: "m-picker-header__time-transition"
            }, {
              default: _(() => [
                (g(!0), A(
                  de,
                  null,
                  ce(M.value, (P, H) => (g(), A(
                    "span",
                    {
                      key: `${P}-${H}`,
                      class: "m-picker-header__time-text"
                    },
                    me(P),
                    1
                    /* TEXT */
                  ))),
                  128
                  /* KEYED_FRAGMENT */
                ))
              ]),
              _: 1
              /* STABLE */
            }, 8, ["name"])
          ],
          2
          /* CLASS */
        )) : W("v-if", !0)
      ],
      2
      /* CLASS */
    ));
  }
}), It = /* @__PURE__ */ se(Ct, [["__scopeId", "data-v-53e90633"]]), Tt = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "1em",
  height: "1em",
  fill: "none",
  viewBox: "0 0 24 24"
}, qt = /* @__PURE__ */ ne(
  "path",
  {
    stroke: "currentColor",
    "stroke-linecap": "round",
    "stroke-linejoin": "round",
    "stroke-width": "1.5",
    d: "M15.75 19.5 8.25 12l7.5-7.5"
  },
  null,
  -1
  /* HOISTED */
), Ht = [
  qt
];
function Pt(t, r) {
  return g(), A("svg", Tt, [...Ht]);
}
const Ae = { render: Pt }, Lt = { class: "m-picker-calendar-switcher" }, At = /* @__PURE__ */ le({
  __name: "MazPickerCalendarSwitcher",
  props: {
    calendarDate: { type: String, default: void 0 },
    locale: { type: String, required: !0 },
    double: { type: Boolean, required: !0 }
  },
  emits: [
    "previous",
    "next",
    "open-month-switcher",
    "open-year-switcher",
    "update:calendar-date"
  ],
  setup(t, { emit: r }) {
    const e = t, u = r, d = y(() => D(e.calendarDate)), V = y(() => e.double ? `${De(
      te(d.value.format(), e.locale, { month: "long" })
    )} - ${De(
      te(d.value.add(1, "month").format(), e.locale, {
        month: "long"
      })
    )}` : De(te(d.value.format(), e.locale, { month: "long" }))), Y = y(
      () => te(d.value.format(), e.locale, { year: "numeric" })
    );
    function m() {
      u("update:calendar-date", D(e.calendarDate).subtract(1, "month").format());
    }
    function T() {
      u("update:calendar-date", D(e.calendarDate).add(1, "month").format());
    }
    return (M, b) => (g(), A("div", Lt, [
      R(pe, {
        size: "xs",
        color: "transparent",
        type: "button",
        onClick: m
      }, {
        default: _(() => [
          R(G(Ae), { class: "maz-text-lg" })
        ]),
        _: 1
        /* STABLE */
      }),
      R(pe, {
        size: "sm",
        color: "transparent",
        type: "button",
        class: "m-picker-calendar-switcher__date",
        onClick: b[0] || (b[0] = (C) => M.$emit("open-month-switcher", C))
      }, {
        default: _(() => [
          ze(
            me(V.value),
            1
            /* TEXT */
          )
        ]),
        _: 1
        /* STABLE */
      }),
      R(pe, {
        size: "sm",
        color: "transparent",
        type: "button",
        class: "m-picker-calendar-switcher__date",
        onClick: b[1] || (b[1] = (C) => M.$emit("open-year-switcher", C))
      }, {
        default: _(() => [
          ze(
            me(Y.value),
            1
            /* TEXT */
          )
        ]),
        _: 1
        /* STABLE */
      }),
      R(pe, {
        size: "xs",
        color: "transparent",
        type: "button",
        onClick: T
      }, {
        default: _(() => [
          R(G(Ae), { class: "maz-rotate-180 maz-text-lg" })
        ]),
        _: 1
        /* STABLE */
      })
    ]));
  }
}), jt = /* @__PURE__ */ se(At, [["__scopeId", "data-v-a672bb60"]]);
function _e(t, r) {
  let e;
  return function(...u) {
    clearTimeout(e), e = setTimeout(() => {
      t.apply(this, u);
    }, r);
  };
}
const Wt = /* @__PURE__ */ le({
  __name: "MazPickerCalendarGrid",
  props: {
    modelValue: {
      type: [String, Object],
      default: void 0
    },
    calendarDate: { type: String, required: !0 },
    hasTime: { type: Boolean, required: !0 },
    locale: { type: String, required: !0 },
    firstDayOfWeek: { type: Number, required: !0 },
    color: { type: String, required: !0 },
    minDate: { type: String, default: void 0 },
    maxDate: { type: String, default: void 0 },
    disabledWeekly: { type: Array, required: !0 },
    disabledDates: { type: Array, required: !0 },
    hoverredDay: { type: Object, default: void 0 },
    disabled: { type: Boolean, required: !0 }
  },
  emits: ["update:model-value", "update:hoverred-day"],
  setup(t, { emit: r }) {
    tt((n) => ({
      "25d899bc": T.value,
      "2ca33be4": b.value,
      "536b1d30": M.value
    }));
    const e = t, u = r, d = Q(), V = Q("maz-slidenext"), Y = y(() => [e.calendarDate]), m = y(() => e.modelValue && typeof e.modelValue == "object"), T = y(() => `var(--maz-color-${e.color}-alpha-20)`), M = y(() => `var(--maz-color-${e.color}-alpha)`), b = y(() => `var(--maz-color-${e.color}-alpha-20)`), C = y({
      get: () => e.modelValue,
      set: (n) => u("update:model-value", n)
    }), P = y(
      () => Array.from({ length: Dt(e.calendarDate) }, (n, o) => o + 1).map((n) => ({
        label: n,
        date: D(e.calendarDate).set("date", n)
      }))
    ), H = y(() => (pt(e.calendarDate) - e.firstDayOfWeek + 7) % 7);
    function j(n) {
      const o = e.modelValue;
      o.start && !o.end && n && n.isAfter(o.start) ? u("update:hoverred-day", n) : u("update:hoverred-day");
    }
    function S(n) {
      const o = e.modelValue;
      return !o.start || !e.hoverredDay ? void 0 : D(n).isBetween(o.start, e.hoverredDay, "date", "(]") ? 3 : void 0;
    }
    function v(n) {
      if (e.hoverredDay)
        return D(n).isSame(e.hoverredDay);
    }
    function B(n) {
      var o;
      return e.modelValue && e.modelValue && typeof e.modelValue == "object" && (o = e.modelValue) != null && o.start ? he(n, e.modelValue.start, "date") : !1;
    }
    function f(n) {
      var o;
      return e.modelValue && e.modelValue && typeof e.modelValue == "object" && (o = e.modelValue) != null && o.end ? he(n, e.modelValue.end, "date") : !1;
    }
    function F(n) {
      const o = e.modelValue;
      return typeof o == "object" ? o.start && he(n, o.start, "date") || o.end && he(n, o.end, "date") || I(n) ? e.color : "transparent" : z(n) ? e.color : "transparent";
    }
    function U(n) {
      if (e.modelValue && typeof e.modelValue == "object") {
        if (e.modelValue.start && he(n, e.modelValue.start, "date"))
          return 1;
        if (e.modelValue.end) {
          if (he(n, e.modelValue.end, "date"))
            return 1;
          if (I(n))
            return 2;
        }
      } else if (z(n))
        return 1;
      return 0;
    }
    function p(n) {
      m.value && j();
      const o = n.format();
      if (typeof C.value == "object") {
        let c = C.value;
        c.start && c.end && (c = {
          start: void 0,
          end: void 0
        });
        const s = D(o).isBefore(c.start, "date");
        C.value = !c.start || s ? {
          start: o,
          end: void 0
        } : {
          start: c.start,
          end: o
        };
      } else
        C.value = o;
    }
    function w(n) {
      return gt(n);
    }
    function z(n) {
      if (!e.modelValue)
        return !1;
      const o = e.modelValue;
      return he(n, o, "date");
    }
    function I(n) {
      const o = e.modelValue;
      return !o.start || !o.end ? !1 : D(n).isBetween(o.start, o.end, "date", "()");
    }
    function E(n) {
      return e.minDate ? D(n).isBefore(e.minDate, "date") : !1;
    }
    function L(n) {
      var o;
      return (o = e.disabledWeekly) != null && o.length ? e.disabledWeekly.some((c) => Ke(n, c)) : !1;
    }
    function O(n) {
      var o;
      return (o = e.disabledDates) != null && o.length ? e.disabledDates.some((c) => he(n, c, "date")) : !1;
    }
    function N(n) {
      return e.maxDate ? D(n).isAfter(e.maxDate, "date") : !1;
    }
    const l = _e(() => {
      d.value && (d.value.style.minHeight = "");
    }, 400);
    function h() {
      var n;
      d.value && (d.value.style.minHeight = `${((n = d.value) == null ? void 0 : n.clientHeight) || 176}px`, l());
    }
    return be(
      () => e.calendarDate,
      (n, o) => {
        V.value = D(n).isAfter(o, "date") ? "maz-slidenext" : "maz-slideprev", h();
      }
    ), (n, o) => (g(), A(
      "div",
      {
        ref_key: "MazPickerGrid",
        ref: d,
        class: "maz-picker-calendar-grid"
      },
      [
        R(Me, { name: V.value }, {
          default: _(() => [
            (g(!0), A(
              de,
              null,
              ce([Y.value], (c, s) => (g(), A(
                "div",
                {
                  key: `${c[s]}`,
                  class: X(["maz-picker-calendar-grid__container", { "--is-range": m.value }])
                },
                [
                  (g(!0), A(
                    de,
                    null,
                    ce(H.value, ($) => (g(), A("div", { key: $ }))),
                    128
                    /* KEYED_FRAGMENT */
                  )),
                  (g(!0), A(
                    de,
                    null,
                    ce(P.value, ({ label: $, date: a }, i) => (g(), x(pe, {
                      key: i,
                      size: "mini",
                      color: F(a),
                      type: "button",
                      disabled: t.disabled || E(a) || N(a) || L(a) || O(a),
                      class: X({
                        "--is-today": w(a),
                        "--is-first": B(a),
                        "--is-last": f(a) || m.value && v(a),
                        "--is-selected": U(a) === 1,
                        "--is-between": U(a) === 2,
                        "--is-between-hoverred": m.value ? S(a) === 3 : void 0
                      }),
                      onClick: (k) => p(a),
                      onMouseover: (k) => m.value ? j(a) : void 0,
                      onMouseleave: o[0] || (o[0] = (k) => m.value ? j() : void 0),
                      onFocus: (k) => m.value ? j(a) : void 0,
                      onBlur: o[1] || (o[1] = (k) => m.value ? j() : void 0)
                    }, {
                      default: _(() => [
                        ne(
                          "span",
                          null,
                          me($),
                          1
                          /* TEXT */
                        )
                      ]),
                      _: 2
                      /* DYNAMIC */
                    }, 1032, ["color", "disabled", "class", "onClick", "onMouseover", "onFocus"]))),
                    128
                    /* KEYED_FRAGMENT */
                  ))
                ],
                2
                /* CLASS */
              ))),
              128
              /* KEYED_FRAGMENT */
            ))
          ]),
          _: 1
          /* STABLE */
        }, 8, ["name"])
      ],
      512
      /* NEED_PATCH */
    ));
  }
}), Et = /* @__PURE__ */ se(Wt, [["__scopeId", "data-v-72921ea8"]]), Ut = { class: "maz-picker-calendar-days" }, Ft = /* @__PURE__ */ le({
  __name: "MazPickerCalendarDays",
  props: {
    locale: { type: String, required: !0 },
    firstDayOfWeek: { type: Number, required: !0 }
  },
  setup(t) {
    const r = t, e = y(() => bt(r.locale, r.firstDayOfWeek));
    return (u, d) => (g(), A("div", Ut, [
      (g(!0), A(
        de,
        null,
        ce(e.value, (V) => (g(), A(
          "span",
          { key: V },
          me(V),
          1
          /* TEXT */
        ))),
        128
        /* KEYED_FRAGMENT */
      ))
    ]));
  }
}), xt = /* @__PURE__ */ se(Ft, [["__scopeId", "data-v-4315c0cd"]]), Nt = { class: "maz-picker-calendar-month --has-padding" }, Zt = /* @__PURE__ */ le({
  __name: "MazPickerCalendarMonth",
  props: {
    modelValue: {
      type: [String, Object],
      default: void 0
    },
    color: { type: String, required: !0 },
    locale: { type: String, required: !0 },
    hasTime: { type: Boolean, required: !0 },
    firstDayOfWeek: { type: Number, required: !0 },
    calendarDate: { type: String, required: !0 },
    offsetMonth: { type: Number, default: 0 },
    minDate: { type: String, default: void 0 },
    maxDate: { type: String, default: void 0 },
    disabledWeekly: { type: Array, required: !0 },
    disabledDates: { type: Array, required: !0 },
    hoverredDay: { type: Object, default: void 0 },
    disabled: { type: Boolean, required: !0 }
  },
  emits: ["update:model-value", "update:calendar-date", "update:hoverred-day"],
  setup(t, { emit: r }) {
    const e = t, u = r, d = y({
      get: () => e.modelValue,
      set: (T) => u("update:model-value", T)
    }), V = y({
      get: () => e.hoverredDay,
      set: (T) => u("update:hoverred-day", T)
    });
    function Y(T) {
      return D(e.calendarDate).add(T, "month").format();
    }
    const m = y({
      get: () => Y(e.offsetMonth),
      set: (T) => u("update:calendar-date", T)
    });
    return (T, M) => (g(), A("div", Nt, [
      R(xt, {
        locale: t.locale,
        "first-day-of-week": t.firstDayOfWeek,
        class: "maz-picker-calendar-month__days"
      }, null, 8, ["locale", "first-day-of-week"]),
      R(Et, {
        modelValue: d.value,
        "onUpdate:modelValue": M[0] || (M[0] = (b) => d.value = b),
        "hoverred-day": V.value,
        "onUpdate:hoverredDay": M[1] || (M[1] = (b) => V.value = b),
        locale: t.locale,
        color: t.color,
        "has-time": t.hasTime,
        "calendar-date": m.value,
        "first-day-of-week": t.firstDayOfWeek,
        "min-date": t.minDate,
        "max-date": t.maxDate,
        "disabled-weekly": t.disabledWeekly,
        "disabled-dates": t.disabledDates,
        disabled: t.disabled
      }, null, 8, ["modelValue", "hoverred-day", "locale", "color", "has-time", "calendar-date", "first-day-of-week", "min-date", "max-date", "disabled-weekly", "disabled-dates", "disabled"])
    ]));
  }
}), Rt = /* @__PURE__ */ se(Zt, [["__scopeId", "data-v-ad901a82"]]), Gt = { class: "maz-picker-calendar flex" }, Jt = { class: "maz-picker-calendar__months" }, Qt = /* @__PURE__ */ le({
  __name: "MazPickerCalendar",
  props: {
    modelValue: {
      type: [String, Object],
      default: void 0
    },
    calendarDate: { type: String, required: !0 },
    color: { type: String, required: !0 },
    locale: { type: String, required: !0 },
    firstDayOfWeek: { type: Number, required: !0 },
    double: { type: Boolean, required: !0 },
    minDate: { type: String, default: void 0 },
    maxDate: { type: String, default: void 0 },
    disabledWeekly: { type: Array, required: !0 },
    disabledDates: { type: Array, required: !0 },
    shortcuts: {
      type: Array,
      required: !0
    },
    noShortcuts: { type: Boolean, required: !0 },
    hasTime: { type: Boolean, required: !0 },
    shortcut: { type: String, default: void 0 },
    disabled: { type: Boolean, required: !0 }
  },
  emits: ["update:model-value", "update:calendar-date"],
  setup(t, { emit: r }) {
    const e = t, u = r, d = ue(() => import("./MazPickerMonthSwitcher-DZPHWTJt.mjs")), V = ue(() => import("./MazPickerYearSwitcher-Ynp2qf3F.mjs")), Y = ue(() => import("./MazPickerShortcuts-CBEmYjPG.mjs")), m = Q(), T = y(() => e.modelValue && typeof e.modelValue == "object"), M = y(
      () => !e.noShortcuts && e.shortcuts.length > 0 && T.value
    ), b = Q(!1), C = Q(!1), P = y({
      get: () => e.modelValue,
      set: (S) => u("update:model-value", S)
    }), H = y(() => Array.from({ length: e.double ? 2 : 1 }, (S, v) => v)), j = y({
      get: () => e.calendarDate,
      set: (S) => u("update:calendar-date", S)
    });
    return (S, v) => (g(), A("div", Gt, [
      M.value ? (g(), x(G(Y), {
        key: 0,
        modelValue: P.value,
        "onUpdate:modelValue": v[0] || (v[0] = (B) => P.value = B),
        color: t.color,
        shortcuts: t.shortcuts,
        shortcut: t.shortcut,
        double: t.double,
        disabled: t.disabled
      }, null, 8, ["modelValue", "color", "shortcuts", "shortcut", "double", "disabled"])) : W("v-if", !0),
      ne(
        "div",
        {
          class: X(["maz-picker-calendar__main", { "--has-double": t.double }])
        },
        [
          R(jt, {
            "calendar-date": j.value,
            "onUpdate:calendarDate": v[1] || (v[1] = (B) => j.value = B),
            locale: t.locale,
            double: t.double,
            onOpenMonthSwitcher: v[2] || (v[2] = (B) => b.value = !0),
            onOpenYearSwitcher: v[3] || (v[3] = (B) => C.value = !0)
          }, null, 8, ["calendar-date", "locale", "double"]),
          R(Te, { name: "maz-picker-slide" }, {
            default: _(() => [
              b.value ? (g(), x(G(d), {
                key: 0,
                "calendar-date": j.value,
                "onUpdate:calendarDate": v[4] || (v[4] = (B) => j.value = B),
                color: t.color,
                double: t.double,
                locale: t.locale,
                onClose: v[5] || (v[5] = (B) => b.value = !1)
              }, null, 8, ["calendar-date", "color", "double", "locale"])) : W("v-if", !0)
            ]),
            _: 1
            /* STABLE */
          }),
          R(Te, { name: "maz-picker-slide" }, {
            default: _(() => [
              C.value ? (g(), x(G(V), {
                key: 0,
                "calendar-date": j.value,
                "onUpdate:calendarDate": v[6] || (v[6] = (B) => j.value = B),
                color: t.color,
                locale: t.locale,
                onClose: v[7] || (v[7] = (B) => C.value = !1)
              }, null, 8, ["calendar-date", "color", "locale"])) : W("v-if", !0)
            ]),
            _: 1
            /* STABLE */
          }),
          ne("div", Jt, [
            (g(!0), A(
              de,
              null,
              ce(H.value, (B) => (g(), x(Rt, {
                key: B,
                modelValue: P.value,
                "onUpdate:modelValue": v[8] || (v[8] = (f) => P.value = f),
                "hoverred-day": m.value,
                "onUpdate:hoverredDay": v[9] || (v[9] = (f) => m.value = f),
                "calendar-date": j.value,
                locale: t.locale,
                "has-time": t.hasTime,
                color: t.color,
                "offset-month": B,
                "first-day-of-week": t.firstDayOfWeek,
                "min-date": t.minDate,
                "max-date": t.maxDate,
                disabled: t.disabled,
                "disabled-weekly": t.disabledWeekly,
                "disabled-dates": t.disabledDates
              }, null, 8, ["modelValue", "hoverred-day", "calendar-date", "locale", "has-time", "color", "offset-month", "first-day-of-week", "min-date", "max-date", "disabled", "disabled-weekly", "disabled-dates"]))),
              128
              /* KEYED_FRAGMENT */
            ))
          ])
        ],
        2
        /* CLASS */
      )
    ]));
  }
}), Xt = /* @__PURE__ */ se(Qt, [["__scopeId", "data-v-df2822d9"]]), Kt = { class: "m-picker-container__wrapper" }, _t = /* @__PURE__ */ le({
  __name: "MazPickerContainer",
  props: {
    modelValue: {
      type: [String, Object],
      default: void 0
    },
    calendarDate: { type: String, required: !0 },
    color: { type: String, required: !0 },
    locale: { type: String, required: !0 },
    noHeader: { type: Boolean, default: !1 },
    firstDayOfWeek: { type: Number, required: !0 },
    double: { type: Boolean, required: !0 },
    hasDate: { type: Boolean, required: !0 },
    minDate: { type: String, default: void 0 },
    maxDate: { type: String, default: void 0 },
    inline: { type: Boolean, required: !0 },
    noShortcuts: { type: Boolean, required: !0 },
    shortcuts: {
      type: Array,
      required: !0
    },
    shortcut: { type: String, default: void 0 },
    hasTime: { type: Boolean, required: !0 },
    isOpen: { type: Boolean, required: !0 },
    format: { type: String, required: !0 },
    isHour12: { type: Boolean, required: !0 },
    formatterOptions: {
      type: Object,
      required: !0
    },
    minuteInterval: { type: Number, required: !0 },
    disabled: { type: Boolean, required: !0 },
    disabledWeekly: { type: Array, required: !0 },
    disabledHours: { type: Array, required: !0 },
    disabledDates: { type: Array, required: !0 }
  },
  emits: ["update:model-value", "update:calendar-date", "close"],
  setup(t, { emit: r }) {
    const e = t, u = r, d = y({
      get: () => e.modelValue,
      set: (Y) => {
        u("update:model-value", Y);
      }
    }), V = y({
      get: () => e.calendarDate,
      set: (Y) => u("update:calendar-date", Y)
    });
    return (Y, m) => (g(), A(
      "div",
      {
        class: X(["m-picker-container", {
          "--has-double": t.double,
          "--is-inline": t.inline,
          "--has-date": t.hasDate
        }])
      },
      [
        t.noHeader ? W("v-if", !0) : (g(), x(It, {
          key: 0,
          color: t.color,
          "has-time": t.hasTime,
          "model-value": t.modelValue,
          locale: t.locale,
          "calendar-date": V.value,
          "has-date": t.hasDate,
          "formatter-options": t.formatterOptions,
          double: t.double,
          "no-shortcuts": t.noShortcuts,
          class: "m-picker-container__header"
        }, null, 8, ["color", "has-time", "model-value", "locale", "calendar-date", "has-date", "formatter-options", "double", "no-shortcuts"])),
        ne("div", Kt, [
          t.hasDate ? (g(), x(Xt, {
            key: 0,
            modelValue: d.value,
            "onUpdate:modelValue": m[0] || (m[0] = (T) => d.value = T),
            "calendar-date": V.value,
            "onUpdate:calendarDate": m[1] || (m[1] = (T) => V.value = T),
            color: t.color,
            locale: t.locale,
            "has-time": t.hasTime,
            double: t.double,
            "min-date": t.minDate,
            "max-date": t.maxDate,
            "first-day-of-week": t.firstDayOfWeek,
            disabled: t.disabled,
            "disabled-weekly": t.disabledWeekly,
            "disabled-dates": t.disabledDates,
            shortcuts: t.shortcuts,
            shortcut: t.shortcut,
            "no-shortcuts": t.noShortcuts,
            class: "m-picker-container__calendar"
          }, null, 8, ["modelValue", "calendar-date", "color", "locale", "has-time", "double", "min-date", "max-date", "first-day-of-week", "disabled", "disabled-weekly", "disabled-dates", "shortcuts", "shortcut", "no-shortcuts"])) : W("v-if", !0),
          t.hasTime ? (g(), x(Ot, {
            key: 1,
            modelValue: d.value,
            "onUpdate:modelValue": m[2] || (m[2] = (T) => d.value = T),
            "calendar-date": V.value,
            "onUpdate:calendarDate": m[3] || (m[3] = (T) => V.value = T),
            "is-open": t.isOpen,
            color: t.color,
            locale: t.locale,
            "min-date": t.minDate,
            "max-date": t.maxDate,
            "has-date": t.hasDate,
            format: t.format,
            disabled: t.disabled,
            "disabled-hours": t.disabledHours,
            "minute-interval": t.minuteInterval,
            "formatter-options": t.formatterOptions,
            "is-hour12": t.isHour12,
            class: "m-picker-container__time"
          }, null, 8, ["modelValue", "calendar-date", "is-open", "color", "locale", "min-date", "max-date", "has-date", "format", "disabled", "disabled-hours", "minute-interval", "formatter-options", "is-hour12"])) : W("v-if", !0)
        ])
      ],
      2
      /* CLASS */
    ));
  }
}), ea = /* @__PURE__ */ se(_t, [["__scopeId", "data-v-5ddb3f1f"]]), ta = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "1em",
  height: "1em",
  fill: "none",
  viewBox: "0 0 24 24"
}, aa = /* @__PURE__ */ ne(
  "path",
  {
    stroke: "currentColor",
    "stroke-linecap": "round",
    "stroke-linejoin": "round",
    "stroke-width": "1.5",
    d: "m19.5 8.25-7.5 7.5-7.5-7.5"
  },
  null,
  -1
  /* HOISTED */
), ra = [
  aa
];
function na(t, r) {
  return g(), A("svg", ta, [...ra]);
}
const oa = { render: na };
function ia({
  componentName: t,
  providedId: r
}) {
  const e = Ue();
  return y(() => r ?? `${t}-${e == null ? void 0 : e.uid}`);
}
const la = {
  key: 0,
  class: "m-input-wrapper-left"
}, sa = { class: "m-input-wrapper-input" }, ua = ["id", "type", "name", "inputmode", "placeholder", "aria-label", "disabled", "readonly", "required"], da = { key: 0 }, ca = {
  key: 1,
  class: "m-input-wrapper-right"
}, fa = /* @__PURE__ */ le({
  inheritAttrs: !1,
  __name: "MazInput",
  props: {
    style: { default: void 0 },
    class: { default: void 0 },
    modelValue: { default: void 0 },
    placeholder: { default: void 0 },
    label: { default: void 0 },
    name: { default: void 0 },
    color: { default: "primary" },
    type: { default: "text" },
    required: { type: Boolean, default: !1 },
    disabled: { type: Boolean, default: !1 },
    readonly: { type: Boolean, default: !1 },
    id: { default: void 0 },
    error: { type: Boolean, default: !1 },
    success: { type: Boolean, default: !1 },
    warning: { type: Boolean, default: !1 },
    hint: { default: void 0 },
    inputClasses: { default: void 0 },
    noBorder: { type: Boolean, default: !1 },
    noRadius: { type: Boolean, default: !1 },
    inputmode: { default: "text" },
    size: { default: "md" },
    debounce: { type: [Boolean, Number], default: !1 },
    debounceDelay: { default: 500 },
    validButton: { type: Boolean, default: !1 },
    validButtonLoading: { type: Boolean, default: !1 },
    autoFocus: { type: Boolean, default: !1 },
    borderActive: { type: Boolean, default: !1 },
    leftIcon: { type: [String, Function, Object], default: void 0 },
    rightIcon: { type: [String, Function, Object], default: void 0 },
    roundedSize: { default: void 0 },
    block: { type: Boolean }
  },
  emits: ["update:model-value", "focus", "blur", "click", "change"],
  setup(t, { emit: r }) {
    const e = t, u = r, d = ue(() => Promise.resolve().then(() => zt)), V = ue(() => import("./MazIcon-BDGxCy9M.mjs")), Y = ue(() => import("./eye-slash-tjfxO0LK.mjs")), m = ue(() => import("./eye-DqO_Jx56.mjs")), T = ue(() => import("./check-CL-vqe79.mjs")), M = Q(!1), b = Q(!1), C = Q(), P = ia({
      componentName: "MazInput",
      providedId: e.id
    });
    Fe(() => {
      var l;
      e.autoFocus && ((l = C.value) == null || l.focus());
    });
    const H = y(() => e.type === "password"), j = y(() => M.value ? "text" : e.type), S = y(() => {
      if (!e.noBorder) {
        if (e.error)
          return "maz-border-danger";
        if (e.success)
          return "maz-border-success";
        if (e.warning)
          return "maz-border-warning";
        if (b.value || e.borderActive) {
          if (e.color === "black")
            return "maz-border-black";
          if (e.color === "danger")
            return "maz-border-danger";
          if (e.color === "info")
            return "maz-border-info";
          if (e.color === "primary")
            return "maz-border-primary";
          if (e.color === "secondary")
            return "maz-border-secondary";
          if (e.color === "success")
            return "maz-border-success";
          if (e.color === "warning")
            return "maz-border-warning";
          if (e.color === "white")
            return "maz-border-white";
        }
        return "--default-border";
      }
    }), v = je(), B = y(() => {
      const { required: l, placeholder: h } = e;
      if (h)
        return l ? `${h} *` : h;
    }), f = y(() => p.value !== void 0 && p.value !== ""), F = _e(
      (l) => {
        u("update:model-value", l);
      },
      typeof e.debounce == "number" ? e.debounce : e.debounceDelay ?? 500
    );
    function U(l) {
      if (e.debounce)
        return F(l);
      u("update:model-value", l);
    }
    const p = y({
      get: () => e.modelValue,
      set: (l) => U(l)
    }), w = y(() => (!!e.label || !!e.hint) && (b.value || !!f.value || !!e.placeholder || ["date", "month", "week"].includes(e.type))), z = y(() => !!e.label || !!e.hint);
    function I() {
      return !!v["right-icon"] || H.value || !!v["valid-button"] || e.validButton || !!e.rightIcon;
    }
    function E() {
      return !!v["left-icon"] || !!e.leftIcon;
    }
    function L(l) {
      u("focus", l), b.value = !0;
    }
    function O(l) {
      u("blur", l), b.value = !1;
    }
    function N(l) {
      return u("change", l);
    }
    return (l, h) => (g(), A(
      "div",
      {
        class: X(["m-input", [
          {
            "--is-focused": b.value || l.borderActive,
            "--should-up": w.value,
            "--has-label": z.value,
            "--is-disabled": l.disabled,
            "--is-readonly": l.readonly,
            "--has-z-2": l.error || l.warning || l.success,
            "--has-state": l.error || l.warning || l.success,
            "--block": l.block
          },
          e.class,
          `--${l.color}`,
          `--${l.size}`
        ]]),
        style: Se(l.style)
      },
      [
        ne(
          "div",
          {
            class: X(["m-input-wrapper", [
              l.inputClasses,
              S.value,
              l.roundedSize ? `--rounded-${l.roundedSize}` : { "maz-rounded": !l.noRadius }
            ]])
          },
          [
            E() ? (g(), A("div", la, [
              W(`
          @slot left-icon - The icon to display on the left of the input
        `),
              l.$slots["left-icon"] || l.leftIcon ? ye(l.$slots, "left-icon", { key: 0 }, () => [
                typeof l.leftIcon == "string" ? (g(), x(G(V), {
                  key: 0,
                  name: l.leftIcon,
                  class: "maz-text-xl maz-text-muted"
                }, null, 8, ["name"])) : l.leftIcon ? (g(), x(ke(l.leftIcon), {
                  key: 1,
                  class: "maz-text-xl maz-text-muted"
                })) : W("v-if", !0)
              ], !0) : W("v-if", !0)
            ])) : W("v-if", !0),
            ne("div", sa, [
              qe(ne("input", xe({
                id: G(P),
                ref_key: "input",
                ref: C,
                "onUpdate:modelValue": h[0] || (h[0] = (n) => p.value = n),
                type: j.value,
                name: l.name
              }, l.$attrs, {
                inputmode: l.inputmode,
                placeholder: B.value,
                "aria-label": l.label || l.placeholder,
                disabled: l.disabled,
                readonly: l.readonly,
                required: l.required,
                class: "m-input-input"
              }, at({
                blur: O,
                focus: L,
                change: N
              }, !0), {
                onClick: h[1] || (h[1] = (n) => l.$emit("click", n))
              }), null, 16, ua), [
                [rt, p.value]
              ]),
              l.label || l.hint ? (g(), A(
                "span",
                {
                  key: 0,
                  class: X(["m-input-label", [
                    {
                      "maz-text-danger-600": l.error,
                      "maz-text-success-600": l.success,
                      "maz-text-warning-600": l.warning
                    }
                  ]])
                },
                [
                  ze(
                    me(l.hint || l.label) + " ",
                    1
                    /* TEXT */
                  ),
                  l.required ? (g(), A("sup", da, "*")) : W("v-if", !0)
                ],
                2
                /* CLASS */
              )) : W("v-if", !0)
            ]),
            I() ? (g(), A("div", ca, [
              W(`
          @slot right-icon - The icon to display on the right of the input
        `),
              l.$slots["right-icon"] || l.rightIcon ? ye(l.$slots, "right-icon", { key: 0 }, () => [
                typeof l.rightIcon == "string" ? (g(), x(G(V), {
                  key: 0,
                  name: l.rightIcon,
                  class: "maz-text-xl maz-text-muted"
                }, null, 8, ["name"])) : l.rightIcon ? (g(), x(ke(l.rightIcon), {
                  key: 1,
                  class: "maz-text-xl maz-text-muted"
                })) : W("v-if", !0)
              ], !0) : W("v-if", !0),
              H.value ? (g(), x(G(d), {
                key: 1,
                color: "transparent",
                tabindex: "-1",
                size: "mini",
                onClick: h[2] || (h[2] = Ee((n) => M.value = !M.value, ["stop"]))
              }, {
                default: _(() => [
                  M.value ? (g(), x(G(Y), {
                    key: 0,
                    class: "maz-text-xl maz-text-muted"
                  })) : (g(), x(G(m), {
                    key: 1,
                    class: "maz-text-xl maz-text-muted"
                  }))
                ]),
                _: 1
                /* STABLE */
              })) : W("v-if", !0),
              W(`
          @slot valid-button - Replace the valid button by your own
        `),
              l.$slots["valid-button"] || l.validButton ? ye(l.$slots, "valid-button", { key: 2 }, () => [
                R(G(d), {
                  color: "transparent",
                  disabled: l.disabled,
                  tabindex: "-1",
                  loading: l.validButtonLoading,
                  class: "m-input-valid-button",
                  size: "mini",
                  type: "submit"
                }, {
                  default: _(() => [
                    R(G(T), { class: "maz-text-2xl maz-text-normal" })
                  ]),
                  _: 1
                  /* STABLE */
                }, 8, ["disabled", "loading"])
              ], !0) : W("v-if", !0)
            ])) : W("v-if", !0)
          ],
          2
          /* CLASS */
        )
      ],
      6
      /* CLASS, STYLE */
    ));
  }
}), ma = /* @__PURE__ */ se(fa, [["__scopeId", "data-v-e9affc48"]]), va = /* @__PURE__ */ le({
  inheritAttrs: !1,
  __name: "MazPicker",
  props: {
    style: { default: void 0 },
    class: { default: void 0 },
    modelValue: { default: void 0 },
    format: { default: "YYYY-MM-DD" },
    open: { type: Boolean, default: !1 },
    label: { default: void 0 },
    placeholder: { default: void 0 },
    inputDateStyle: { default: () => ({
      dateStyle: "full"
    }) },
    locale: { default: void 0 },
    noHeader: { type: Boolean, default: !1 },
    disabled: { type: Boolean, default: !1 },
    firstDayOfWeek: { default: 0 },
    autoClose: { type: Boolean, default: !1 },
    customElementSelector: { default: void 0 },
    double: { type: Boolean, default: !1 },
    inline: { type: Boolean, default: !1 },
    color: { default: "primary" },
    pickerPosition: { default: void 0 },
    time: { type: Boolean, default: !1 },
    onlyTime: { type: Boolean, default: !1 },
    minuteInterval: { default: 5 },
    noUseBrowserLocale: { type: Boolean, default: !1 },
    noFetchLocal: { type: Boolean, default: !1 },
    noShortcuts: { type: Boolean, default: !1 },
    shortcuts: { default: () => [
      {
        label: "Last 7 days",
        identifier: "last7Days",
        value: {
          start: D().subtract(6, "day").format("YYYY-MM-DD"),
          end: D().format("YYYY-MM-DD")
        }
      },
      {
        label: "Last 30 days",
        identifier: "last30Days",
        value: {
          start: D().subtract(29, "day").format("YYYY-MM-DD"),
          end: D().format("YYYY-MM-DD")
        }
      },
      {
        label: "This week",
        identifier: "thisWeek",
        value: {
          start: D().startOf("week").format("YYYY-MM-DD"),
          end: D().endOf("week").format("YYYY-MM-DD")
        }
      },
      {
        label: "Last week",
        identifier: "lastWeek",
        value: {
          start: D().subtract(1, "week").startOf("week").format("YYYY-MM-DD"),
          end: D().subtract(1, "week").endOf("week").format("YYYY-MM-DD")
        }
      },
      {
        label: "This month",
        identifier: "thisMonth",
        value: {
          start: D().set("date", 1).format("YYYY-MM-DD"),
          end: D().set("date", D().daysInMonth()).format("YYYY-MM-DD")
        }
      },
      {
        label: "This year",
        identifier: "thisYear",
        value: {
          start: D().startOf("year").format("YYYY-MM-DD"),
          end: D().endOf("year").format("YYYY-MM-DD")
        }
      },
      {
        label: "Last year",
        identifier: "lastYear",
        value: {
          start: D().subtract(1, "year").startOf("year").format("YYYY-MM-DD"),
          end: D().subtract(1, "year").endOf("year").format("YYYY-MM-DD")
        }
      }
    ] },
    shortcut: { default: void 0 },
    minDate: { default: void 0 },
    maxDate: { default: void 0 },
    disabledWeekly: { default: () => [] },
    disabledDates: { default: () => [] },
    disabledHours: { default: () => [] },
    block: { type: Boolean }
  },
  emits: ["update:model-value", "close"],
  setup(t, { emit: r }) {
    const e = t, u = r;
    D.extend(st), D.extend(dt);
    const d = {
      dateStyle: "full"
    }, V = Ue(), Y = Q(e.locale || "en-US"), m = y(() => e.locale ?? Y.value), T = y(() => `mazPickerContainer-${V == null ? void 0 : V.uid}`), M = Q(), b = y({
      get: () => e.modelValue && typeof e.modelValue == "object" ? {
        start: e.modelValue.start ? D(e.modelValue.start, e.format).format() : void 0,
        end: e.modelValue.end ? D(e.modelValue.end, e.format).format() : void 0
      } : e.modelValue ? D(e.modelValue, e.format).format() : void 0,
      set: (a) => {
        if (e.disabled)
          return;
        s(a);
        const i = typeof a == "object";
        e.autoClose && (!i || i && a.end) && O();
      }
    }), C = y(() => e.time || e.onlyTime), P = y(() => e.double && !e.onlyTime), H = y(() => !e.onlyTime), j = y(() => typeof b.value == "object");
    We(() => {
      j.value && C.value && console.error("[maz-ui](MazPicker) You can't use time picker with range picker"), C.value && !(e.format.includes("h") || e.format.includes("H")) && console.error(
        '[maz-ui](MazPicker) When you use the time picker, you must provided a format with time - Ex: "YYYY-MM-DD HH:mm"'
      ), e.format.includes("h") && !(e.format.includes("a") || e.format.includes("A")) && console.error(
        '[maz-ui](MazPicker) if you use the 12 format "h" or "hh", you must add "a" or "A" at the end of the format - Ex: "YYYY-MM-DD hh:mm a"'
      );
    });
    function S(a) {
      const i = (typeof a == "object" ? a.start : a) ?? D().format();
      return e.minDate && D(i).isBefore(e.minDate) ? e.minDate : e.maxDate && D(i).isAfter(e.maxDate) ? e.minDate ?? e.maxDate : i;
    }
    const v = Q(S(b.value)), B = y(
      () => e.format.includes("a") || e.format.includes("A") || e.format.includes("h")
    ), f = y(() => ({
      ...d,
      ...e.inputDateStyle,
      timeStyle: e.inputDateStyle.timeStyle ?? C.value ? "short" : void 0,
      hour12: e.inputDateStyle.hour12 ?? B.value
    })), F = y(() => {
      if (b.value)
        return e.onlyTime ? b.value ? te(D(b.value).format(), m.value, {
          timeStyle: f.value.timeStyle,
          hour12: f.value.hour12
        }) : void 0 : typeof b.value == "object" ? yt({
          value: b.value,
          locale: m.value,
          options: f.value
        }) : He({
          value: D(b.value).format(),
          locale: m.value,
          options: f.value
        });
    }), U = Q(!1), p = Q(!1), w = Q({
      vertical: "bottom",
      horizontal: "left"
    }), z = y(() => (U.value || e.open || p.value) && !e.disabled || e.inline), I = Q(!1);
    Fe(async () => {
      if (I.value = !0, e.customElementSelector && h(e.customElementSelector), !e.locale) {
        const a = $t();
        if (!e.noUseBrowserLocale && a)
          Y.value = a;
        else if (!e.noFetchLocal) {
          const i = await Mt();
          i && (Y.value = i);
        }
      }
    }), nt(() => {
      e.customElementSelector && n(e.customElementSelector);
    });
    async function E() {
      if (e.pickerPosition) {
        const a = e.pickerPosition.includes("right") ? "right" : "left", i = e.pickerPosition.includes("top") ? "top" : "bottom";
        return {
          horizontal: a,
          vertical: i
        };
      } else
        return {
          horizontal: "left",
          vertical: await L(M.value)
        };
    }
    async function L(a) {
      if (typeof window > "u")
        return "bottom";
      const i = 30;
      await ge();
      const k = document.querySelector(`#${T.value}`), q = a == null ? void 0 : a.getBoundingClientRect(), Z = window.innerHeight, K = ((k == null ? void 0 : k.clientHeight) ?? 0) - i, oe = (q && Z - q.bottom) ?? 0, ee = (q && q.top) ?? 0, ie = oe && oe >= K, ae = ee && ee >= K;
      return !ie && (ae || ee >= oe) ? "top" : "bottom";
    }
    function O() {
      U.value = !1, p.value = !1, u("close");
    }
    function N() {
      e.customElementSelector || O();
    }
    function l() {
      p.value = !p.value;
    }
    function h(a) {
      const i = document.querySelector(a);
      if (i)
        i.addEventListener("click", l);
      else
        throw new Error(
          `[maz-ui](MazPicker) impossible to find custom element with selector "${a}"`
        );
    }
    function n(a) {
      const i = document.querySelector(a);
      i == null || i.removeEventListener("click", l);
    }
    function o(a) {
      if (e.minDate || e.maxDate) {
        if (typeof a == "string") {
          const { newValue: i, newCurrentDate: k } = Ie({
            value: a,
            minDate: e.minDate,
            maxDate: e.maxDate,
            format: e.format
          });
          i && s(i), k && c(k);
        } else if (typeof a == "object" && (a.start || a.end)) {
          let i = a.start, k = a.end;
          if (a.start) {
            const { newValue: q, newCurrentDate: Z } = Ie({
              value: a.start,
              minDate: e.minDate,
              maxDate: e.maxDate,
              format: e.format
            });
            q && (i = q), Z && c(Z);
          }
          if (a.end) {
            const { newValue: q } = Ie({
              value: a.end,
              minDate: e.minDate,
              maxDate: e.maxDate,
              format: e.format
            });
            q && (k = q);
          }
          s({
            start: i,
            end: k
          });
        }
      }
    }
    function c(a) {
      a && !D(v.value).isSame(a, "month") && (v.value = a);
    }
    function s(a) {
      if (typeof a == "object") {
        const i = kt(a, e.format);
        u("update:model-value", i), i.start && c(i.start);
      } else
        u("update:model-value", Pe(a, e.format));
    }
    be(
      () => [b.value, e.minDate, e.maxDate],
      (a, i) => {
        const k = a[0], q = i == null ? void 0 : i[0];
        typeof k == "object" && (k.start || k.end) ? (!q || typeof q == "object" && (q.start !== k.start || q.end !== k.end)) && (s(k), o(k)) : typeof k == "string" && k !== q && (s(k), o(k));
      },
      { immediate: !0 }
    );
    function $(a) {
      a.code === "Escape" && z.value && (a.preventDefault(), O());
    }
    return be(
      () => z.value,
      async (a) => {
        a ? (w.value = await E(), !e.inline && I.value && document.addEventListener("keydown", $)) : !e.inline && I.value && document.removeEventListener("keydown", $);
      },
      { immediate: !0 }
    ), be(
      () => [b.value, e.disabledWeekly, e.disabledDates],
      (a) => {
        const i = a[0], k = a[1], q = a[2];
        typeof i == "object" && (i.start || i.end) ? ((i.start && we({ value: i.start, disabledWeekly: k }) || i.start && $e({ value: i.start, disabledDates: q }) || i.end && we({ value: i.end, disabledWeekly: k }) || i.end && $e({ value: i.end, disabledDates: q })) && (b.value = { start: void 0, end: void 0 }), (i.end && we({ value: i.end, disabledWeekly: k }) || i.end && $e({ value: i.end, disabledDates: q })) && (b.value = { start: i.start, end: void 0 })) : typeof i == "string" && (we({ value: i, disabledWeekly: k }) || $e({ value: i, disabledDates: q })) && (b.value = void 0);
      },
      { immediate: !0 }
    ), (a, i) => qe((g(), A(
      "div",
      {
        ref_key: "MazPicker",
        ref: M,
        class: X(["m-picker", [
          `m-picker--${a.color}`,
          `m-picker--${w.value.vertical}`,
          `m-picker--${w.value.horizontal}`,
          {
            "--is-open": z.value,
            "--is-disabled": a.disabled,
            "--block": a.block
          },
          e.class
        ]]),
        role: "none",
        style: Se(a.style)
      },
      [
        !a.customElementSelector && !a.inline ? (g(), x(ma, xe({
          key: 0,
          "model-value": F.value,
          readonly: ""
        }, a.$attrs, {
          block: "",
          autocomplete: "off",
          class: "m-picker__input",
          label: a.label,
          disabled: a.disabled,
          placeholder: a.placeholder,
          color: a.color,
          onClick: i[1] || (i[1] = (k) => U.value = !U.value)
        }), {
          "right-icon": _(() => [
            ne("button", {
              type: "button",
              tabindex: "-1",
              class: "m-picker__button",
              onClick: i[0] || (i[0] = (k) => U.value = !U.value)
            }, [
              R(G(oa), { class: "m-picker__button__chevron maz-text-lg" })
            ])
          ]),
          _: 1
          /* STABLE */
        }, 16, ["model-value", "label", "disabled", "placeholder", "color"])) : W("v-if", !0),
        R(Te, {
          name: w.value.vertical === "top" ? "maz-slideinvert" : "maz-slide",
          persisted: ""
        }, {
          default: _(() => [
            qe(R(ea, {
              id: T.value,
              modelValue: b.value,
              "onUpdate:modelValue": i[2] || (i[2] = (k) => b.value = k),
              "calendar-date": v.value,
              "onUpdate:calendarDate": i[3] || (i[3] = (k) => v.value = k),
              "is-open": z.value,
              color: a.color,
              locale: m.value,
              "has-date": H.value,
              double: P.value,
              "has-time": C.value,
              "formatter-options": f.value,
              "no-header": a.noHeader,
              "min-date": a.minDate,
              format: a.format,
              "is-hour12": B.value,
              "max-date": a.maxDate,
              "disabled-weekly": a.disabledWeekly,
              inline: a.inline,
              "first-day-of-week": a.firstDayOfWeek,
              shortcuts: a.shortcuts,
              shortcut: a.shortcut,
              disabled: a.disabled,
              "disabled-hours": a.disabledHours,
              "disabled-dates": a.disabledDates,
              "minute-interval": a.minuteInterval,
              "no-shortcuts": a.noShortcuts,
              onClose: O
            }, null, 8, ["id", "modelValue", "calendar-date", "is-open", "color", "locale", "has-date", "double", "has-time", "formatter-options", "no-header", "min-date", "format", "is-hour12", "max-date", "disabled-weekly", "inline", "first-day-of-week", "shortcuts", "shortcut", "disabled", "disabled-hours", "disabled-dates", "minute-interval", "no-shortcuts"]), [
              [ot, z.value]
            ])
          ]),
          _: 1
          /* STABLE */
        }, 8, ["name"])
      ],
      6
      /* CLASS, STYLE */
    )), [
      [G(ft), N]
    ]);
  }
}), pa = /* @__PURE__ */ se(va, [["__scopeId", "data-v-19f02535"]]);
export {
  Ae as C,
  pe as M,
  se as _,
  te as a,
  pa as b,
  De as c,
  D as d,
  he as i
};
