// maz-ui/components/chunks/MazStepper-DxrzKz06.mjs
import { defineComponent as J, useCssVars as K, defineAsyncComponent as f, computed as h, useSlots as Q, ref as U, openBlock as a, createElementBlock as c, Fragment as b, renderList as X, normalizeClass as M, createElementVNode as u, createBlock as w, resolveDynamicComponent as A, createCommentVNode as l, renderSlot as v, unref as P, createTextVNode as y, toDisplayString as g, createVNode as Y, withCtx as Z } from "vue";
import '../assets/MazStepper.css';const ee = { class: "m-stepper" }, te = ["id", "disabled", "onClick"], ne = { class: "m-stepper__header__wrapper" }, oe = { class: "m-stepper__count --primary" }, se = { class: "m-stepper__count__circle" }, ae = { class: "m-stepper__header__content" }, le = { class: "m-stepper__title" }, ie = {
  key: 0,
  class: "m-stepper__subtitle"
}, re = {
  key: 0,
  class: "m-stepper__right"
}, ce = { class: "m-stepper__content__wrapper" }, ue = /* @__PURE__ */ J({
  __name: "MazStepper",
  props: {
    modelValue: { default: void 0 },
    steps: { default: void 0 },
    color: { default: "primary" },
    disabledNextSteps: { type: Boolean },
    disabledPreviousSteps: { type: Boolean },
    autoValidateSteps: { type: Boolean },
    allStepsOpened: { type: Boolean },
    allStepsValidated: { type: Boolean },
    canCloseSteps: { type: Boolean }
  },
  emits: ["update:model-value"],
  setup(m, { emit: $ }) {
    K((t) => ({
      "13f82dd2": R.value,
      e4bbae2c: q.value
    }));
    const n = m, C = $, k = f(() => import("./MazIcon-BDGxCy9M.mjs")), W = f(() => import("./MazExpandAnimation-ChSSdQAq.mjs")), T = f(() => import("./check-circle-CtCMBYyp.mjs")), j = f(
      () => import("./exclamation-circle-DzBJ8vGa.mjs")
    ), L = f(() => import("./exclamation-triangle-BfZLmRbP.mjs")), R = h(() => `var(--maz-color-${n.color})`), q = h(() => `var(--maz-color-${n.color}-contrast)`), z = Q(), S = h(
      () => Object.keys(z).filter((t) => t.startsWith("content-")).length
    ), E = U(1), s = h({
      get: () => n.modelValue ?? E.value,
      set: (t) => {
        E.value = t, C("update:model-value", t);
      }
    });
    function I(t) {
      return N(t) ? { icon: T, class: "--success" } : O(t) ? { icon: L, class: "--warning" } : D(t) ? { icon: j, class: "--error" } : { class: "--normal" };
    }
    function p(t) {
      var o, e;
      return (e = (o = n.steps) == null ? void 0 : o[t - 1]) == null ? void 0 : e.icon;
    }
    function V(t, o) {
      var e, i;
      return (i = (e = n.steps) == null ? void 0 : e[o - 1]) == null ? void 0 : i[t];
    }
    function x(t) {
      s.value === t && n.canCloseSteps ? s.value = 0 : t < 1 ? s.value = 1 : t > S.value ? s.value = S.value : s.value = t;
    }
    function B(t, o) {
      var r, _;
      const e = t === "titleInfo" ? "title-info" : t, i = Object.keys(z).filter((H) => H.startsWith(`${e}-`)).includes(`${e}-${o}`), d = !!((_ = (r = n.steps) == null ? void 0 : r[o - 1]) != null && _[t]);
      return i || d;
    }
    function N(t) {
      var d, r;
      const o = (r = (d = n.steps) == null ? void 0 : d[t - 1]) == null ? void 0 : r.success, e = D(t) || O(t), i = n.autoValidateSteps && t < s.value && !e;
      return o ?? (i || n.allStepsValidated);
    }
    function F(t) {
      var r, _;
      const o = (_ = (r = n.steps) == null ? void 0 : r[t - 1]) == null ? void 0 : _.disabled, e = s.value === t && !n.canCloseSteps, i = n.disabledNextSteps && t > s.value, d = n.disabledPreviousSteps && t < s.value;
      return o ?? (e || i || d || n.allStepsOpened);
    }
    function D(t) {
      var o, e;
      return (e = (o = n.steps) == null ? void 0 : o[t - 1]) == null ? void 0 : e.error;
    }
    function O(t) {
      var o, e;
      return (e = (o = n.steps) == null ? void 0 : o[t - 1]) == null ? void 0 : e.warning;
    }
    function G(t) {
      return t === S.value;
    }
    return (t, o) => (a(), c("div", ee, [
      (a(!0), c(
        b,
        null,
        X(S.value, (e) => (a(), c(
          b,
          { key: e },
          [
            B("title", e) ? (a(), c("button", {
              key: 0,
              id: `header-step-${e}`,
              type: "button",
              disabled: F(e),
              class: M(["m-stepper__header", [
                {
                  "--is-current-step": e === s.value || t.allStepsOpened,
                  "--disabled": e !== s.value && !t.allStepsOpened && F(e)
                },
                `${I(e).class}`
              ]]),
              onClick: (i) => x(e)
            }, [
              u("div", ne, [
                u("span", oe, [
                  u("div", se, [
                    I(e).icon ? (a(), w(A(I(e).icon), {
                      key: 0,
                      class: "icon maz-text-xl"
                    })) : l("v-if", !0)
                  ]),
                  l(`
              @slot icon-\${step} - Replace step number in the circle by an icon for the step
            `),
                  v(t.$slots, `icon-${e}`, {}, () => [
                    p(e) ? (a(), c(
                      b,
                      { key: 0 },
                      [
                        typeof p(e) == "string" ? (a(), w(P(k), {
                          key: 0,
                          name: p(e)
                        }, null, 8, ["name"])) : p(e) ? (a(), w(A(p(e)), { key: 1 })) : l("v-if", !0)
                      ],
                      64
                      /* STABLE_FRAGMENT */
                    )) : (a(), c(
                      b,
                      { key: 1 },
                      [
                        y(
                          g(e),
                          1
                          /* TEXT */
                        )
                      ],
                      64
                      /* STABLE_FRAGMENT */
                    ))
                  ], !0)
                ]),
                u("div", ae, [
                  u("span", le, [
                    l(`
                @slot title-\${step} - Title of the step
              `),
                    v(t.$slots, `title-${e}`, {}, () => [
                      y(
                        g(V("title", e)),
                        1
                        /* TEXT */
                      )
                    ], !0)
                  ]),
                  B("subtitle", e) ? (a(), c("span", ie, [
                    l(`
                @slot title-\${step} - Subtitle of the step
              `),
                    v(t.$slots, `subtitle-${e}`, {}, () => [
                      y(
                        g(V("subtitle", e)),
                        1
                        /* TEXT */
                      )
                    ], !0)
                  ])) : l("v-if", !0)
                ])
              ]),
              B("titleInfo", e) ? (a(), c("span", re, [
                l(`
            @slot title-info-\${step} - Info of the right of the step
          `),
                v(t.$slots, `title-info-${e}`, {}, () => [
                  y(
                    g(V("titleInfo", e)),
                    1
                    /* TEXT */
                  )
                ], !0)
              ])) : l("v-if", !0)
            ], 10, te)) : l("v-if", !0),
            u(
              "div",
              {
                class: M(["m-stepper__content", {
                  "--no-border": G(e)
                }])
              },
              [
                Y(P(W), {
                  "model-value": t.allStepsOpened || s.value === e,
                  "aria-labelledby": `header-step-${e}`
                }, {
                  default: Z(() => [
                    u("div", ce, [
                      l(` @slot content-\${step} - Content of the step
                @binding {boolean} validated - If the step is validated
                @binding {boolean} error - If the step has an error
                @binding {boolean} warning - If the step has a warning
                @binding {Function} previous-step - Function to go to the previous step
                @binding {Function} next-step - Function to go to the next step
              `),
                      v(t.$slots, `content-${e}`, {
                        validated: N(e),
                        error: D(e),
                        warning: O(e),
                        nextStep: () => x(e + 1),
                        previousStep: () => x(e - 1)
                      }, void 0, !0)
                    ])
                  ]),
                  _: 2
                  /* DYNAMIC */
                }, 1032, ["model-value", "aria-labelledby"])
              ],
              2
              /* CLASS */
            )
          ],
          64
          /* STABLE_FRAGMENT */
        ))),
        128
        /* KEYED_FRAGMENT */
      ))
    ]));
  }
}), de = (m, $) => {
  const n = m.__vccOpts || m;
  for (const [C, k] of $)
    n[C] = k;
  return n;
}, _e = /* @__PURE__ */ de(ue, [["__scopeId", "data-v-c6a6f6ee"]]);
export {
  _e as M,
  de as _
};
