// maz-ui/components/chunks/MazTransitionExpand-CYyEf_hd.mjs
import { defineComponent as e, useCssVars as p, openBlock as d, createBlock as h, TransitionGroup as u, withCtx as y, renderSlot as g, nextTick as s } from "vue";
import { _ as l } from "./MazCard-BBwohtCJ.mjs";
import '../assets/MazTransitionExpand.css';const f = /* @__PURE__ */ e({
  __name: "MazTransitionExpand",
  props: {
    animationDuration: { default: "300ms" }
  },
  setup(x) {
    p((t) => ({
      "366a0c86": t.animationDuration
    }));
    function n(t) {
      const i = getComputedStyle(t).width;
      t.style.width = i, t.style.position = "absolute", t.style.visibility = "hidden", t.style.height = "auto", t.style.top = "bottom";
      const o = getComputedStyle(t).height;
      t.style.width = "", t.style.position = "", t.style.visibility = "", t.style.height = "0px", getComputedStyle(t).height, s(() => {
        t.style.height = o;
      });
    }
    function a(t) {
      t.style.height = "auto";
    }
    function r(t) {
      const i = getComputedStyle(t).height;
      t.style.height = i, getComputedStyle(t).height, s(() => {
        t.style.height = "0px";
      });
    }
    return (t, i) => (d(), h(u, {
      class: "m-transition-expand",
      name: "maz-expand",
      tag: "div",
      onEnter: i[0] || (i[0] = (o) => n(o)),
      onAfterEnter: i[1] || (i[1] = (o) => a(o)),
      onLeave: i[2] || (i[2] = (o) => r(o))
    }, {
      default: y(() => [
        g(t.$slots, "default", {}, void 0, !0)
      ]),
      _: 3
      /* FORWARDED */
    }));
  }
}), v = /* @__PURE__ */ l(f, [["__scopeId", "data-v-4be123cd"]]);
export {
  v as default
};
