// maz-ui/components/chunks/MazTabsBar-xcIIXHx5.mjs
import { inject as I, ref as i, defineComponent as N, defineAsyncComponent as O, computed as V, onBeforeMount as W, onMounted as z, watch as j, openBlock as f, createElementBlock as m, normalizeClass as p, createElementVNode as q, normalizeStyle as B, Fragment as H, renderList as U, createCommentVNode as g, renderSlot as S, createTextVNode as C, toDisplayString as T, createBlock as Q, unref as X, withCtx as D } from "vue";
import '../assets/MazTabsBar.css';function F(r, n) {
  const l = I(r, n);
  if (!l)
    throw new TypeError(`[maz-ui](injectStrict) Could not resolve ${r}`);
  return l;
}
function x(r) {
  return new Promise((n) => setTimeout(n, r));
}
const G = (r, n) => {
  const l = r.__vccOpts || r;
  for (const [o, c] of n)
    l[o] = c;
  return l;
};
i("system");
i("system");
const J = ["disabled", "onClick"], K = /* @__PURE__ */ N({
  __name: "MazTabsBar",
  props: {
    items: {},
    persistent: { type: Boolean },
    queryParam: { default: "tab" },
    block: { type: Boolean },
    noElevation: { type: Boolean },
    autoScroll: { type: Boolean, default: !0 }
  },
  setup(r) {
    const n = r, l = O(() => import("./MazBadge-CwA0md19.mjs")), { currentTab: o, updateCurrentTab: c } = F("maz-tabs");
    function k(e) {
      c(e + 1), n.persistent && M(e + 1);
    }
    const v = i(), h = i([]);
    function y(e) {
      return o.value === e + 1;
    }
    function L({
      mazBtn: e,
      index: t
    }) {
      h.value[t] = e && "$el" in e ? e.$el : e;
    }
    const d = V(
      () => n.items.map((e) => ({
        label: typeof e == "string" ? e : e.label,
        disabled: typeof e == "string" ? !1 : e.disabled ?? !1,
        badge: typeof e == "string" ? void 0 : e.badge
      }))
    ), _ = i(), b = i(!1);
    async function w() {
      if (!n.autoScroll)
        return;
      await x(150);
      const e = v.value, t = h.value[o.value - 1];
      if (!e || !t)
        return;
      const a = 50;
      if (t.offsetLeft - a < e.scrollLeft || t.offsetLeft + t.offsetWidth > e.scrollLeft + e.clientWidth) {
        const E = window.getComputedStyle(e, "padding-left").paddingLeft, R = Number(E.slice(0, -2));
        e.scrollTo({
          left: t.offsetLeft - R - a,
          behavior: b.value ? "smooth" : "instant"
        });
      }
      if (typeof o.value != "number")
        return;
      const s = (t == null ? void 0 : t.offsetWidth) ?? 0, u = (t == null ? void 0 : t.offsetHeight) ?? 0, A = (t == null ? void 0 : t.offsetLeft) ?? 0;
      _.value = {
        transform: `translateX(${A}px)`,
        width: `${s}px`,
        height: `${u}px`
      }, b.value = !0;
    }
    function P(e, t) {
      return t ? {} : o.value === e + 1 ? "color: var(--maz-color-text)" : "color: var(--maz-color-muted)";
    }
    W(() => {
      (o.value < 1 || o.value > d.value.length) && console.error(
        `[maz-ui](MazTabsBar) The model-value should be between 1 and ${d.value.length}`
      );
    }), z(() => {
      n.persistent && w();
    }), j(
      () => [o.value, d.value],
      () => {
        w();
      }
    );
    function $() {
      const e = new URL(window.location.href);
      return Number(e.searchParams.get(n.queryParam));
    }
    function M(e) {
      const t = new URL(window.location.href);
      t.searchParams.set(n.queryParam, String(e)), window.history.replaceState({}, document.title, t.toString());
    }
    return z(async () => {
      n.persistent && c($() || o.value || 1);
    }), (e, t) => (f(), m(
      "div",
      {
        ref_key: "tabsBarRef",
        ref: v,
        class: p(["m-tabs-bar", {
          "--block": e.block,
          "--elevation": !e.noElevation
        }])
      },
      [
        q(
          "div",
          {
            class: p(["m-tabs-bar__indicator", { "--animated": b.value }]),
            style: B([_.value])
          },
          null,
          6
          /* CLASS, STYLE */
        ),
        (f(!0), m(
          H,
          null,
          U(d.value, (a, s) => (f(), m("button", {
            key: s,
            ref_for: !0,
            ref: (u) => L({ mazBtn: u, index: s }),
            class: p([{ "--active": y(s), "--disabled": a.disabled }, "m-tabs-bar__item"]),
            disabled: a.disabled,
            style: B(P(s, a.disabled)),
            onClick: (u) => a.disabled ? void 0 : k(s)
          }, [
            g(`
          @slot item - Content of item to display in the tabs bar
            @binding {MazTabsBarItem[]} item - all data of the item
            @binding {boolean} active - \`true\` if the tab is active
            @binding {number} index - index of the item
        `),
            S(e.$slots, "item", {
              item: a,
              active: y(s),
              index: s
            }, () => [
              C(
                T(a.label) + " ",
                1
                /* TEXT */
              ),
              a.badge ? (f(), Q(X(l), {
                key: 0,
                color: a.badge.color,
                pastel: a.badge.pastel,
                outline: a.badge.outline,
                "rounded-size": a.badge.roundedSize,
                size: a.badge.size ?? "0.7rem",
                class: "m-tabs-bar__item__badge"
              }, {
                default: D(() => [
                  g(`
              @slot badge-content - Content in the badge
                @binding {string | number | boolean} content - content of the badge provided in item
            `),
                  S(e.$slots, "badge-content", {
                    content: a.badge.content
                  }, () => [
                    C(
                      T(a.badge.content),
                      1
                      /* TEXT */
                    )
                  ], !0)
                ]),
                _: 2
                /* DYNAMIC */
              }, 1032, ["color", "pastel", "outline", "rounded-size", "size"])) : g("v-if", !0)
            ], !0)
          ], 14, J))),
          128
          /* KEYED_FRAGMENT */
        ))
      ],
      2
      /* CLASS */
    ));
  }
}), Z = /* @__PURE__ */ G(K, [["__scopeId", "data-v-68598b04"]]);
export {
  Z as M,
  G as _
};
