// maz-ui/components/MazDrawer.mjs
import { defineComponent as V, ref as S, onMounted as g, watch as N, openBlock as r, createBlock as h, Teleport as T, createVNode as f, Transition as O, withCtx as v, createElementBlock as D, normalizeClass as d, createElementVNode as a, withModifiers as P, mergeProps as K, renderSlot as k, createCommentVNode as L, defineAsyncComponent as B, normalizeStyle as F, createTextVNode as q, toDisplayString as I, unref as z } from "vue";
import './assets/MazDrawer.css';const A = "--backdrop-present", U = /* @__PURE__ */ V({
  inheritAttrs: !1,
  __name: "MazBackdrop",
  props: {
    modelValue: { type: Boolean, default: !1 },
    teleportSelector: { default: "body" },
    beforeClose: { type: Function, default: void 0 },
    persistent: { type: Boolean, default: !1 },
    noCloseOnEscKey: { type: Boolean, default: !1 },
    transitionName: { default: "backdrop-anim" },
    backdropClass: { default: void 0 },
    backdropContentClass: { default: void 0 }
  },
  emits: ["open", "close", "update:model-value", "before-close"],
  setup(b, { expose: i, emit: c }) {
    const e = b, t = c;
    function n() {
      document.documentElement.classList.add(A);
    }
    async function M() {
      document.querySelector(".m-backdrop.--present") || document.documentElement.classList.remove(A);
    }
    const s = S(e.modelValue);
    function l() {
      C(!1);
    }
    async function C(o) {
      var p;
      o || (t("before-close"), await ((p = e.beforeClose) == null ? void 0 : p.call(e))), s.value = o;
    }
    function y() {
      t("open");
    }
    function E() {
      t("update:model-value", !1), t("close"), u();
    }
    function $() {
      e.persistent || l();
    }
    function m(o) {
      !e.noCloseOnEscKey && o.key === "Escape" && !e.persistent && l();
    }
    function w() {
      n(), document.addEventListener("keyup", m, !1);
    }
    function u() {
      document.removeEventListener("keyup", m), M();
    }
    return g(() => {
      e.modelValue ? w() : u();
    }), N(
      () => e.modelValue,
      (o) => {
        s.value = o, o ? w() : u();
      }
    ), i({
      onBackdropAnimationEnter: y,
      onBackdropAnimationLeave: E,
      onBackdropClicked: $,
      close: l,
      present: s,
      toggleModal: C,
      onKeyPress: m
    }), (o, p) => (r(), h(T, { to: o.teleportSelector }, [
      f(O, {
        appear: "",
        name: o.transitionName,
        onAfterEnter: y,
        onAfterLeave: E
      }, {
        default: v(() => [
          s.value ? (r(), D(
            "div",
            {
              key: 0,
              class: d(["m-backdrop --present", [o.backdropClass]]),
              tabindex: "-1",
              role: "dialog"
            },
            [
              a(
                "button",
                {
                  class: d(["m-backdrop-overlay", { "--disabled": o.persistent }]),
                  tabindex: "-1",
                  onClick: P($, ["self"])
                },
                null,
                2
                /* CLASS */
              ),
              a(
                "div",
                K({
                  class: ["m-backdrop-content", o.backdropContentClass]
                }, o.$attrs, {
                  role: "document",
                  tabindex: "0"
                }),
                [
                  k(o.$slots, "default", { close: l })
                ],
                16
                /* FULL_PROPS */
              )
            ],
            2
            /* CLASS */
          )) : L("v-if", !0)
        ]),
        _: 3
        /* FORWARDED */
      }, 8, ["name"])
    ], 8, ["to"]));
  }
}), X = { class: "m-drawer-header__title" }, j = {
  key: 0,
  class: "m-drawer-header__close"
}, G = { class: "m-drawer-body" }, J = /* @__PURE__ */ V({
  __name: "MazDrawer",
  props: {
    title: { default: void 0 },
    variant: { default: "right" },
    size: { default: "30rem" },
    backdropClass: { default: void 0 },
    noClose: { type: Boolean, default: !1 }
  },
  emits: ["before-close", "open", "close", "update:model-value"],
  setup(b) {
    const i = B(() => import("./chunks/MazBtn-BQ7aH6Qe.mjs").then((e) => e.M)), c = B(() => import("./chunks/x-mark-BP8zWe_C.mjs"));
    return (e, t) => (r(), h(U, {
      "backdrop-class": ["m-drawer", `--${e.variant}`, e.backdropClass],
      style: F({
        "--maz-drawer-size": e.size
      }),
      onClose: t[0] || (t[0] = (n) => e.$emit("close")),
      onOpen: t[1] || (t[1] = (n) => e.$emit("open")),
      onBeforeClose: t[2] || (t[2] = (n) => e.$emit("before-close")),
      "onUpdate:modelValue": t[3] || (t[3] = (n) => e.$emit("update:model-value", n))
    }, {
      default: v(({ close: n }) => [
        a(
          "div",
          {
            class: d(["m-drawer-content-wrap", [`--${e.variant}`]])
          },
          [
            a(
              "header",
              {
                class: d(["m-drawer-header", [e.$slots.title || e.title ? "--between" : "--end"]])
              },
              [
                a("h4", X, [
                  k(e.$slots, "title", { close: n }, () => [
                    q(
                      I(e.title),
                      1
                      /* TEXT */
                    )
                  ])
                ]),
                e.noClose ? L("v-if", !0) : (r(), D("div", j, [
                  f(z(i), {
                    size: "sm",
                    color: "transparent",
                    onClick: n
                  }, {
                    default: v(() => [
                      f(z(c), { class: "icon maz-text-lg" })
                    ]),
                    _: 2
                    /* DYNAMIC */
                  }, 1032, ["onClick"])
                ]))
              ],
              2
              /* CLASS */
            ),
            a("div", G, [
              k(e.$slots, "default", { close: n })
            ])
          ],
          2
          /* CLASS */
        )
      ]),
      _: 3
      /* FORWARDED */
    }, 8, ["backdrop-class", "style"]));
  }
});
export {
  J as default
};
