// maz-ui/components/MazExpandAnimation.mjs
import { defineComponent as d, mergeModels as u, useCssVars as m, useModel as f, ref as l, onMounted as c, onUnmounted as v, openBlock as p, createElementBlock as _, normalizeClass as E, createElementVNode as x, mergeProps as h, createCommentVNode as g, renderSlot as M } from "vue";
import './assets/MazExpandAnimation.css';const V = ["aria-hidden"], A = /* @__PURE__ */ d({
  inheritAttrs: !1,
  __name: "MazExpandAnimation",
  props: /* @__PURE__ */ u({
    duration: { default: "300ms" },
    timingFunction: { default: "ease-in-out" }
  }, {
    modelValue: { type: Boolean },
    modelModifiers: {}
  }),
  emits: ["update:modelValue"],
  setup(i) {
    m((e) => ({
      "3628b947": e.duration,
      "239277b5": e.timingFunction
    }));
    const n = f(i, "modelValue"), a = l(!n.value), t = l();
    function s() {
      n.value || (a.value = !0);
    }
    function r() {
      n.value && (a.value = !1);
    }
    return c(() => {
      var e, o;
      (e = t.value) == null || e.addEventListener("transitionstart", s, !1), (o = t.value) == null || o.addEventListener("transitionend", r, !1);
    }), v(() => {
      var e, o;
      (e = t.value) == null || e.removeEventListener("transitionstart", s, !1), (o = t.value) == null || o.removeEventListener("transitionend", r, !1);
    }), (e, o) => (p(), _("div", {
      ref_key: "expandAnimationRef",
      ref: t,
      class: E(["m-expand-animation", { "m-expand-animation--expanded": n.value }]),
      "aria-hidden": !n.value,
      role: "region"
    }, [
      x(
        "div",
        h({ class: "m-expand-animation__inner" }, e.$attrs, {
          class: {
            "--overflow-hidden": a.value
          }
        }),
        [
          g(" @slot Default Slot - Display content "),
          M(e.$slots, "default", {}, void 0, !0)
        ],
        16
        /* FULL_PROPS */
      )
    ], 10, V));
  }
}), b = (i, n) => {
  const a = i.__vccOpts || i;
  for (const [t, s] of n)
    a[t] = s;
  return a;
}, y = /* @__PURE__ */ b(A, [["__scopeId", "data-v-b7ca6b97"]]);
export {
  y as default
};
