// maz-ui/components/MazLink.mjs
import { defineComponent as p, defineAsyncComponent as s, openBlock as n, createBlock as l, resolveDynamicComponent as d, mergeProps as c, withCtx as h, createCommentVNode as t, renderSlot as r, unref as f, createVNode as y } from "vue";
import './assets/MazLink.css';const m = /* @__PURE__ */ p({
  __name: "MazLink",
  props: {
    id: { default: void 0 },
    title: { default: void 0 },
    href: { default: "#" },
    to: { default: void 0 },
    color: { default: "primary" },
    target: { default: "_slef" },
    download: { default: void 0 },
    rel: { default: void 0 },
    ariaLabel: { default: void 0 },
    underline: { type: Boolean, default: !1 },
    underlineOnlyHover: { type: Boolean, default: !0 },
    autoExternal: { type: Boolean },
    leftIcon: { type: [String, Function, Object], default: void 0 },
    rightIcon: { type: [String, Function, Object], default: void 0 }
  },
  setup(a) {
    const i = s(
      () => import("./chunks/arrow-top-right-on-square-CkcBq4qD.mjs")
    ), o = s(() => import("./chunks/MazIcon-BDGxCy9M.mjs"));
    return (e, u) => (n(), l(d(e.to ? "router-link" : "a"), c({
      id: e.id,
      class: ["m-link", [
        {
          "--underline": e.underline,
          "--underline-only-hover": !e.underline && e.underlineOnlyHover
        },
        `--${e.color}`
      ]],
      to: e.to,
      href: e.href,
      title: e.title,
      target: e.target,
      rel: e.rel,
      download: e.download,
      "aria-label": e.ariaLabel
    }, e.$attrs), {
      default: h(() => [
        t(`
      @slot left-icon - The icon to display on the left of the text
    `),
        r(e.$slots, "left-icon", {}, () => [
          typeof e.leftIcon == "string" ? (n(), l(f(o), {
            key: 0,
            name: e.leftIcon
          }, null, 8, ["name"])) : e.leftIcon ? (n(), l(d(e.leftIcon), { key: 1 })) : t("v-if", !0)
        ], !0),
        t(`
      @slot Text of the link
    `),
        r(e.$slots, "default", {}, void 0, !0),
        t(`
      @slot right-icon - The icon to display on the left of the text
    `),
        r(e.$slots, "right-icon", {}, () => [
          typeof e.rightIcon == "string" ? (n(), l(f(o), {
            key: 0,
            name: e.rightIcon
          }, null, 8, ["name"])) : e.rightIcon ? (n(), l(d(e.rightIcon), { key: 1 })) : t("v-if", !0)
        ], !0),
        t(`
      @slot external-icon - Replace the default external icon
    `),
        e.autoExternal && e.target === "_blank" ? r(e.$slots, "external-icon", { key: 0 }, () => [
          y(f(i))
        ], !0) : t("v-if", !0)
      ]),
      _: 3
      /* FORWARDED */
    }, 16, ["id", "class", "to", "href", "title", "target", "rel", "download", "aria-label"]));
  }
}), v = (a, i) => {
  const o = a.__vccOpts || a;
  for (const [e, u] of i)
    o[e] = u;
  return o;
}, k = /* @__PURE__ */ v(m, [["__scopeId", "data-v-4f837a44"]]);
export {
  k as default
};
