// maz-ui/components/MazLoadingBar.mjs
import { defineComponent as r, useCssVars as s, computed as n, openBlock as l, createElementBlock as p, pushScopeId as d, popScopeId as _, createElementVNode as i } from "vue";
import './assets/MazLoadingBar.css';const m = (o) => (d("data-v-be315052"), o = o(), _(), o), u = { class: "m-loading-bar" }, h = /* @__PURE__ */ m(() => /* @__PURE__ */ i(
  "div",
  null,
  null,
  -1
  /* HOISTED */
)), v = [
  h
], f = /* @__PURE__ */ r({
  __name: "MazLoadingBar",
  props: {
    color: { default: "primary" },
    height: { default: "0.125rem" }
  },
  setup(o) {
    s((t) => ({
      "6349f7b6": t.height,
      "02343323": e.value.alpha,
      c6029158: e.value.main
    }));
    const a = o, e = n(() => ({
      alpha: `var(--maz-color-${a.color}-alpha-20)`,
      main: `var(--maz-color-${a.color})`
    }));
    return (t, c) => (l(), p("div", u, v));
  }
}), g = (o, a) => {
  const e = o.__vccOpts || o;
  for (const [t, c] of a)
    e[t] = c;
  return e;
}, z = /* @__PURE__ */ g(f, [["__scopeId", "data-v-be315052"]]);
export {
  z as default
};
