// maz-ui/components/MazSlider.mjs
import { defineComponent as te, ref as y, computed as f, watch as F, onMounted as ne, onBeforeUnmount as ae, nextTick as g, openBlock as b, createElementBlock as w, Fragment as T, createCommentVNode as oe, createElementVNode as _, normalizeStyle as B, normalizeClass as j, renderList as K, toDisplayString as ue } from "vue";
import './assets/MazSlider.css';function le(o, l) {
  let n;
  return function(...c) {
    clearTimeout(n), n = setTimeout(() => {
      o.apply(this, c);
    }, l);
  };
}
function se(o) {
  const l = document.documentElement, n = document.body, c = o.getBoundingClientRect();
  return {
    y: c.top + (window.pageYOffset || l.scrollTop) - (l.clientTop || n.clientTop || 0),
    x: c.left + (window.pageXOffset || l.scrollLeft) - (l.clientLeft || n.clientLeft || 0)
  };
}
function re(o, l, n = !1) {
  const c = "targetTouches" in o ? o.targetTouches[0] : o, v = se(l), i = {
    x: c.pageX - v.x,
    y: c.pageY - v.y
  };
  return {
    x: n ? l.offsetWidth - i.x : i.x,
    y: n ? l.offsetHeight - i.y : i.y
  };
}
function q(o, l, n, c) {
  return c === "minus" ? l ? o >= l : !0 : n ? o <= n : !0;
}
function I(o, l, n) {
  const c = o + 1, i = l < c ? c - l : l - c;
  return 100 / n * i / 100;
}
const ie = ["aria-valuenow", "aria-valuemin", "aria-valuemax"], ce = ["data-label", "onMousedownPassive", "onTouchstartPassive", "onFocusPassive", "onBlurPassive", "onTouchendPassive", "onKeydownPassive"], de = /* @__PURE__ */ te({
  __name: "MazSlider",
  props: {
    /** Array of cursors values */
    modelValue: {
      type: [Number, Array],
      required: !0,
      validator: (o) => ["number"].includes(typeof o) || Array.isArray(o) || o === null
    },
    /** array of cursors label */
    labels: { type: Array, default: void 0 },
    /** min value of sliders */
    min: { type: Number, default: 0 },
    /** max value of sliders */
    max: { type: Number, default: 100 },
    /** height size of slider bar */
    size: { type: String, default: void 0 },
    /** remove div in different colors */
    noDivider: { type: Boolean, default: !1 },
    /** become a logarithmic slider (exponential) */
    log: { type: Boolean, default: !1 },
    /** main slider color */
    color: {
      type: String,
      default: "primary"
    },
    /** disables cursor animation when active */
    noCursorAnim: { type: Boolean, default: !1 }
  },
  emits: ["update:model-value"],
  setup(o, { emit: l }) {
    const n = o, c = l, v = y(), i = y(), m = y(), d = y(), C = y([]), k = y([]), p = f(() => typeof n.modelValue == "number" ? [n.modelValue] : n.modelValue ? n.modelValue : [0]), M = f(() => Math.log(n.min || 1)), X = f(() => Math.log(n.max)), L = f(() => {
      const { min: e, max: t } = n;
      return (X.value - M.value) / (t - e);
    }), A = f(() => {
      const { min: e, max: t } = n;
      return t - e;
    }), Y = f(() => ({
      paddingTop: n.labels ? "2.5em" : "1em"
    })), H = f(() => Array.isArray(n.modelValue));
    F(
      () => n.modelValue,
      () => d.value = p.value,
      { immediate: !0 }
    ), F(
      () => [p.value, n.min, n.max, n.log].join(","),
      () => z(!0)
    );
    const $ = le(() => z(), 300);
    ne(() => {
      z(!0), window.addEventListener("resize", $);
    }), ae(() => {
      window.removeEventListener("resize", $);
    });
    async function z(e) {
      e === !0 && U(), await Q(), await g();
      for (const [t, s] of p.value.entries())
        P(t);
    }
    function R(e, t) {
      e.key === "ArrowLeft" ? d.value && q(d.value[t] - 1, d.value[t - 1], d.value[t + 1], "minus") && (d.value[t]--, x(d.value)) : e.key === "ArrowRight" && d.value && q(d.value[t] + 1, d.value[t - 1], d.value[t + 1], "plus") && (d.value[t]++, x(d.value));
    }
    function W(e) {
      i.value = void 0, D(e);
    }
    function U() {
      const { min: e, max: t } = n, s = p.value.map(
        (a) => a < e ? e : a > t ? t : a
      );
      x(s), d.value = s;
    }
    function x(e) {
      const t = H.value ? [...e] : e[0];
      c("update:model-value", t);
    }
    function G(e) {
      const { labels: t } = n;
      return t ? t[e] : void 0;
    }
    function P(e) {
      D(e), n.noDivider || J();
    }
    async function D(e) {
      var s;
      await g();
      const t = document.querySelectorAll(".m-slider .m-slider__btn");
      if (t) {
        const a = t[e], u = (a == null ? void 0 : a.clientWidth) + 16, r = {
          left: typeof ((s = m.value) == null ? void 0 : s[e]) == "number" ? `${m.value[e] - u / 2 || 1}px` : ""
        };
        C.value[e] = r;
      }
    }
    function J() {
      if (m.value) {
        const e = [...m.value];
        e.push(0);
        const t = e.length, s = Math.round(t / 2);
        k.value = e.map((a, u) => ({
          left: u === 0 ? 0 : `${e[u - 1]}px`,
          // ATTENTION: buttonPositions.value[i - 1]
          right: u + 1 === t ? 0 : `calc( 100% - ${a}px )`,
          backgroundColor: s === u + 1 ? void 0 : u < s ? `rgba(255, 255, 255, ${I(u, s, t)})` : `rgba(0, 0, 0, ${I(u, s, t)})`
          // darken
        }));
      }
    }
    async function Q() {
      var u, r;
      await g();
      const { min: e, max: t, log: s } = n, a = (u = v.value) == null ? void 0 : u.clientWidth;
      a ? m.value = (r = d.value) == null ? void 0 : r.map(
        (h) => s ? a / t * (e + (Math.log(h) - M.value) / L.value) : a / A.value * (h - e)
      ) : console.warn("[maz-ui][MazSlider] ref component not found");
    }
    async function E() {
      var u, r, h;
      await g();
      const { max: e, min: t, log: s } = n, a = (u = v.value) == null ? void 0 : u.clientWidth;
      if (a)
        return s ? (r = m.value) == null ? void 0 : r.map((V) => {
          const Z = V / (a / e), ee = Math.exp((Z - t) * L.value + M.value);
          return Math.round(ee);
        }) : (h = m.value) == null ? void 0 : h.map(
          (V) => Math.round(V / (a / A.value)) + t
        );
      console.warn("[maz-ui][MazSlider] ref component not found");
    }
    function S(e, t) {
      i.value === void 0 && (i.value = t, P(t));
    }
    async function O() {
      if (i.value === void 0)
        return;
      const e = await E();
      e && x(e), i.value = void 0;
    }
    async function N(e) {
      var s;
      if (await g(), i.value === void 0)
        return;
      const t = (s = v.value) == null ? void 0 : s.clientWidth;
      if (m.value) {
        const a = m.value[i.value - 1] || 0, u = m.value[i.value + 1] || t;
        if (!v.value)
          return console.error("[maz-ui](MazSlider/handleMousemove) MazSlider not available");
        m.value[i.value] = (() => {
          const r = re(e, v.value).x;
          return r < a ? a : u && r > u ? u : r;
        })(), d.value = await E(), P(i.value);
      }
    }
    return (e, t) => (b(), w(
      T,
      null,
      [
        oe(" eslint-disable vuejs-accessibility/mouse-events-have-key-events "),
        _(
          "div",
          {
            style: B([Y.value, { fontSize: o.size }]),
            class: j(["m-slider", [`m-slider--${o.color}`]]),
            role: "button",
            tabindex: "-1",
            onMousemovePassive: N,
            onMouseupPassive: O,
            onMouseleavePassive: O,
            onTouchmovePassive: N
          },
          [
            _("div", {
              ref_key: "MazSlider",
              ref: v,
              class: "m-slider__bar",
              role: "slider",
              "aria-valuenow": o.modelValue.toString(),
              "aria-valuemin": o.min,
              "aria-valuemax": o.max
            }, [
              (b(!0), w(
                T,
                null,
                K(k.value, (s, a) => (b(), w(
                  "div",
                  {
                    key: `divider-${a}`,
                    style: B([s]),
                    class: "m-slider__divider"
                  },
                  null,
                  4
                  /* STYLE */
                ))),
                128
                /* KEYED_FRAGMENT */
              )),
              (b(!0), w(
                T,
                null,
                K(p.value, (s, a) => {
                  var u;
                  return b(), w("button", {
                    key: `cursor-${a}`,
                    type: "button",
                    "data-label": G(a),
                    class: j(["m-slider__btn", {
                      "active-cursor": a === i.value && !o.noCursorAnim
                    }]),
                    style: B([C.value[a]]),
                    onMousedownPassive: (r) => S(r, a),
                    onTouchstartPassive: (r) => S(r, a),
                    onFocusPassive: (r) => S(r, a),
                    onBlurPassive: (r) => W(a),
                    onTouchendPassive: (r) => W(a),
                    onKeydownPassive: (r) => R(r, a)
                  }, [
                    _(
                      "span",
                      null,
                      ue((u = d.value) == null ? void 0 : u[a]),
                      1
                      /* TEXT */
                    )
                  ], 46, ce);
                }),
                128
                /* KEYED_FRAGMENT */
              ))
            ], 8, ie)
          ],
          38
          /* CLASS, STYLE, NEED_HYDRATION */
        )
      ],
      2112
      /* STABLE_FRAGMENT, DEV_ROOT_FRAGMENT */
    ));
  }
}), ve = (o, l) => {
  const n = o.__vccOpts || o;
  for (const [c, v] of l)
    n[c] = v;
  return n;
}, fe = /* @__PURE__ */ ve(de, [["__scopeId", "data-v-433a5afb"]]);
export {
  fe as default
};
