// maz-ui/components/MazTabsContentItem.mjs
import { inject as p, defineComponent as d, ref as c, watch as f, computed as i, openBlock as l, createBlock as v, Transition as _, withCtx as T, withDirectives as z, createElementVNode as C, renderSlot as w, vShow as h } from "vue";
import './assets/MazTabsContentItem.css';function y(e, n) {
  const t = p(e, n);
  if (!t)
    throw new TypeError(`[maz-ui](injectStrict) Could not resolve ${e}`);
  return t;
}
const I = /* @__PURE__ */ d({
  __name: "MazTabsContentItem",
  props: {
    tab: { type: Number, required: !0 }
  },
  setup(e) {
    const n = e, t = c("maz-tab-transition"), o = c(), { currentTab: r } = y("maz-tabs");
    f(
      () => r.value,
      async (a, s) => {
        if (typeof a == "number" && typeof s == "number") {
          const b = s < a;
          t.value = b ? "maz-tab-transition" : "maz-tab-reverse-transition";
        }
      },
      { immediate: !0 }
    );
    const m = i(() => n.tab - 1), u = i(() => r.value - 1 === m.value);
    return (a, s) => (l(), v(_, {
      name: t.value,
      persisted: ""
    }, {
      default: T(() => [
        z(C(
          "div",
          {
            ref_key: "MazTabsContentItem",
            ref: o,
            class: "m-tabs-content-item"
          },
          [
            w(a.$slots, "default", {}, void 0, !0)
          ],
          512
          /* NEED_PATCH */
        ), [
          [h, u.value]
        ])
      ]),
      _: 3
      /* FORWARDED */
    }, 8, ["name"]));
  }
}), M = (e, n) => {
  const t = e.__vccOpts || e;
  for (const [o, r] of n)
    t[o] = r;
  return t;
}, S = /* @__PURE__ */ M(I, [["__scopeId", "data-v-7d9ca863"]]);
export {
  S as default
};
